diff --git a/src/core/qgsapplication.cpp b/src/core/qgsapplication.cpp
index 8ff69f0..dd73661 100644
--- a/src/core/qgsapplication.cpp
+++ b/src/core/qgsapplication.cpp
@@ -60,7 +60,6 @@
 #include "qgsbookmarkmanager.h"
 #include "qgsstylemodel.h"
 
-#include "gps/qgsgpsconnectionregistry.h"
 #include "processing/qgsprocessingregistry.h"
 
 #include "layout/qgspagesizeregistry.h"
@@ -2142,11 +2141,6 @@ QgsLayoutItemRegistry *QgsApplication::layoutItemRegistry()
   return members()->mLayoutItemRegistry;
 }
 
-QgsGpsConnectionRegistry *QgsApplication::gpsConnectionRegistry()
-{
-  return members()->mGpsConnectionRegistry;
-}
-
 QgsPluginLayerRegistry *QgsApplication::pluginLayerRegistry()
 {
   return members()->mPluginLayerRegistry;
@@ -2225,7 +2219,6 @@ QgsApplication::ApplicationMembers::ApplicationMembers()
   mCalloutRegistry = new QgsCalloutRegistry();
   mRendererRegistry = new QgsRendererRegistry();
   mRasterRendererRegistry = new QgsRasterRendererRegistry();
-  mGpsConnectionRegistry = new QgsGpsConnectionRegistry();
   mPluginLayerRegistry = new QgsPluginLayerRegistry();
   mProcessingRegistry = new QgsProcessingRegistry();
   mPageSizeRegistry = new QgsPageSizeRegistry();
@@ -2249,7 +2242,6 @@ QgsApplication::ApplicationMembers::~ApplicationMembers()
   delete mAnnotationRegistry;
   delete mColorSchemeRegistry;
   delete mFieldFormatterRegistry;
-  delete mGpsConnectionRegistry;
   delete mMessageLog;
   delete mPaintEffectRegistry;
   delete mPluginLayerRegistry;
