--- a/src/3d/qgspointcloudlayer3drenderer.cpp	2025-07-15 20:09:37.389537550 +0300
+++ b/src/3d/qgspointcloudlayer3drenderer.cpp	2025-07-30 19:58:18.106960871 +0300
@@ -21,7 +21,7 @@
 
 #include "qgspointcloudindex.h"
 #include "qgspointcloudlayer.h"
-#include "qgsvirtualpointcloudentity_p.h"
+// #include "qgsvirtualpointcloudentity_p.h"
 #include "qgsxmlutils.h"
 #include "qgsapplication.h"
 #include "qgs3dsymbolregistry.h"
@@ -161,11 +161,14 @@
   const QgsCoordinateTransform coordinateTransform( pcl->crs(), map.crs(), map.transformContext() );
 
   Qt3DCore::QEntity *entity = nullptr;
+  /*
   if ( pcl->dataProvider()->index() )
   {
+  */
     entity = new QgsPointCloudLayerChunkedEntity( pcl->dataProvider()->index(), map, coordinateTransform, dynamic_cast<QgsPointCloud3DSymbol *>( mSymbol->clone() ), maximumScreenError(), showBoundingBoxes(),
         static_cast< const QgsPointCloudLayerElevationProperties * >( pcl->elevationProperties() )->zScale(),
         static_cast< const QgsPointCloudLayerElevationProperties * >( pcl->elevationProperties() )->zOffset(), mPointBudget );
+  /*
   }
   else if ( !pcl->dataProvider()->subIndexes().isEmpty() )
   {
@@ -173,6 +176,7 @@
         static_cast< const QgsPointCloudLayerElevationProperties * >( pcl->elevationProperties() )->zScale(),
         static_cast< const QgsPointCloudLayerElevationProperties * >( pcl->elevationProperties() )->zOffset(), mPointBudget );
   }
+  */
   return entity;
 }
 
