--- a/src/3d/qgspointcloudlayer3drenderer.cpp	2025-05-15 15:26:35.343376278 +0300
+++ b/src/3d/qgspointcloudlayer3drenderer.cpp	2025-05-15 14:47:00.479284362 +0300
@@ -20,7 +20,7 @@
 
 #include "qgspointcloudindex.h"
 #include "qgspointcloudlayer.h"
-#include "qgsvirtualpointcloudentity_p.h"
+//#include "qgsvirtualpointcloudentity_p.h"
 #include "qgsxmlutils.h"
 #include "qgs3dsymbolregistry.h"
 #include "qgspointcloud3dsymbol.h"
@@ -159,7 +159,7 @@
   const QgsCoordinateTransform coordinateTransform( pcl->crs3D(), map->crs(), map->transformContext() );
 
   Qt3DCore::QEntity *entity = nullptr;
-  if ( pcl->dataProvider()->index() )
+  if ( true || pcl->dataProvider()->index() )
   {
     entity = new QgsPointCloudLayerChunkedEntity( map, pcl->dataProvider()->index(), coordinateTransform, dynamic_cast<QgsPointCloud3DSymbol *>( mSymbol->clone() ), static_cast<float>( maximumScreenError() ), showBoundingBoxes(), static_cast<const QgsPointCloudLayerElevationProperties *>( pcl->elevationProperties() )->zScale(), static_cast<const QgsPointCloudLayerElevationProperties *>( pcl->elevationProperties() )->zOffset(), mPointBudget );
   }
