--- a/src/app/layers/qgsapplayerhandling.cpp	2025-07-15 20:09:37.484539579 +0300
+++ b/src/app/layers/qgsapplayerhandling.cpp	2025-07-30 19:58:18.107960906 +0300
@@ -41,7 +41,7 @@
 #include "qgsgui.h"
 #include "qgsmessagelog.h"
 #include "qgsapplication.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsprojectstorageregistry.h"
 #include "qgsprojectstorage.h"
 #include "qgsmaplayerfactory.h"
@@ -562,6 +562,9 @@
 
   QgsDebugMsgLevel( "completeBaseName: " + base, 2 );
 
+  return nullptr;
+
+  /*
   // create the layer
   const QgsVectorTileLayer::LayerOptions options( QgsProject::instance()->transformContext() );
   std::unique_ptr<QgsVectorTileLayer> layer( new QgsVectorTileLayer( uri, base, options ) );
@@ -584,6 +587,7 @@
   QgisApp::instance()->activateDeactivateLayerRelatedActions( QgisApp::instance()->activeLayer() );
 
   return layer.release();
+  */
 }
 
 bool QgsAppLayerHandling::askUserForZipItemLayers( const QString &path, const QList<Qgis::LayerType> &acceptableTypes )
