--- a/src/app/qgspluginregistry.cpp	2025-05-15 15:26:35.397914871 +0300
+++ b/src/app/qgspluginregistry.cpp	2025-05-15 14:47:00.628289558 +0300
@@ -568,6 +568,11 @@
     corePlugins << QStringLiteral( "MetaSearch" );
     corePlugins << QStringLiteral( "grassprovider" );
 
+    // NextGIS plugins
+    corePlugins << "quick_map_services";
+    corePlugins << "nextgis_connect";
+    corePlugins << "ngq_rosreestr_tools";
+
     // make the required core plugins enabled by default:
     const auto constCorePlugins = corePlugins;
     for ( const QString &corePlugin : constCorePlugins )
