--- a/src/app/qgisapp.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/app/qgisapp.h	2022-04-06 18:40:09.000000000 +0300
@@ -95,7 +95,7 @@
 class QgsUserInputWidget;
 class QgsVectorLayer;
 class QgsVectorLayerTools;
-class QgsVectorTileLayer;
+// class QgsVectorTileLayer;
 class QgsWelcomePage;
 class QgsOptionsWidgetFactory;
 class QgsStatusBar;
@@ -106,7 +106,7 @@
 class QgsUserProfileManagerWidgetFactory;
 class QgsHandleBadLayersHandler;
 class QgsNetworkAccessManager;
-class QgsGpsConnection;
+// class QgsGpsConnection;
 class QgsApplicationExitBlockerInterface;
 class QgsAbstractMapToolHandler;
 class QgsAppMapTools;
@@ -120,7 +120,7 @@
 
 class QgsBrowserDockWidget;
 class QgsAdvancedDigitizingDockWidget;
-class QgsGpsInformationWidget;
+// class QgsGpsInformationWidget;
 class QgsStatisticalSummaryDockWidget;
 class QgsMapCanvasTracer;
 class QgsTemporalControllerDockWidget;
@@ -140,7 +140,7 @@
 class QgsDataSourceManagerDialog;
 class QgsBrowserGuiModel;
 class QgsBrowserModel;
-class QgsGeoCmsProviderRegistry;
+// class QgsGeoCmsProviderRegistry;
 class QgsLayoutCustomDropHandler;
 class QgsProxyProgressTask;
 class QgsNetworkRequestParameters;
@@ -593,7 +593,7 @@
 #endif
     QAction *actionAddWmsLayer() { return mActionAddWmsLayer; }
     QAction *actionAddXyzLayer() { return mActionAddXyzLayer; }
-    QAction *actionAddVectorTileLayer() { return mActionAddVectorTileLayer; }
+    // QAction *actionAddVectorTileLayer() { return mActionAddVectorTileLayer; }
     QAction *actionAddPointCloudLayer() { return mActionAddPointCloudLayer; }
     QAction *actionAddWcsLayer() { return mActionAddWcsLayer; }
 #ifdef HAVE_SPATIALITE
@@ -908,7 +908,7 @@
      * Any existing GPS connection used by the widget will be disconnect and replaced with this connection. The connection
      * is automatically registered within the QgsApplication::gpsConnectionRegistry().
      */
-    void setGpsPanelConnection( QgsGpsConnection *connection );
+    // void setGpsPanelConnection( QgsGpsConnection *connection );
 
   public slots:
     //! save current vector layer
@@ -1203,7 +1203,7 @@
      * in the Map Legend so it should be formed in a meaningful way.
      * \since QGIS 3.14
      */
-    QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName );
+    // QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName );
 
     /**
      * Adds a vector tile layer directly without prompting user for location
@@ -1574,9 +1574,9 @@
     //! Save project as
     void fileSaveAs();
     //! Export project in dxf format
-    void dxfExport();
+    // void dxfExport();
     //! Import layers in dwg format
-    void dwgImport();
+    // void dwgImport();
 
     /**
      * Open the project file corresponding to the
@@ -2168,7 +2168,7 @@
     void postProcessAddedLayer( QgsMapLayer *layer );
 
     //! Open a vector tile layer - this is the generic function which takes all parameters
-    QgsVectorTileLayer *addVectorTileLayerPrivate( const QString &uri, const QString &baseName, bool guiWarning = true );
+    // QgsVectorTileLayer *addVectorTileLayerPrivate( const QString &uri, const QString &baseName, bool guiWarning = true );
 
     //! Open a point cloud layer - this is the generic function which takes all parameters
     QgsPointCloudLayer *addPointCloudLayerPrivate( const QString &uri,
@@ -2661,7 +2661,7 @@
     QList<QgsDecorationItem *> mDecorationItems;
 
     //! Persistent GPS toolbox
-    QgsGpsInformationWidget *mpGpsWidget = nullptr;
+    // QgsGpsInformationWidget *mpGpsWidget = nullptr;
 
     QgsMessageBarItem *mLastMapToolMessage = nullptr;
 
