--- a/src/app/mainwin.cpp	2025-07-15 20:09:37.489539685 +0300
+++ b/src/app/mainwin.cpp	2025-08-18 15:05:37.440369285 +0300
@@ -20,6 +20,7 @@
 #include <fstream>
 #include <list>
 #include <memory>
+#include <filesystem>
 
 void showError( std::string message, std::string title )
 {
@@ -50,6 +51,72 @@
   return basename;
 }
 
+std::string getQgisRootPath()
+{
+    const auto baseName = moduleExeBaseName();
+    const auto binPathPos = baseName.rfind("bin");
+    if (binPathPos != std::string::npos)
+    {
+        return baseName.substr(0, binPathPos);
+    }
+    return {};
+}
+
+
+bool isRunningFromBuildDir()
+{
+    const auto qgisRootPath = getQgisRootPath();
+    bool result = false;
+    if (!qgisRootPath.empty())
+    {
+        std::ifstream f(qgisRootPath + "/qgisbuildpath.txt");
+        result = f.good();
+    }
+    return result;
+}
+
+
+std::string pythonPathEnvString()
+{
+  std::string pythonPathEnv;
+  const auto qgisRootPath = getQgisRootPath();
+  if (qgisRootPath.empty())
+  {
+      return pythonPathEnv;
+  }
+
+  if (!isRunningFromBuildDir())
+  {
+      pythonPathEnv = "PYTHONPATH=";
+      pythonPathEnv += qgisRootPath + "lib\\python38;";
+      pythonPathEnv += qgisRootPath + "lib\\python38\\site-packages;";
+      pythonPathEnv += qgisRootPath + "lib\\python38\\lib-dynload;";
+  }
+  else
+  {
+      const auto baseName = moduleExeBaseName();
+
+      std::string binDir = baseName.substr(0, baseName.rfind('\\'));
+      std::error_code ec;
+
+      pythonPathEnv = "PYTHONPATH=";
+      for (const auto& entry : std::filesystem::recursive_directory_iterator(binDir, ec))
+      {
+          if (!entry.is_directory())
+              continue;
+
+          const std::string dirName = entry.path().filename().string();
+          if (dirName != "site-packages")
+              continue;
+
+          pythonPathEnv += entry.path().string() + ";";
+      }
+  }
+
+  return pythonPathEnv;
+}
+
+
 int CALLBACK WinMain( HINSTANCE /*hInstance*/, HINSTANCE /*hPrevInstance*/, LPSTR /*lpCmdLine*/, int /*nCmdShow*/ )
 {
   std::string exename( moduleExeBaseName() );
@@ -129,6 +196,13 @@
     return EXIT_FAILURE;
   }
 
+  const std::string& pythonPathEnv = pythonPathEnvString();
+  if (_putenv(pythonPathEnv.c_str()) < 0)
+  {
+      showError("Could not set PYTHONPATH environment variable", "Error loading QGIS");
+      return EXIT_FAILURE;
+  }
+
 #ifndef _MSC_VER // MinGW
 #pragma GCC diagnostic ignored "-Wcast-function-type"
 #endif
@@ -168,10 +242,10 @@
 
 
 #ifdef _MSC_VER
-  HINSTANCE hGetProcIDDLL = LoadLibrary( "qgis_app.dll" );
+  HINSTANCE hGetProcIDDLL = LoadLibrary( "ngqgis_app.dll" );
 #else
 // MinGW
-  HINSTANCE hGetProcIDDLL = LoadLibrary( "libqgis_app.dll" );
+  HINSTANCE hGetProcIDDLL = LoadLibrary( "libngqgis_app.dll" );
 #endif
 
   if ( !hGetProcIDDLL )
@@ -188,7 +262,7 @@
       0,
       NULL );
 
-    std::string message = "Could not load qgis_app.dll \n Windows Error: " + std::string( errorText )
+    std::string message = "Could not load ngqgis_app.dll \n Windows Error: " + std::string( errorText )
                           + "\n Help: \n\n Check " + basename + ".env for correct environment paths";
     showError( message, "Error loading QGIS" );
 
@@ -207,7 +281,7 @@
 
   if ( !realmain )
   {
-    showError( "Could not locate main function in qgis_app.dll", "Error loading QGIS" );
+    showError( "Could not locate main function in ngqgis_app.dll", "Error loading QGIS" );
     return EXIT_FAILURE;
   }
 
