--- a/src/app/qgslayerstylingwidget.cpp	2025-07-15 20:09:37.514540219 +0300
+++ b/src/app/qgslayerstylingwidget.cpp	2025-07-30 19:58:18.114015920 +0300
@@ -37,9 +37,11 @@
 #include "qgsmaplayerstylemanager.h"
 #include "qgsstyle.h"
 #include "qgsvectorlayer.h"
+/*
 #include "qgsvectortilelayer.h"
 #include "qgsvectortilebasiclabelingwidget.h"
 #include "qgsvectortilebasicrendererwidget.h"
+*/
 #include "qgsmeshlayer.h"
 #include "qgsproject.h"
 #include "qgsundowidget.h"
@@ -259,6 +261,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QListWidgetItem *symbolItem = new QListWidgetItem( QgsApplication::getThemeIcon( QStringLiteral( "propertyicons/symbology.svg" ) ), QString() );
       symbolItem->setData( Qt::UserRole, Symbology );
       symbolItem->setToolTip( tr( "Symbology" ) );
@@ -267,6 +270,7 @@
       labelItem->setData( Qt::UserRole, VectorLabeling );
       labelItem->setToolTip( tr( "Labels" ) );
       mOptionsListWidget->addItem( labelItem );
+      */
       break;
     }
 
@@ -685,6 +689,7 @@
 
       case Qgis::LayerType::VectorTile:
       {
+        /*
         QgsVectorTileLayer *vtLayer = qobject_cast<QgsVectorTileLayer *>( mCurrentLayer );
         switch ( row )
         {
@@ -707,6 +712,7 @@
           default:
             break;
         }
+        */
         break;
       }
 
