--- a/src/app/qgsmaptoolselectutils.cpp
+++ b/src/app/qgsmaptoolselectutils.cpp
@@ -21,7 +21,7 @@
 #include "qgsmessagebar.h"
 #include "qgsmapcanvas.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsvectorlayerfeatureiterator.h"
 #include "qgsfeature.h"
 #include "qgsgeometry.h"
@@ -50,14 +50,20 @@
     switch ( layer->type() )
     {
       case Qgis::LayerType::Vector:
-      case Qgis::LayerType::VectorTile:
-        // supported
-        break;
+        break;
+//      case (int)Qgis::LayerType::VectorTile:
+//        // supported
+//        break;
       case Qgis::LayerType::Raster:
+        break;
       case Qgis::LayerType::Plugin:
+        break;
       case Qgis::LayerType::Mesh:
+        break;
       case Qgis::LayerType::Annotation:
+        break;
       case Qgis::LayerType::PointCloud:
+        break;
       case Qgis::LayerType::Group:
         layer = nullptr; //not supported
         break;
@@ -119,10 +125,10 @@
         }
         break;
       }
-      case Qgis::LayerType::VectorTile:
-        // mixed layer type, so aim for somewhere between the vector layer polygon/point sizes
-        boxSize = 2;
-        break;
+//      case Qgis::LayerType::VectorTile:
+//        // mixed layer type, so aim for somewhere between the vector layer polygon/point sizes
+//        boxSize = 2;
+//        break;
 
       case Qgis::LayerType::Raster:
       case Qgis::LayerType::Plugin:
@@ -251,7 +257,7 @@
       vlayer->selectByIds( selectedFeatures, behavior );
       break;
     }
-
+/*
     case Qgis::LayerType::VectorTile:
     {
       QgsVectorTileLayer *vtLayer = qobject_cast< QgsVectorTileLayer *>( layer );
@@ -274,7 +280,7 @@
 
       vtLayer->selectByGeometry( selectGeomTrans, context, behavior, Qgis::SelectGeometryRelationship::Intersect, flags, &renderContext );
       break;
-    }
+    }*/
 
     case Qgis::LayerType::Raster:
     case Qgis::LayerType::Plugin:
@@ -309,7 +315,7 @@
       vLayer->selectByIds( selectedFeatures, selectBehavior );
       break;
     }
-
+/*
     case Qgis::LayerType::VectorTile:
     {
       QgsVectorTileLayer *vtLayer = qobject_cast< QgsVectorTileLayer * >( layer );
@@ -327,7 +333,7 @@
 
       vtLayer->selectByGeometry( selectGeomTrans, context, selectBehavior, doContains ? Qgis::SelectGeometryRelationship::Within : Qgis::SelectGeometryRelationship::Intersect, Qgis::SelectionFlags(), &renderContext );
       break;
-    }
+    }*/
 
     case Qgis::LayerType::Raster:
     case Qgis::LayerType::Plugin:

