--- a/src/app/qgisapp.cpp	2025-05-15 15:26:35.373721126 +0300
+++ b/src/app/qgisapp.cpp	2025-05-15 14:47:00.605288756 +0300
@@ -112,8 +112,8 @@
 #include "options/qgscodeeditoroptions.h"
 #include "options/qgselevationoptions.h"
 #include "options/qgsfontoptions.h"
-#include "options/qgsgpsdeviceoptions.h"
-#include "options/qgsgpsoptions.h"
+// #include "options/qgsgpsdeviceoptions.h"
+// #include "options/qgsgpsoptions.h"
 #include "options/qgsideoptions.h"
 #include "options/qgscustomprojectionoptions.h"
 #include "options/qgsrasterrenderingoptions.h"
@@ -241,9 +241,9 @@
 #include "qgsdatumtransformdialog.h"
 #include "qgsdoublespinbox.h"
 #include "qgsdockwidget.h"
-#include "qgsdxfexport.h"
-#include "qgsdxfexportdialog.h"
-#include "qgsdwgimportdialog.h"
+// #include "qgsdxfexport.h"
+// #include "qgsdxfexportdialog.h"
+// #include "qgsdwgimportdialog.h"
 #include "qgsdecorationtitle.h"
 #include "qgsdecorationcopyright.h"
 #include "qgsdecorationimage.h"
@@ -265,13 +265,13 @@
 #include "qgsformannotation.h"
 #include "qgsguiutils.h"
 #include "qgsprojectionselectiondialog.h"
-#include "qgsgpsinformationwidget.h"
-#include "qgsappgpsconnection.h"
-#include "qgsappgpsdigitizing.h"
-#include "qgsappgpslogging.h"
-#include "qgsappgpssettingsmenu.h"
-#include "qgsgpstoolbar.h"
-#include "qgsgpscanvasbridge.h"
+// #include "qgsgpsinformationwidget.h"
+// #include "qgsappgpsconnection.h"
+// #include "qgsappgpsdigitizing.h"
+// #include "qgsappgpslogging.h"
+// #include "qgsappgpssettingsmenu.h"
+// #include "qgsgpstoolbar.h"
+// #include "qgsgpscanvasbridge.h"
 #include "qgsguivectorlayertools.h"
 #include "qgslayerdefinition.h"
 #include "qgslayertree.h"
@@ -327,7 +327,7 @@
 #include "qgsmapoverviewcanvas.h"
 #include "qgsmapsettings.h"
 #include "qgsmaptip.h"
-#include "qgsmbtiles.h"
+// #include "qgsmbtiles.h"
 #include "qgsmergeattributesdialog.h"
 #include "qgsmessageviewer.h"
 #include "qgsmessagebar.h"
@@ -407,13 +407,14 @@
 #include "qgsvectorlayer.h"
 #include "qgsvectorlayerproperties.h"
 #include "qgsvectorlayerdigitizingproperties.h"
-#include "qgsvectortilelayer.h"
-#include "qgsvectortilelayerproperties.h"
+// #include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayerproperties.h"
 #include "qgspointcloudlayerproperties.h"
 #include "qgstiledscenelayerproperties.h"
 #include "qgsmapthemes.h"
 #include "qgsmessagelogviewer.h"
 #include "qgsmaplayeractionregistry.h"
+#include "actions/qgsmaplayeraction.h"
 #include "qgswelcomepage.h"
 #include "qgsrecentprojectsmenueventfilter.h"
 #include "qgsversioninfo.h"
@@ -1479,7 +1480,7 @@
 
   addDockWidget( Qt::LeftDockWidgetArea, mBookMarksDockWidget );
   mBookMarksDockWidget->hide();
-
+  /*
   // create the GPS tool on starting QGIS - this is like the browser
   mGpsConnection = new QgsAppGpsConnection( this );
   mGpsSettingsMenu = new QgsAppGpsSettingsMenu( this );
@@ -1531,7 +1532,7 @@
   mpGpsDock->setWidget( gpsStack );
   mpGpsDock->setToggleVisibilityAction( mGpsToolBar->showInfoAction() );
   mpGpsDock->hide();
-
+  */
 
   mLastMapToolMessage = nullptr;
 
@@ -1765,6 +1766,14 @@
   // do main window customization - after window state has been restored, before the window is shown
   startProfile( tr( "Update customization on main window" ) );
   QgsCustomization::instance()->updateMainWindow( mToolbarMenu, mPanelMenu );
+  mActionNewMeshLayer->setVisible(false);
+  mActionAddMeshLayer->setVisible(false);
+  mActionAddAfsLayer->setVisible(false);
+  mActionAddVectorTileLayer->setVisible(false);
+  mActionAddPointCloudLayer->setVisible(false);
+  mActionDxfExport->setVisible(false);
+  mActionDwgImport->setVisible(false);
+  getMeshMenu(QStringLiteral())->menuAction()->setVisible(false);
   endProfile();
 
   mSplash->showMessage( tr( "Populate saved styles" ), Qt::AlignHCenter | Qt::AlignBottom, splashTextColor );
@@ -1952,8 +1961,8 @@
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsRenderingOptionsFactory>() ) );
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsVectorRenderingOptionsFactory>() ) );
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsRasterRenderingOptionsFactory>() ) );
-  mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsGpsOptionsFactory>() ) );
-  mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsGpsDeviceOptionsFactory>() ) );
+  // mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsGpsOptionsFactory>() ) );
+  // mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsGpsDeviceOptionsFactory>() ) );
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsCustomProjectionOptionsFactory>() ) );
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsElevationOptionsFactory>() ) );
   mOptionWidgetFactories.emplace_back( QgsScopedOptionsWidgetFactory( std::make_unique<QgsFontOptionsFactory>() ) );
@@ -2110,7 +2119,7 @@
   mAdvancedDigitizingDockWidget = nullptr;
 
   delete mpMaptip;
-
+  /*
   delete mpGpsWidget;
   mpGpsWidget = nullptr;
 
@@ -2131,7 +2140,7 @@
 
   delete mGpsConnection;
   mGpsConnection = nullptr;
-
+  */
   delete mOverviewMapCursor;
 
   delete mTracer;
@@ -2487,6 +2496,7 @@
       if ( QgsMapLayer *layer = QgsAppLayerHandling::addLayer<QgsTiledSceneLayer>( uri, u.name, u.providerKey, addToLegend ) )
         addedLayers << layer;
     }
+    /*
     else if ( u.layerType == QLatin1String( "vector-tile" ) )
     {
       QgsTemporaryCursorOverride busyCursor( Qt::WaitCursor );
@@ -2585,6 +2595,7 @@
         addedLayers << layer;
       }
     }
+    */
     else if ( u.layerType == QLatin1String( "plugin" ) )
     {
       QgsMapLayer *layer = QgsAppLayerHandling::addLayer<QgsPluginLayer>( uri, u.name, u.providerKey, addToLegend, false );
@@ -2899,8 +2910,10 @@
   connect( mActionNewReport, &QAction::triggered, this, &QgisApp::newReport );
   connect( mActionShowLayoutManager, &QAction::triggered, this, &QgisApp::showLayoutManager );
   connect( mActionExit, &QAction::triggered, this, &QgisApp::fileExit );
-  connect( mActionDxfExport, &QAction::triggered, this, &QgisApp::dxfExport );
-  connect( mActionDwgImport, &QAction::triggered, this, &QgisApp::dwgImport );
+  // connect( mActionDxfExport, &QAction::triggered, this, &QgisApp::dxfExport );
+  // connect( mActionDwgImport, &QAction::triggered, this, [ = ] {
+  //   openFile(QFileDialog::getOpenFileName(this, tr("Open"), QString(), tr("AutoCAD DWG/DXF(*.dxf *.dwg)")));
+  // });
 
   // Edit Menu Items
 
@@ -3023,9 +3036,9 @@
   connect( mActionAddHanaLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "hana" ) ); } );
   connect( mActionAddWmsLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "wms" ) ); } );
   connect( mActionAddXyzLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "xyz" ) ); } );
-  connect( mActionAddVectorTileLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "vectortile" ) ); } );
-  connect( mActionAddPointCloudLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "pointcloud" ) ); } );
-  connect( mActionAddGpsLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "gpx" ) ); } );
+  // connect( mActionAddVectorTileLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "vectortile" ) ); } );
+  // connect( mActionAddPointCloudLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "pointcloud" ) ); } );
+  // connect( mActionAddGpsLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "gpx" ) ); } );
   connect( mActionAddWcsLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "wcs" ) ); } );
 #ifdef HAVE_SPATIALITE
   connect( mActionAddWfsLayer, &QAction::triggered, this, [=] { dataSourceManager( QStringLiteral( "WFS" ) ); } );
@@ -3529,6 +3542,7 @@
                       << mSelectionToolBar
                       << mPluginToolBar
                       << mHelpToolBar
+                      << mNGAccountToolBar
                       << mRasterToolBar
                       << mVectorToolBar
                       << mDatabaseToolBar
@@ -4115,7 +4129,9 @@
   for ( QToolBar *toolbar : constToolbars )
   {
     QString className = toolbar->parent()->metaObject()->className();
-    if ( className == QLatin1String( "QgisApp" ) )
+    QString objectName = toolbar->parent()->objectName();
+
+    if ( className == QLatin1String( "QgisApp" ) || objectName == QLatin1String( "QgisApp" ))
     {
       toolbar->setIconSize( iconSize );
     }
@@ -5028,6 +5044,7 @@
   return QgsLayerTreeRegistryBridge::InsertionPoint( insertGroup, index );
 }
 
+/*
 void QgisApp::setGpsPanelConnection( QgsGpsConnection *connection )
 {
   mGpsConnection->setConnection( connection );
@@ -5037,6 +5054,7 @@
 {
   return mGpsSettingsMenu;
 }
+*/
 
 void QgisApp::autoSelectAddedLayer( QList<QgsMapLayer *> layers )
 {
@@ -5363,7 +5381,7 @@
   QgsPluginRegistry::instance()->unloadAll();
 }
 
-#include "ui_defaults.h"
+#include "ngui_defaults.h"
 
 void QgisApp::restoreWindowState()
 {
@@ -5640,7 +5658,7 @@
   return QgsAppLayerHandling::addLayer<L>( uri, baseName, provider );
 }
 template QgsPointCloudLayer *QgisApp::addLayer<QgsPointCloudLayer>( const QString &uri, const QString &baseName, const QString &provider );
-template QgsVectorTileLayer *QgisApp::addLayer<QgsVectorTileLayer>( const QString &uri, const QString &baseName, const QString &provider );
+// template QgsVectorTileLayer *QgisApp::addLayer<QgsVectorTileLayer>( const QString &uri, const QString &baseName, const QString &provider );
 template QgsTiledSceneLayer *QgisApp::addLayer<QgsTiledSceneLayer>( const QString &uri, const QString &baseName, const QString &provider );
 template QgsPluginLayer *QgisApp::addLayer<QgsPluginLayer>( const QString &uri, const QString &baseName, const QString &provider );
 
@@ -6668,6 +6686,7 @@
   mProjectLastModified = fullPath.lastModified();
 } // QgisApp::fileSaveAs
 
+/*
 void QgisApp::dxfExport()
 {
   QgsDxfExportDialog d;
@@ -6754,6 +6773,7 @@
   QgsDwgImportDialog d;
   d.exec();
 }
+*/
 
 void QgisApp::openTemplate( const QString &fileName )
 {
@@ -8193,7 +8213,7 @@
       break;
 
     case Qgis::LayerType::VectorTile:
-      QgsVectorTileLayerProperties( qobject_cast<QgsVectorTileLayer *>( layer ), mMapCanvas, visibleMessageBar() ).saveStyleToFile();
+      // QgsVectorTileLayerProperties( qobject_cast<QgsVectorTileLayer *>( layer ), mMapCanvas, visibleMessageBar() ).saveStyleToFile();
       break;
 
     case Qgis::LayerType::PointCloud:
@@ -9773,8 +9793,8 @@
   {
     if ( QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( it.value() ) )
       vl->removeSelection();
-    else if ( QgsVectorTileLayer *vtl = qobject_cast<QgsVectorTileLayer *>( it.value() ) )
-      vtl->removeSelection();
+    // else if ( QgsVectorTileLayer *vtl = qobject_cast<QgsVectorTileLayer *>( it.value() ) )
+    //   vtl->removeSelection();
   }
 }
 
@@ -9800,12 +9820,12 @@
       break;
     }
 
-    case Qgis::LayerType::VectorTile:
-    {
-      QgsVectorTileLayer *vtlayer = qobject_cast<QgsVectorTileLayer *>( layer );
-      vtlayer->removeSelection();
-      break;
-    }
+   case Qgis::LayerType::VectorTile:
+   {
+    //  QgsVectorTileLayer *vtlayer = qobject_cast<QgsVectorTileLayer *>( layer );
+    //  vtlayer->removeSelection();
+     break;
+   }
 
     case Qgis::LayerType::Raster:
     case Qgis::LayerType::Plugin:
@@ -9969,8 +9989,8 @@
       break;
 
     case Qgis::LayerType::VectorTile:
-      clipboard()->replaceWithCopyOf( qobject_cast<QgsVectorTileLayer *>( layerContainingSelection ) );
-      break;
+      // clipboard()->replaceWithCopyOf( qobject_cast<QgsVectorTileLayer *>( layerContainingSelection ) );
+      // break;
 
     case Qgis::LayerType::Raster:
     case Qgis::LayerType::Plugin:
@@ -12126,7 +12146,7 @@
 {
   QgsScopedRuntimeProfile profile( tr( "Loading Python support" ) );
 
-  QString pythonlibName( QStringLiteral( "qgispython" ) );
+  QString pythonlibName( QStringLiteral( "ngqgispython" ) );
 #if defined( Q_OS_UNIX )
   pythonlibName.prepend( QgsApplication::libraryPath() );
 #endif
@@ -12338,6 +12358,7 @@
     sOptionsPagesMap.insert( QCoreApplication::translate( "QgsOptionsBase", "Locator" ), QStringLiteral( "mOptionsLocatorSettings" ) );
     sOptionsPagesMap.insert( QCoreApplication::translate( "QgsOptionsBase", "Acceleration" ), QStringLiteral( "mOptionsPageAcceleration" ) );
     sOptionsPagesMap.insert( QCoreApplication::translate( "QgsOptionsBase", "Advanced" ), QCoreApplication::translate( "QgsOptionsBase", "Advanced" ) );
+    sOptionsPagesMap.insert( QCoreApplication::translate( "QgsOptionsBase", "NextGIS" ), QCoreApplication::translate( "QgsOptionsBase", "NextGIS" ) );
   } );
 
   QMap<QString, QString> pages = sOptionsPagesMap;
@@ -12434,7 +12455,7 @@
     mMagnifierWidget->setDefaultFactor( factor );
     mMagnifierWidget->updateMagnification( factor );
 
-    mWelcomePage->updateNewsFeedVisibility();
+    // mWelcomePage->updateNewsFeedVisibility();
   }
 }
 
@@ -12631,7 +12652,7 @@
 void QgisApp::reportaBug()
 {
   QgsSettings settings;
-  QString reportaBugUrl = settings.value( QStringLiteral( "qgis/reportaBugUrl" ), tr( "https://qgis.org/en/site/getinvolved/development/bugreporting.html" ) ).toString();
+  QString reportaBugUrl = settings.value( QStringLiteral( "qgis/reportaBugUrl" ), tr( "https://nextgis.com/support" ) ).toString();
   openURL( reportaBugUrl, false );
 }
 
@@ -14520,12 +14541,14 @@
       }
 
       case Qgis::LayerType::VectorTile:
+      /*
       {
         QgsVectorTileLayer *vtLayer = qobject_cast<QgsVectorTileLayer *>( layer );
         const int selectedCount = vtLayer->selectedFeatureCount();
         showStatusMessage( tr( "%n feature(s) selected on layer %1.", "number of selected features", selectedCount ).arg( layer->name() ) );
         break;
       }
+      */
 
       case Qgis::LayerType::Raster:
       case Qgis::LayerType::Plugin:
@@ -14656,10 +14679,12 @@
       }
 
       case Qgis::LayerType::VectorTile:
+      /*
       {
         QgsVectorTileLayer *layer = qobject_cast<QgsVectorTileLayer *>( activeLayer() );
         return layer->selectedFeatureCount() > 0;
       }
+      */
 
       case Qgis::LayerType::Raster:
       case Qgis::LayerType::Plugin:
@@ -14687,12 +14712,14 @@
         break;
       }
       case Qgis::LayerType::VectorTile:
+      /*
       {
         QgsVectorTileLayer *layer = qobject_cast<QgsVectorTileLayer *>( mapLayer );
         if ( layer->selectedFeatureCount() > 0 )
           return true;
         break;
       }
+      */
 
       case Qgis::LayerType::Raster:
       case Qgis::LayerType::Plugin:
@@ -15307,6 +15334,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QgsVectorTileLayer *vtLayer = qobject_cast<QgsVectorTileLayer *>( layer );
       const bool layerHasSelection = vtLayer->selectedFeatureCount() > 0;
       mActionLocalHistogramStretch->setEnabled( false );
@@ -15374,6 +15402,7 @@
       mActionIdentify->setEnabled( true );
       mDigitizingTechniqueManager->enableDigitizingTechniqueActions( false );
       enableMeshEditingTools( false );
+      */
       break;
     }
 
@@ -16341,6 +16370,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QgsVectorTileLayerProperties vectorTileLayerPropertiesDialog( qobject_cast<QgsVectorTileLayer *>( mapLayer ), mMapCanvas, visibleMessageBar(), this );
       if ( !page.isEmpty() )
         vectorTileLayerPropertiesDialog.setCurrentPage( page );
@@ -16354,9 +16384,9 @@
         mMapStyleWidget->updateCurrentWidgetLayer();
       }
       mMapStyleWidget->blockUpdates( false ); // delete since dialog cannot be reused without updating code
+      */
       break;
     }
-
     case Qgis::LayerType::PointCloud:
     {
       QgsPointCloudLayerProperties pointCloudLayerPropertiesDialog( qobject_cast<QgsPointCloudLayer *>( mapLayer ), mMapCanvas, visibleMessageBar(), this );
