--- a/src/app/qgisappinterface.h	2025-05-15 15:26:35.382574966 +0300
+++ b/src/app/qgisappinterface.h	2025-05-15 14:47:00.607288826 +0300
@@ -68,7 +68,7 @@
     QgsRasterLayer *addRasterLayer( const QString &rasterLayerPath, const QString &baseName ) override;
     QgsRasterLayer *addRasterLayer( const QString &url, const QString &baseName, const QString &providerKey ) override;
     QgsMeshLayer *addMeshLayer( const QString &url, const QString &baseName, const QString &providerKey ) override;
-    QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName ) override;
+    // QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName ) override;
     QgsPointCloudLayer *addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey ) override;
     QgsTiledSceneLayer *addTiledSceneLayer( const QString &url, const QString &baseName, const QString &providerKey ) override;
     bool addProject( const QString &projectName ) override;
@@ -258,7 +258,7 @@
     QAction *actionAddPgLayer() override;
     QAction *actionAddWmsLayer() override;
     QAction *actionAddXyzLayer() override;
-    QAction *actionAddVectorTileLayer() override;
+    // QAction *actionAddVectorTileLayer() override;
     QAction *actionAddPointCloudLayer() override;
     QAction *actionAddAfsLayer() override;
     QAction *actionAddAmsLayer() override;
@@ -315,7 +315,7 @@
     void takeAppScreenShots( const QString &saveDirectory, const int categories = 0 ) override;
     QgsBrowserGuiModel *browserModel() override;
     QgsLayerTreeRegistryBridge::InsertionPoint layerTreeInsertionPoint() override;
-    void setGpsPanelConnection( QgsGpsConnection *connection ) override;
+    // void setGpsPanelConnection( QgsGpsConnection *connection ) override;
     QList<QgsMapDecoration *> activeDecorations() override;
     QgsUserProfileManager *userProfileManager() override;
     void blockActiveLayerChanges( bool blocked ) override;
