--- a/src/app/qgsidentifyresultsdialog.h	2025-07-15 20:09:37.514540219 +0300
+++ b/src/app/qgsidentifyresultsdialog.h	2025-07-30 19:58:18.114015920 +0300
@@ -164,11 +164,13 @@
      * Adds results from vector tile layer
      * \since QGIS 3.14
      */
+    /*
     void addFeature( QgsVectorTileLayer *layer,
                      const QString &label,
                      const QgsFields &fields,
                      const QgsFeature &feature,
                      const QMap< QString, QString > &derivedAttributes );
+    */
 
     /**
      * Adds results from point cloud layer
@@ -307,7 +309,7 @@
     QgsVectorLayer *vectorLayer( QTreeWidgetItem *item );
     QgsRasterLayer *rasterLayer( QTreeWidgetItem *item );
     QgsMeshLayer *meshLayer( QTreeWidgetItem *item );
-    QgsVectorTileLayer *vectorTileLayer( QTreeWidgetItem *item );
+    // QgsVectorTileLayer *vectorTileLayer( QTreeWidgetItem *item );
     QgsPointCloudLayer *pointCloudLayer( QTreeWidgetItem *item );
     QTreeWidgetItem *featureItem( QTreeWidgetItem *item );
     QTreeWidgetItem *layerItem( QTreeWidgetItem *item );
