--- a/src/app/qgslayerstylingwidget.cpp	2025-05-15 15:26:35.390697493 +0300
+++ b/src/app/qgslayerstylingwidget.cpp	2025-05-15 14:47:00.616289140 +0300
@@ -39,9 +39,9 @@
 #include "qgsmaplayerstylemanager.h"
 #include "qgsstyle.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
-#include "qgsvectortilebasiclabelingwidget.h"
-#include "qgsvectortilebasicrendererwidget.h"
+// #include "qgsvectortilelayer.h"
+// #include "qgsvectortilebasiclabelingwidget.h"
+// #include "qgsvectortilebasicrendererwidget.h"
 #include "qgsmeshlayer.h"
 #include "qgsmeshlabelingwidget.h"
 #include "qgsproject.h"
@@ -266,6 +266,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QListWidgetItem *symbolItem = new QListWidgetItem( QgsApplication::getThemeIcon( QStringLiteral( "propertyicons/symbology.svg" ) ), QString() );
       symbolItem->setData( Qt::UserRole, Symbology );
       symbolItem->setToolTip( tr( "Symbology" ) );
@@ -274,6 +275,7 @@
       labelItem->setData( Qt::UserRole, VectorLabeling );
       labelItem->setToolTip( tr( "Labels" ) );
       mOptionsListWidget->addItem( labelItem );
+      */
       break;
     }
 
@@ -753,6 +755,7 @@
 
       case Qgis::LayerType::VectorTile:
       {
+        /*
         QgsVectorTileLayer *vtLayer = qobject_cast<QgsVectorTileLayer *>( mCurrentLayer );
         switch ( row )
         {
@@ -775,6 +778,7 @@
           default:
             break;
         }
+        */
         break;
       }
 
