--- a/src/app/qgisappinterface.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/app/qgisappinterface.cpp	2022-04-06 18:40:09.000000000 +0300
@@ -163,12 +163,12 @@
 {
   return qgis->addMeshLayer( url, baseName, providerKey );
 }
-
+/*
 QgsVectorTileLayer *QgisAppInterface::addVectorTileLayer( const QString &url, const QString &baseName )
 {
   return qgis->addVectorTileLayer( url, baseName );
 }
-
+*/
 QgsPointCloudLayer *QgisAppInterface::addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey )
 {
   return qgis->addPointCloudLayer( url, baseName, providerKey );
@@ -744,7 +744,7 @@
 QAction *QgisAppInterface::actionAddPgLayer() { return qgis->actionAddPgLayer(); }
 QAction *QgisAppInterface::actionAddWmsLayer() { return qgis->actionAddWmsLayer(); }
 QAction *QgisAppInterface::actionAddXyzLayer() { return qgis->actionAddXyzLayer(); }
-QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
+// QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
 QAction *QgisAppInterface::actionAddPointCloudLayer() { return qgis->actionAddPointCloudLayer(); }
 QAction *QgisAppInterface::actionAddAfsLayer() { return qgis->actionAddAfsLayer(); }
 QAction *QgisAppInterface::actionAddAmsLayer() { return qgis->actionAddAfsLayer(); }
@@ -908,12 +908,12 @@
 {
   return qgis->layerTreeInsertionPoint();
 }
-
+/*
 void QgisAppInterface::setGpsPanelConnection( QgsGpsConnection *connection )
 {
   qgis->setGpsPanelConnection( connection );
 }
-
+*/
 QList<QgsMapDecoration *> QgisAppInterface::activeDecorations()
 {
   return qgis->activeDecorations();
