--- a/src/app/qgisapp.cpp	2023-08-22 11:32:52.639350417 +0300
+++ b/src/app/qgisapp.cpp	2023-09-25 09:51:22.240160215 +0300
@@ -114,7 +114,7 @@
 #include "qgsgeometrycheckregistry.h"
 
 #include "options/qgscodeeditoroptions.h"
-#include "options/qgsgpsdeviceoptions.h"
+// #include "options/qgsgpsdeviceoptions.h"
 #include "options/qgscustomprojectionoptions.h"
 
 #ifdef HAVE_3D
@@ -224,9 +224,9 @@
 #include "qgsdatumtransformdialog.h"
 #include "qgsdoublespinbox.h"
 #include "qgsdockwidget.h"
-#include "qgsdxfexport.h"
-#include "qgsdxfexportdialog.h"
-#include "qgsdwgimportdialog.h"
+// #include "qgsdxfexport.h"
+// #include "qgsdxfexportdialog.h"
+// #include "qgsdwgimportdialog.h"
 #include "qgsdecorationtitle.h"
 #include "qgsdecorationcopyright.h"
 #include "qgsdecorationimage.h"
@@ -250,7 +250,7 @@
 #include "qgsguiutils.h"
 #include "qgshtmlannotation.h"
 #include "qgsprojectionselectiondialog.h"
-#include "qgsgpsinformationwidget.h"
+// #include "qgsgpsinformationwidget.h"
 #include "qgsguivectorlayertools.h"
 #include "qgslabelingwidget.h"
 #include "qgsdiagramproperties.h"
@@ -308,7 +308,7 @@
 #include "qgsmapoverviewcanvas.h"
 #include "qgsmapsettings.h"
 #include "qgsmaptip.h"
-#include "qgsmbtiles.h"
+// #include "qgsmbtiles.h"
 #include "qgsmenuheader.h"
 #include "qgsmergeattributesdialog.h"
 #include "qgsmessageviewer.h"
@@ -392,8 +392,8 @@
 #include "qgsvectorlayer.h"
 #include "qgsvectorlayerproperties.h"
 #include "qgsvectorlayerdigitizingproperties.h"
-#include "qgsvectortilelayer.h"
-#include "qgsvectortilelayerproperties.h"
+// #include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayerproperties.h"
 #include "qgspointcloudlayerproperties.h"
 #include "qgsmapthemes.h"
 #include "qgsmessagelogviewer.h"
@@ -1370,7 +1370,7 @@
 
   addDockWidget( Qt::LeftDockWidgetArea, mBookMarksDockWidget );
   mBookMarksDockWidget->hide();
-
+/*
   // create the GPS tool on starting QGIS - this is like the browser
   mpGpsWidget = new QgsGpsInformationWidget( mMapCanvas );
   QgsPanelWidgetStack *gpsStack = new QgsPanelWidgetStack();
@@ -1391,7 +1391,7 @@
   // now add our widget to the dock - ownership of the widget is passed to the dock
   mpGpsDock->setWidget( gpsStack );
   mpGpsDock->hide();
-
+*/
   mLastMapToolMessage = nullptr;
 
   mLogViewer = new QgsMessageLogViewer( this );
@@ -1627,6 +1627,8 @@
   // do main window customization - after window state has been restored, before the window is shown
   startProfile( tr( "Update customization on main window" ) );
   QgsCustomization::instance()->updateMainWindow( mToolbarMenu, mPanelMenu );
+  mActionAddAfsLayer->setVisible(false);
+  mActionAddVectorTileLayer->setVisible(false);
   endProfile();
 
   mSplash->showMessage( tr( "Populate saved styles" ), Qt::AlignHCenter | Qt::AlignBottom, splashTextColor );
@@ -1819,7 +1821,7 @@
   } );
 
   mCodeEditorWidgetFactory.reset( std::make_unique< QgsCodeEditorOptionsFactory >() );
-  mBabelGpsDevicesWidgetFactory.reset( std::make_unique< QgsGpsDeviceOptionsFactory >() );
+//   mBabelGpsDevicesWidgetFactory.reset( std::make_unique< QgsGpsDeviceOptionsFactory >() );
   mCustomProjectionsWidgetFactory.reset( std::make_unique< QgsCustomProjectionOptionsFactory >() );
 
 #ifdef HAVE_3D
@@ -1884,7 +1886,7 @@
 
   delete mpMaptip;
 
-  delete mpGpsWidget;
+//   delete mpGpsWidget;
 
   delete mOverviewMapCursor;
 
@@ -2171,7 +2173,7 @@
     {
       addPointCloudLayer( uri, u.name, u.providerKey );
     }
-    else if ( u.layerType == QLatin1String( "vector-tile" ) )
+    /*else if ( u.layerType == QLatin1String( "vector-tile" ) )
     {
       QgsTemporaryCursorOverride busyCursor( Qt::WaitCursor );
 
@@ -2200,7 +2202,7 @@
                                message, Qgis::MessageLevel::Warning );
       }
       addMapLayer( layer );
-    }
+    }*/
     else if ( u.layerType == QLatin1String( "plugin" ) )
     {
       addPluginLayer( uri, u.name, u.providerKey );
@@ -2495,7 +2497,7 @@
     connect( mDataSourceManagerDialog, SIGNAL( addVectorLayers( QStringList const &, QString const &, QString const & ) ),
              this, SLOT( addVectorLayers( QStringList const &, QString const &, QString const & ) ) );
     connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addMeshLayer, this, &QgisApp::addMeshLayer );
-    connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addVectorTileLayer, this, &QgisApp::addVectorTileLayer );
+    // connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addVectorTileLayer, this, &QgisApp::addVectorTileLayer );
     connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addPointCloudLayer, this, &QgisApp::addPointCloudLayer );
     connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::showStatusMessage, this, &QgisApp::showStatusMessage );
     connect( mDataSourceManagerDialog, &QgsDataSourceManagerDialog::addDatabaseLayers, this, &QgisApp::addDatabaseLayers );
@@ -2695,8 +2697,8 @@
   connect( mActionNewReport, &QAction::triggered, this, &QgisApp::newReport );
   connect( mActionShowLayoutManager, &QAction::triggered, this, &QgisApp::showLayoutManager );
   connect( mActionExit, &QAction::triggered, this, &QgisApp::fileExit );
-  connect( mActionDxfExport, &QAction::triggered, this, &QgisApp::dxfExport );
-  connect( mActionDwgImport, &QAction::triggered, this, &QgisApp::dwgImport );
+//   connect( mActionDxfExport, &QAction::triggered, this, &QgisApp::dxfExport );
+//   connect( mActionDwgImport, &QAction::triggered, this, &QgisApp::dwgImport );
 
   // Edit Menu Items
 
@@ -2843,7 +2845,7 @@
   connect( mActionAddHanaLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "hana" ) ); } );
   connect( mActionAddWmsLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "wms" ) ); } );
   connect( mActionAddXyzLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "xyz" ) ); } );
-  connect( mActionAddVectorTileLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "vectortile" ) ); } );
+//   connect( mActionAddVectorTileLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "vectortile" ) ); } );
   connect( mActionAddPointCloudLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "pointcloud" ) ); } );
   connect( mActionAddWcsLayer, &QAction::triggered, this, [ = ] { dataSourceManager( QStringLiteral( "wcs" ) ); } );
 #ifdef HAVE_SPATIALITE
@@ -4302,7 +4304,7 @@
   mActionCustomProjection->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionCustomProjection.svg" ) ) );
   mActionAddWmsLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddWmsLayer.svg" ) ) );
   mActionAddXyzLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddXyzLayer.svg" ) ) );
-  mActionAddVectorTileLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddVectorTileLayer.svg" ) ) );
+//   mActionAddVectorTileLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddVectorTileLayer.svg" ) ) );
   mActionAddWcsLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddWcsLayer.svg" ) ) );
 #ifdef HAVE_SPATIALITE
   mActionAddWfsLayer->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "/mActionAddWfsLayer.svg" ) ) );
@@ -5047,12 +5049,12 @@
   }
   return QgsLayerTreeRegistryBridge::InsertionPoint( insertGroup, index );
 }
-
+/*
 void QgisApp::setGpsPanelConnection( QgsGpsConnection *connection )
 {
   mpGpsWidget->setConnection( connection );
 }
-
+*/
 void QgisApp::autoSelectAddedLayer( QList<QgsMapLayer *> layers )
 {
   if ( !layers.isEmpty() )
@@ -5360,7 +5362,7 @@
   QgsPluginRegistry::instance()->unloadAll();
 }
 
-#include "ui_defaults.h"
+#include "ngui_defaults.h"
 
 void QgisApp::restoreWindowState()
 {
@@ -5952,7 +5954,7 @@
       meshLayer->loadDefaultMetadata( ok );
       break;
     }
-
+/*
     case QgsMapLayerType::VectorTileLayer:
     {
       bool ok = false;
@@ -5965,7 +5967,7 @@
 
       break;
     }
-
+*/
     case QgsMapLayerType::AnnotationLayer:
     case QgsMapLayerType::GroupLayer:
       break;
@@ -6007,17 +6009,17 @@
     }
   }
 }
-
+/*
 QgsVectorTileLayer *QgisApp::addVectorTileLayer( const QString &url, const QString &baseName )
 {
   return addVectorTileLayerPrivate( url, baseName );
 }
-
+*/
 QgsPointCloudLayer *QgisApp::addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey )
 {
   return addPointCloudLayerPrivate( url, baseName, providerKey );
 }
-
+/*
 QgsVectorTileLayer *QgisApp::addVectorTileLayerPrivate( const QString &url, const QString &baseName, const bool guiWarning )
 {
   QgsCanvasRefreshBlocker refreshBlocker;
@@ -6055,7 +6057,7 @@
 
   return layer.release();
 }
-
+*/
 QgsPointCloudLayer *QgisApp::addPointCloudLayerPrivate( const QString &uri, const QString &baseName, const QString &providerKey, bool guiWarning )
 {
   QgsCanvasRefreshBlocker refreshBlocker;
@@ -7381,7 +7383,7 @@
   }
   mProjectLastModified = fullPath.lastModified();
 } // QgisApp::fileSaveAs
-
+/*
 void QgisApp::dxfExport()
 {
   QgsDxfExportDialog d;
@@ -7447,7 +7449,7 @@
   QgsDwgImportDialog d;
   d.exec();
 }
-
+*/
 void QgisApp::openLayerDefinition( const QString &path )
 {
   QString errorMessage;
@@ -7651,36 +7653,36 @@
     }
   }
 
-  if ( fileName.endsWith( QStringLiteral( ".mbtiles" ), Qt::CaseInsensitive ) )
-  {
-    QgsMbTiles reader( fileName );
-    if ( reader.open() )
-    {
-      if ( reader.metadataValue( "format" ) == QLatin1String( "pbf" ) )
-      {
-        // these are vector tiles
-        QUrlQuery uq;
-        uq.addQueryItem( QStringLiteral( "type" ), QStringLiteral( "mbtiles" ) );
-        uq.addQueryItem( QStringLiteral( "url" ), fileName );
-        const QgsVectorTileLayer::LayerOptions options( QgsProject::instance()->transformContext() );
-        std::unique_ptr<QgsVectorTileLayer> vtLayer( new QgsVectorTileLayer( uq.toString(), fileInfo.completeBaseName(), options ) );
-        if ( vtLayer->isValid() )
-        {
-          QgsProject::instance()->addMapLayer( vtLayer.release() );
-          return true;
-        }
-      }
-      else // raster tiles
-      {
-        // prefer to use WMS provider's implementation to open MBTiles rasters
-        QUrlQuery uq;
-        uq.addQueryItem( QStringLiteral( "type" ), QStringLiteral( "mbtiles" ) );
-        uq.addQueryItem( QStringLiteral( "url" ), QUrl::fromLocalFile( fileName ).toString() );
-        if ( addRasterLayer( uq.toString(), fileInfo.completeBaseName(), QStringLiteral( "wms" ) ) )
-          return true;
-      }
-    }
-  }
+//   if ( fileName.endsWith( QStringLiteral( ".mbtiles" ), Qt::CaseInsensitive ) )
+//   {
+//     QgsMbTiles reader( fileName );
+//     if ( reader.open() )
+//     {
+//       if ( reader.metadataValue( "format" ) == QLatin1String( "pbf" ) )
+//       {
+//         // these are vector tiles
+//         QUrlQuery uq;
+//         uq.addQueryItem( QStringLiteral( "type" ), QStringLiteral( "mbtiles" ) );
+//         uq.addQueryItem( QStringLiteral( "url" ), fileName );
+//         const QgsVectorTileLayer::LayerOptions options( QgsProject::instance()->transformContext() );
+//         std::unique_ptr<QgsVectorTileLayer> vtLayer( new QgsVectorTileLayer( uq.toString(), fileInfo.completeBaseName(), options ) );
+//         if ( vtLayer->isValid() )
+//         {
+//           QgsProject::instance()->addMapLayer( vtLayer.release() );
+//           return true;
+//         }
+//       }
+//       else // raster tiles
+//       {
+//         // prefer to use WMS provider's implementation to open MBTiles rasters
+//         QUrlQuery uq;
+//         uq.addQueryItem( QStringLiteral( "type" ), QStringLiteral( "mbtiles" ) );
+//         uq.addQueryItem( QStringLiteral( "url" ), QUrl::fromLocalFile( fileName ).toString() );
+//         if ( addRasterLayer( uq.toString(), fileInfo.completeBaseName(), QStringLiteral( "wms" ) ) )
+//           return true;
+//       }
+//     }
+//   }
 
   QList< QgsProviderSublayerModel::NonLayerItem > nonLayerItems;
   if ( QgsProjectStorage *ps = QgsApplication::projectStorageRegistry()->projectStorageFromUri( fileName ) )
@@ -13012,7 +13014,7 @@
 {
   QgsScopedRuntimeProfile profile( tr( "Loading Python support" ) );
 
-  QString pythonlibName( QStringLiteral( "qgispython" ) );
+  QString pythonlibName( QStringLiteral( "ngqgispython" ) );
 #if defined(Q_OS_UNIX)
   pythonlibName.prepend( QgsApplication::libraryPath() );
 #endif
@@ -13321,7 +13323,7 @@
     mMagnifierWidget->setDefaultFactor( factor );
     mMagnifierWidget->updateMagnification( factor );
 
-    mWelcomePage->updateNewsFeedVisibility();
+    // mWelcomePage->updateNewsFeedVisibility();
   }
 }
 
@@ -16163,7 +16165,7 @@
     }
 
     break;
-
+/*
     case QgsMapLayerType::VectorTileLayer:
       mActionLocalHistogramStretch->setEnabled( false );
       mActionFullHistogramStretch->setEnabled( false );
@@ -16233,7 +16235,7 @@
       enableDigitizeTechniqueActions( false );
       enableMeshEditingTools( false );
       break;
-
+*/
     case QgsMapLayerType::PointCloudLayer:
       mActionLocalHistogramStretch->setEnabled( false );
       mActionFullHistogramStretch->setEnabled( false );
@@ -17169,7 +17171,7 @@
       delete vectorLayerPropertiesDialog; // delete since dialog cannot be reused without updating code
       break;
     }
-
+/*
     case QgsMapLayerType::VectorTileLayer:
     {
       QgsVectorTileLayerProperties vectorTileLayerPropertiesDialog( qobject_cast<QgsVectorTileLayer *>( mapLayer ), mMapCanvas, visibleMessageBar(), this );
@@ -17187,7 +17189,7 @@
       mMapStyleWidget->blockUpdates( false ); // delete since dialog cannot be reused without updating code
       break;
     }
-
+*/
     case QgsMapLayerType::PointCloudLayer:
     {
       QgsPointCloudLayerProperties pointCloudLayerPropertiesDialog( qobject_cast<QgsPointCloudLayer *>( mapLayer ), mMapCanvas, visibleMessageBar(), this );
