--- a/src/app/browser/qgsinbuiltdataitemproviders.cpp	2025-05-15 15:26:35.349144193 +0300
+++ b/src/app/browser/qgsinbuiltdataitemproviders.cpp	2025-05-15 14:47:00.569287500 +0300
@@ -64,8 +64,8 @@
 #include "qgsrelationshipsitem.h"
 #include "qgsprovidersqlquerybuilder.h"
 #include "qgsdbrelationshipwidget.h"
-#include "qgsdbqueryhistoryprovider.h"
-#include "qgshistoryproviderregistry.h"
+// #include "qgsdbqueryhistoryprovider.h"
+// #include "qgshistoryproviderregistry.h"
 
 #include <QFileInfo>
 #include <QMenu>
@@ -1648,10 +1648,10 @@
 
 QgsDatabaseItemGuiProvider::QgsDatabaseItemGuiProvider()
 {
-  if ( QgsDatabaseQueryHistoryProvider *historyProvider = qobject_cast<QgsDatabaseQueryHistoryProvider *>( QgsGui::historyProviderRegistry()->providerById( QStringLiteral( "dbquery" ) ) ) )
-  {
-    connect( historyProvider, &QgsDatabaseQueryHistoryProvider::openSqlDialog, this, &QgsDatabaseItemGuiProvider::openSqlDialogGeneric );
-  }
+  // if ( QgsDatabaseQueryHistoryProvider *historyProvider = qobject_cast<QgsDatabaseQueryHistoryProvider *>( QgsGui::historyProviderRegistry()->providerById( QStringLiteral( "dbquery" ) ) ) )
+  // {
+  //   connect( historyProvider, &QgsDatabaseQueryHistoryProvider::openSqlDialog, this, &QgsDatabaseItemGuiProvider::openSqlDialogGeneric );
+  // }
 }
 
 QString QgsDatabaseItemGuiProvider::name()
