--- a/src/app/qgsmaptoolselectutils.cpp	2025-07-15 20:09:37.521540369 +0300
+++ b/src/app/qgsmaptoolselectutils.cpp	2025-07-30 19:58:18.114961156 +0300
@@ -21,7 +21,7 @@
 #include "qgsmessagebar.h"
 #include "qgsmapcanvas.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsvectorlayerfeatureiterator.h"
 #include "qgsfeature.h"
 #include "qgsgeometry.h"
@@ -50,9 +50,9 @@
     switch ( layer->type() )
     {
       case Qgis::LayerType::Vector:
-      case Qgis::LayerType::VectorTile:
         // supported
         break;
+      case Qgis::LayerType::VectorTile:
       case Qgis::LayerType::Raster:
       case Qgis::LayerType::Plugin:
       case Qgis::LayerType::Mesh:
@@ -121,7 +121,7 @@
       }
       case Qgis::LayerType::VectorTile:
         // mixed layer type, so aim for somewhere between the vector layer polygon/point sizes
-        boxSize = 2;
+        // boxSize = 2;
         break;
 
       case Qgis::LayerType::Raster:
@@ -254,6 +254,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QgsVectorTileLayer *vtLayer = qobject_cast< QgsVectorTileLayer *>( layer );
 
       QgsCoordinateTransform ct( canvas->mapSettings().destinationCrs(), layer->crs(), QgsProject::instance() );
@@ -273,6 +274,7 @@
       renderContext.setExpressionContext( expressionContext );
 
       vtLayer->selectByGeometry( selectGeomTrans, context, behavior, Qgis::SelectGeometryRelationship::Intersect, flags, &renderContext );
+      */
       break;
     }
 
@@ -312,6 +314,7 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QgsVectorTileLayer *vtLayer = qobject_cast< QgsVectorTileLayer * >( layer );
       QgsCoordinateTransform ct( canvas->mapSettings().destinationCrs(), layer->crs(), QgsProject::instance() );
       QgsGeometry selectGeomTrans;
@@ -326,6 +329,7 @@
       renderContext.setExpressionContext( expressionContext );
 
       vtLayer->selectByGeometry( selectGeomTrans, context, selectBehavior, doContains ? Qgis::SelectGeometryRelationship::Within : Qgis::SelectGeometryRelationship::Intersect, Qgis::SelectionFlags(), &renderContext );
+      */
       break;
     }
 
