--- a/src/app/qgswelcomepage.cpp	2025-05-15 15:26:35.402822667 +0300
+++ b/src/app/qgswelcomepage.cpp	2025-05-15 14:47:00.642290046 +0300
@@ -27,8 +27,8 @@
 #include "qgsfileutils.h"
 #include "qgstemplateprojectsmodel.h"
 #include "qgsprojectlistitemdelegate.h"
-#include "qgsnewsfeedmodel.h"
-#include "qgsnewsfeedparser.h"
+// #include "qgsnewsfeedmodel.h"
+// #include "qgsnewsfeedparser.h"
 
 #include "qgsprojectstorage.h"
 #include "qgsprojectstorageguiprovider.h"
@@ -45,7 +45,7 @@
 #include <QRegularExpression>
 #include <QUrl>
 
-#define FEED_URL "https://feed.qgis.org/"
+// #define FEED_URL "https://feed.qgis.org/"
 
 QgsWelcomePage::QgsWelcomePage( bool skipVersionCheck, QWidget *parent )
   : QWidget( parent )
@@ -86,7 +86,7 @@
   QWidget *rightContainer = new QWidget();
   QVBoxLayout *rightLayout = new QVBoxLayout;
   rightLayout->setContentsMargins( 0, 0, 0, 0 );
-
+  /*
   if ( !QgsSettings().value( QStringLiteral( "%1/disabled" ).arg( QgsNewsFeedParser::keyForFeed( QStringLiteral( FEED_URL ) ) ), false, QgsSettings::Core ).toBool() )
   {
     mSplitter2 = new QSplitter( Qt::Vertical );
@@ -117,7 +117,7 @@
     newsContainer->setLayout( newsLayout );
     mSplitter2->addWidget( newsContainer );
   }
-
+  */
   QWidget *templateContainer = new QWidget();
   QVBoxLayout *templateLayout = new QVBoxLayout();
   templateLayout->setContentsMargins( 0, 0, 0, 0 );
@@ -168,7 +168,7 @@
        && settings.value( QStringLiteral( "qgis/checkVersion" ), true ).toBool() && !skipVersionCheck )
   {
     connect( mVersionInfo, &QgsVersionInfo::versionInfoAvailable, this, &QgsWelcomePage::versionInfoReceived );
-    mVersionInfo->checkVersion();
+    // mVersionInfo->checkVersion();
   }
 
   mSplitter->restoreState( settings.value( QStringLiteral( "Windows/WelcomePage/SplitState" ), QVariant(), QgsSettings::App ).toByteArray() );
@@ -178,15 +178,15 @@
   connect( mRecentProjectsListView, &QAbstractItemView::activated, this, &QgsWelcomePage::recentProjectItemActivated );
   connect( mTemplateProjectsListView, &QAbstractItemView::activated, this, &QgsWelcomePage::templateProjectItemActivated );
 
-  updateNewsFeedVisibility();
+  // updateNewsFeedVisibility();
 }
 
 QgsWelcomePage::~QgsWelcomePage()
 {
   QgsSettings settings;
   settings.setValue( QStringLiteral( "Windows/WelcomePage/SplitState" ), mSplitter->saveState(), QgsSettings::App );
-  if ( mSplitter2 && mNewsFeedTitle->isVisible() )
-    settings.setValue( QStringLiteral( "Windows/WelcomePage/SplitState2" ), mSplitter2->saveState(), QgsSettings::App );
+  // if ( mSplitter2 && mNewsFeedTitle->isVisible() )
+  //   settings.setValue( QStringLiteral( "Windows/WelcomePage/SplitState2" ), mSplitter2->saveState(), QgsSettings::App );
 
   delete mVersionInfo;
 }
@@ -196,10 +196,12 @@
   mRecentProjectsModel->setRecentProjects( recentProjects );
 }
 
+/*
 QString QgsWelcomePage::newsFeedUrl()
 {
   return QStringLiteral( FEED_URL );
 }
+*/
 
 QgsRecentProjectItemsModel *QgsWelcomePage::recentProjectsModel()
 {
@@ -219,6 +221,7 @@
     QgisApp::instance()->fileNewFromTemplate( index.data( QgsProjectListItemDelegate::NativePathRole ).toString() );
 }
 
+/*
 void QgsWelcomePage::newsItemActivated( const QModelIndex &index )
 {
   if ( !index.isValid() )
@@ -227,6 +230,7 @@
   const QUrl link = index.data( static_cast<int>( QgsNewsFeedModel::CustomRole::Link ) ).toUrl();
   QDesktopServices::openUrl( link );
 }
+*/
 
 void QgsWelcomePage::versionInfoReceived()
 {
@@ -377,7 +381,7 @@
 
   menu->popup( mTemplateProjectsListView->mapToGlobal( point ) );
 }
-
+/*
 void QgsWelcomePage::showContextMenuForNews( QPoint point )
 {
   const QModelIndex index = mNewsFeedListView->indexAt( point );
@@ -458,6 +462,8 @@
 
   return QWidget::eventFilter( obj, event );
 }
+*/
+
 
 void QgsWelcomePage::removeProject( int row )
 {
