--- a/src/app/qgisappinterface.cpp	2025-07-15 20:09:37.506540049 +0300
+++ b/src/app/qgisappinterface.cpp	2025-07-30 19:58:18.114015920 +0300
@@ -160,10 +160,12 @@
   return qgis->addMeshLayer( url, baseName, providerKey );
 }
 
+/*
 QgsVectorTileLayer *QgisAppInterface::addVectorTileLayer( const QString &url, const QString &baseName )
 {
   return qgis->addVectorTileLayer( url, baseName );
 }
+*/
 
 QgsPointCloudLayer *QgisAppInterface::addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey )
 {
@@ -813,7 +815,7 @@
 QAction *QgisAppInterface::actionAddPgLayer() { return qgis->actionAddPgLayer(); }
 QAction *QgisAppInterface::actionAddWmsLayer() { return qgis->actionAddWmsLayer(); }
 QAction *QgisAppInterface::actionAddXyzLayer() { return qgis->actionAddXyzLayer(); }
-QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
+// QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
 QAction *QgisAppInterface::actionAddPointCloudLayer() { return qgis->actionAddPointCloudLayer(); }
 QAction *QgisAppInterface::actionAddAfsLayer() { return qgis->actionAddAfsLayer(); }
 QAction *QgisAppInterface::actionAddAmsLayer() { return qgis->actionAddAfsLayer(); }
@@ -978,10 +980,12 @@
   return qgis->layerTreeInsertionPoint();
 }
 
+/*
 void QgisAppInterface::setGpsPanelConnection( QgsGpsConnection *connection )
 {
   qgis->setGpsPanelConnection( connection );
 }
+*/
 
 QList<QgsMapDecoration *> QgisAppInterface::activeDecorations()
 {
@@ -992,4 +996,3 @@
 {
   return qgis->userProfileManager();
 }
-
