--- a/src/app/qgisappinterface.cpp	2025-04-25 19:26:53.684229269 +0300
+++ b/src/app/qgisappinterface.cpp	2025-04-02 18:34:56.423365411 +0300
@@ -159,12 +159,12 @@
 {
   return qgis->addMeshLayer( url, baseName, providerKey );
 }
-
+/*
 QgsVectorTileLayer *QgisAppInterface::addVectorTileLayer( const QString &url, const QString &baseName )
 {
   return qgis->addVectorTileLayer( url, baseName );
 }
-
+*/
 QgsPointCloudLayer *QgisAppInterface::addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey )
 {
   return qgis->addPointCloudLayer( url, baseName, providerKey );
@@ -813,7 +813,7 @@
 QAction *QgisAppInterface::actionAddPgLayer() { return qgis->actionAddPgLayer(); }
 QAction *QgisAppInterface::actionAddWmsLayer() { return qgis->actionAddWmsLayer(); }
 QAction *QgisAppInterface::actionAddXyzLayer() { return qgis->actionAddXyzLayer(); }
-QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
+// QAction *QgisAppInterface::actionAddVectorTileLayer() { return qgis->actionAddVectorTileLayer(); }
 QAction *QgisAppInterface::actionAddPointCloudLayer() { return qgis->actionAddPointCloudLayer(); }
 QAction *QgisAppInterface::actionAddAfsLayer() { return qgis->actionAddAfsLayer(); }
 QAction *QgisAppInterface::actionAddAmsLayer() { return qgis->actionAddAfsLayer(); }
@@ -977,12 +977,12 @@
 {
   return qgis->layerTreeInsertionPoint();
 }
-
+/*
 void QgisAppInterface::setGpsPanelConnection( QgsGpsConnection *connection )
 {
   qgis->setGpsPanelConnection( connection );
 }
-
+*/
 QList<QgsMapDecoration *> QgisAppInterface::activeDecorations()
 {
   return qgis->activeDecorations();
