--- a/python/core/auto_generated/processing/qgsprocessingutils.sip.in	2025-05-15 15:26:35.293443102 +0300
+++ b/python/core/auto_generated/processing/qgsprocessingutils.sip.in	2025-05-15 14:47:00.291277805 +0300
@@ -44,8 +44,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -78,8 +76,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -103,8 +99,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -130,8 +124,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -157,8 +149,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -184,8 +174,6 @@
 
 .. seealso:: :py:func:`compatiblePointCloudLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleTiledSceneLayers`
 
 .. seealso:: :py:func:`compatibleLayers`
@@ -193,33 +181,6 @@
 .. versionadded:: 3.22
 %End
 
-    static QList<QgsVectorTileLayer *> compatibleVectorTileLayers( QgsProject *project, bool sort = true );
-%Docstring
-Returns a list of vector tile layers from a ``project`` which are
-compatible with the processing framework.
-
-If the ``sort`` argument is ``True`` then the layers will be sorted by
-their :py:func:`QgsMapLayer.name()` value.
-
-.. seealso:: :py:func:`compatibleRasterLayers`
-
-.. seealso:: :py:func:`compatibleVectorLayers`
-
-.. seealso:: :py:func:`compatibleMeshLayers`
-
-.. seealso:: :py:func:`compatiblePluginLayers`
-
-.. seealso:: :py:func:`compatiblePointCloudLayers`
-
-.. seealso:: :py:func:`compatibleAnnotationLayers`
-
-.. seealso:: :py:func:`compatibleTiledSceneLayers`
-
-.. seealso:: :py:func:`compatibleLayers`
-
-.. versionadded:: 3.32
-%End
-
     static QList<QgsTiledSceneLayer *> compatibleTiledSceneLayers( QgsProject *project, bool sort = true );
 %Docstring
 Returns a list of tiled scene layers from a ``project`` which are
@@ -589,8 +550,6 @@
 
 .. seealso:: :py:func:`defaultPointCloudExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.10
 %End
 
@@ -606,8 +565,6 @@
 
 .. seealso:: :py:func:`defaultPointCloudExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.10
 %End
 
@@ -622,27 +579,9 @@
 
 .. seealso:: :py:func:`defaultRasterExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.24
 %End
 
-    static QString defaultVectorTileExtension();
-%Docstring
-Returns the default vector tile extension to use, in the absence of all
-other constraints (e.g. provider based support for extensions).
-
-This method returns a fallback value of "mbtiles".
-
-.. seealso:: :py:func:`defaultVectorExtension`
-
-.. seealso:: :py:func:`defaultRasterExtension`
-
-.. seealso:: :py:func:`defaultPointCloudExtension`
-
-.. versionadded:: 3.32
-%End
-
     static QVariantMap removePointerValuesFromMap( const QVariantMap &map );
 %Docstring
 Removes any raw pointer values from an input ``map``, replacing them
