--- a/python/core/auto_generated/qgspathresolver.sip.in	2025-07-15 20:09:37.229534133 +0300
+++ b/python/core/auto_generated/qgspathresolver.sip.in	2025-07-30 03:21:02.040397058 +0300
@@ -90,9 +90,10 @@
 %End
 %MethodCode
     PyObject *s = 0;
-    Py_BEGIN_ALLOW_THREADS
+    QString id;
     Py_XINCREF( a0 );
-    QString id = QgsPathResolver::setPathPreprocessor( [a0]( const QString &arg )->QString
+    Py_BEGIN_ALLOW_THREADS
+    id = QgsPathResolver::setPathPreprocessor( [a0]( const QString &arg )->QString
     {
       QString res;
       SIP_BLOCK_THREADS
@@ -109,8 +110,8 @@
       return res;
     } );
 
-    s = sipConvertFromNewType( new QString( id ), sipType_QString, 0 );
     Py_END_ALLOW_THREADS
+    s = sipConvertFromNewType( new QString( id ), sipType_QString, 0 );
     return s;
 %End
 
@@ -164,9 +165,10 @@
 %End
 %MethodCode
     PyObject *s = 0;
-    Py_BEGIN_ALLOW_THREADS
+    QString id;
     Py_XINCREF( a0 );
-    QString id = QgsPathResolver::setPathWriter( [a0]( const QString &arg )->QString
+    Py_BEGIN_ALLOW_THREADS
+    id = QgsPathResolver::setPathWriter( [a0]( const QString &arg )->QString
     {
       QString res;
       SIP_BLOCK_THREADS
@@ -183,8 +185,8 @@
       return res;
     } );
 
-    s = sipConvertFromNewType( new QString( id ), sipType_QString, 0 );
     Py_END_ALLOW_THREADS
+    s = sipConvertFromNewType( new QString( id ), sipType_QString, 0 );
     return s;
 %End
 
