--- a/python/core/auto_generated/processing/qgsprocessingparameters.sip.in	2022-02-18 15:06:10.000000000 +0300
+++ b/python/core/auto_generated/processing/qgsprocessingparameters.sip.in	2022-04-30 14:32:32.145601500 +0300
@@ -226,10 +226,8 @@
 %End
 %TypeHeaderCode
 #include "qgsprocessingparameteraggregate.h"
-#include "qgsprocessingparameterdxflayers.h"
 #include "qgsprocessingparameterfieldmap.h"
 #include "qgsprocessingparametertininputlayers.h"
-#include "qgsprocessingparametervectortilewriterlayers.h"
 #include "qgsprocessingparametermeshdataset.h"
 %End
 %ConvertToSubClassCode
@@ -315,10 +313,6 @@
       sipType = sipType_QgsProcessingParameterFieldMapping;
     else if ( sipCpp->type() == QgsProcessingParameterTinInputLayers::typeName() )
       sipType = sipType_QgsProcessingParameterTinInputLayers;
-    else if ( sipCpp->type() == QgsProcessingParameterVectorTileWriterLayers::typeName() )
-      sipType = sipType_QgsProcessingParameterVectorTileWriterLayers;
-    else if ( sipCpp->type() == QgsProcessingParameterDxfLayers::typeName() )
-      sipType = sipType_QgsProcessingParameterDxfLayers;
     else if ( sipCpp->type() == QgsProcessingParameterMeshDatasetGroups::typeName() )
       sipType = sipType_QgsProcessingParameterMeshDatasetGroups;
     else if ( sipCpp->type() == QgsProcessingParameterMeshDatasetTime::typeName() )
