--- a/python/core/auto_generated/raster/qgsbrightnesscontrastfilter.sip.in	2022-02-18 15:06:10.000000000 +0300
+++ b/python/core/auto_generated/raster/qgsbrightnesscontrastfilter.sip.in	2022-05-14 20:21:37.765753700 +0300
@@ -63,7 +63,7 @@
 
     void setBrightness( int brightness );
 %Docstring
-Set brightness level. Acceptable value range is -255…255
+Set brightness level. Acceptable value range is -255...255
 
 .. seealso:: :py:func:`brightness`
 %End
@@ -77,7 +77,7 @@
 
     void setContrast( int contrast );
 %Docstring
-Set contrast level. Acceptable value range is -100…100
+Set contrast level. Acceptable value range is -100...100
 
 .. seealso:: :py:func:`contrast`
 %End
@@ -91,7 +91,7 @@
 
     void setGamma( double gamma );
 %Docstring
-Set gamma value. Acceptable value range is -0.1…10
+Set gamma value. Acceptable value range is -0.1...10
 
 .. seealso:: :py:func:`gamma`
 
