--- a/python/core/auto_generated/processing/qgsprocessingparameters.sip.in	2025-04-25 19:26:53.216218996 +0300
+++ b/python/core/auto_generated/processing/qgsprocessingparameters.sip.in	2025-04-02 18:34:55.730622046 +0300
@@ -234,10 +234,8 @@
 %End
 %TypeHeaderCode
 #include "qgsprocessingparameteraggregate.h"
-#include "qgsprocessingparameterdxflayers.h"
 #include "qgsprocessingparameterfieldmap.h"
 #include "qgsprocessingparametertininputlayers.h"
-#include "qgsprocessingparametervectortilewriterlayers.h"
 #include "qgsprocessingparametermeshdataset.h"
 %End
 %ConvertToSubClassCode
@@ -323,10 +321,6 @@
       sipType = sipType_QgsProcessingParameterFieldMapping;
     else if ( sipCpp->type() == QgsProcessingParameterTinInputLayers::typeName() )
       sipType = sipType_QgsProcessingParameterTinInputLayers;
-    else if ( sipCpp->type() == QgsProcessingParameterVectorTileWriterLayers::typeName() )
-      sipType = sipType_QgsProcessingParameterVectorTileWriterLayers;
-    else if ( sipCpp->type() == QgsProcessingParameterDxfLayers::typeName() )
-      sipType = sipType_QgsProcessingParameterDxfLayers;
     else if ( sipCpp->type() == QgsProcessingParameterMeshDatasetGroups::typeName() )
       sipType = sipType_QgsProcessingParameterMeshDatasetGroups;
     else if ( sipCpp->type() == QgsProcessingParameterMeshDatasetTime::typeName() )
