--- a/python/core/auto_generated/symbology/qgslinesymbollayer.sip.in	2022-02-18 15:06:10.000000000 +0300
+++ b/python/core/auto_generated/symbology/qgslinesymbollayer.sip.in	2022-04-30 15:18:19.872926300 +0300
@@ -76,16 +76,6 @@
 
    virtual double estimateMaxBleed( const QgsRenderContext &context ) const;
 
-    virtual QVector<qreal> dxfCustomDashPattern( Qgis::RenderUnit &unit ) const;
-
-    virtual Qt::PenStyle dxfPenStyle() const;
-
-    virtual double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual double dxfOffset( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual QColor dxfColor( QgsSymbolRenderContext &context ) const;
-
     virtual bool canCauseArtifactsBetweenAdjacentTiles() const;
 
 
