--- a/python/core/auto_generated/symbology/qgsfillsymbollayer.sip.in	2023-08-21 19:30:48.705620174 +0300
+++ b/python/core/auto_generated/symbology/qgsfillsymbollayer.sip.in	2023-10-04 00:28:34.632754956 +0300
@@ -173,22 +173,6 @@
     virtual double estimateMaxBleed( const QgsRenderContext &context ) const;
 
 
-    virtual double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual QColor dxfColor( QgsSymbolRenderContext &context ) const;
-
-    virtual double dxfAngle( QgsSymbolRenderContext &context ) const;
-
-
-    virtual Qt::PenStyle dxfPenStyle() const;
-
-    virtual QColor dxfBrushColor( QgsSymbolRenderContext &context ) const;
-
-    virtual Qt::BrushStyle dxfBrushStyle() const;
-
-    virtual QImage toTiledPatternImage( ) const;
-
-
   protected:
 
 
@@ -890,10 +874,6 @@
 
     virtual QgsMapUnitScale mapUnitScale() const;
 
-    virtual double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual Qt::PenStyle dxfPenStyle() const;
-
     virtual QVariantMap properties() const;
 
 
@@ -1272,8 +1252,6 @@
 
     virtual double estimateMaxBleed( const QgsRenderContext &context ) const;
 
-    virtual QColor dxfColor( QgsSymbolRenderContext &context ) const;
-
     virtual QSet<QString> usedAttributes( const QgsRenderContext &context ) const;
 
     virtual bool hasDataDefinedProperties() const;
