--- a/python/core/auto_generated/symbology/qgsfillsymbollayer.sip.in	2025-05-15 15:26:35.309274477 +0300
+++ b/python/core/auto_generated/symbology/qgsfillsymbollayer.sip.in	2025-05-15 14:47:00.323278921 +0300
@@ -173,20 +173,6 @@
 
     virtual double estimateMaxBleed( const QgsRenderContext &context ) const;
 
-
-    virtual double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual QColor dxfColor( QgsSymbolRenderContext &context ) const;
-
-    virtual double dxfAngle( QgsSymbolRenderContext &context ) const;
-
-
-    virtual Qt::PenStyle dxfPenStyle() const;
-
-    virtual QColor dxfBrushColor( QgsSymbolRenderContext &context ) const;
-
-    virtual Qt::BrushStyle dxfBrushStyle() const;
-
     virtual QImage toTiledPatternImage( ) const;
 
 
@@ -905,10 +891,6 @@
 
     virtual QgsMapUnitScale mapUnitScale() const;
 
-    virtual double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const;
-
-    virtual Qt::PenStyle dxfPenStyle() const;
-
     virtual QVariantMap properties() const;
 
 
@@ -1390,8 +1372,6 @@
 
     virtual double estimateMaxBleed( const QgsRenderContext &context ) const;
 
-    virtual QColor dxfColor( QgsSymbolRenderContext &context ) const;
-
     virtual QSet<QString> usedAttributes( const QgsRenderContext &context ) const;
 
     virtual bool hasDataDefinedProperties() const;
