--- a/python/core/auto_generated/processing/qgsprocessingutils.sip.in	2025-07-15 20:09:37.213533791 +0300
+++ b/python/core/auto_generated/processing/qgsprocessingutils.sip.in	2025-07-30 19:58:18.104501779 +0300
@@ -43,8 +43,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 %End
 
@@ -73,8 +71,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 %End
 
@@ -96,8 +92,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 
 .. versionadded:: 3.6
@@ -121,8 +115,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 
 .. versionadded:: 3.22
@@ -146,8 +138,6 @@
 
 .. seealso:: :py:func:`compatibleAnnotationLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 
 .. versionadded:: 3.22
@@ -171,38 +161,11 @@
 
 .. seealso:: :py:func:`compatiblePointCloudLayers`
 
-.. seealso:: :py:func:`compatibleVectorTileLayers`
-
 .. seealso:: :py:func:`compatibleLayers`
 
 .. versionadded:: 3.22
 %End
 
-    static QList<QgsVectorTileLayer *> compatibleVectorTileLayers( QgsProject *project, bool sort = true );
-%Docstring
-Returns a list of vector tile layers from a ``project`` which are compatible with the processing
-framework.
-
-If the ``sort`` argument is ``True`` then the layers will be sorted by their :py:func:`QgsMapLayer.name()`
-value.
-
-.. seealso:: :py:func:`compatibleRasterLayers`
-
-.. seealso:: :py:func:`compatibleVectorLayers`
-
-.. seealso:: :py:func:`compatibleMeshLayers`
-
-.. seealso:: :py:func:`compatiblePluginLayers`
-
-.. seealso:: :py:func:`compatiblePointCloudLayers`
-
-.. seealso:: :py:func:`compatibleAnnotationLayers`
-
-.. seealso:: :py:func:`compatibleLayers`
-
-.. versionadded:: 3.32
-%End
-
     static QList< QgsMapLayer * > compatibleLayers( QgsProject *project, bool sort = true );
 %Docstring
 Returns a list of map layers from a ``project`` which are compatible with the processing
@@ -504,8 +467,6 @@
 
 .. seealso:: :py:func:`defaultPointCloudExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.10
 %End
 
@@ -521,8 +482,6 @@
 
 .. seealso:: :py:func:`defaultPointCloudExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.10
 %End
 
@@ -537,27 +496,9 @@
 
 .. seealso:: :py:func:`defaultRasterExtension`
 
-.. seealso:: :py:func:`defaultVectorTileExtension`
-
 .. versionadded:: 3.24
 %End
 
-    static QString defaultVectorTileExtension();
-%Docstring
-Returns the default vector tile extension to use, in the absence of all other constraints (e.g.
-provider based support for extensions).
-
-This method returns a fallback value of "mbtiles".
-
-.. seealso:: :py:func:`defaultVectorExtension`
-
-.. seealso:: :py:func:`defaultRasterExtension`
-
-.. seealso:: :py:func:`defaultPointCloudExtension`
-
-.. versionadded:: 3.32
-%End
-
     static QVariantMap removePointerValuesFromMap( const QVariantMap &map );
 %Docstring
 Removes any raw pointer values from an input ``map``, replacing them with
