--- a/src/core/settings/qgssettingsregistry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/settings/qgssettingsregistry.cpp	2022-04-07 00:39:45.000000000 +0300
@@ -16,9 +16,9 @@
 #include "qgssettingsregistry.h"
 
 #include "qgslayout.h"
-#include "qgslocator.h"
+// #include "qgslocator.h"
 #include "qgsnetworkaccessmanager.h"
-#include "qgsnewsfeedparser.h"
+// #include "qgsnewsfeedparser.h"
 #include "qgsprocessing.h"
 #include "qgsapplication.h"
 #include "qgsgeometryoptions.h"
