--- a/src/core/browser/qgsfilebaseddataitemprovider.cpp	2023-08-21 19:39:45.675050657 +0300
+++ b/src/core/browser/qgsfilebaseddataitemprovider.cpp	2023-10-04 12:13:39.000000000 +0300
@@ -156,8 +156,8 @@
     case Qgis::LayerType::Mesh:
       return Qgis::BrowserLayerType::Mesh;
 
-    case Qgis::LayerType::VectorTile:
-      return Qgis::BrowserLayerType::VectorTile;
+    // case Qgis::LayerType::VectorTile:
+    //   return Qgis::BrowserLayerType::VectorTile;
 
     case Qgis::LayerType::PointCloud:
       return Qgis::BrowserLayerType::PointCloud;
