--- a/src/core/geometry/qgspoint.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/geometry/qgspoint.cpp	2022-03-21 09:56:35.000000000 +0300
@@ -30,7 +30,7 @@
 #include <QRegularExpression>
 #include <QJsonObject>
 #include <QJsonArray>
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 
 /***************************************************************************
  * This class is considered CRITICAL and any change MUST be accompanied with
@@ -324,20 +324,20 @@
 
 json QgsPoint::asJsonObject( int precision ) const
 {
-  json j
-  {
-    { "type", "Point" },
-    { "coordinates", json::array() },
-  };
+  json j;
+  j.Add("type", "Point");
+  CPLJSONArray coordinates;
+  
   if ( ! isEmpty() )
   {
-    j["coordinates"].push_back( qgsRound( mX, precision ) );
-    j["coordinates"].push_back( qgsRound( mY, precision ) );
+    coordinates.Add( qgsRound( mX, precision ) );
+    coordinates.Add( qgsRound( mY, precision ) );
     if ( is3D() )
     {
-      j["coordinates"].push_back( qgsRound( mZ, precision ) );
+      coordinates.Add( qgsRound( mZ, precision ) );
     }
   }
+  j.Add("coordinates", coordinates);
   return j;
 }
 
