--- a/src/core/qgsrange.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/qgsrange.h	2022-05-01 00:28:37.555479100 +0300
@@ -42,6 +42,7 @@
  * \note not available in Python bindings (but class provided for template-based inheritance)
  * \since QGIS 3.0
  */
+ 
 template <typename T>
 class QgsRange
 {
@@ -190,6 +191,7 @@
 
 };
 
+typedef QgsRange<int> QgsRangeintBase;
 
 /**
  * \brief QgsRange which stores a range of double values.
