--- a/src/core/geometry/qgsabstractgeometry.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/geometry/qgsabstractgeometry.h	2022-04-07 00:39:42.000000000 +0300
@@ -26,10 +26,12 @@
 #include "qgswkbtypes.h"
 #include "qgswkbptr.h"
 
-#ifndef SIP_RUN
-#include "json_fwd.hpp"
-using namespace nlohmann;
-#endif
+// #ifndef SIP_RUN
+// #include "json_fwd.hpp"
+// using namespace nlohmann;
+// #endif
+#include "cpl_json.h"
+using json = CPLJSONObject;
 
 class QgsMapToPixel;
 class QgsCurve;
