--- a/src/core/symbology/qgsmarkersymbollayer.cpp	2023-08-21 19:39:45.751045266 +0300
+++ b/src/core/symbology/qgsmarkersymbollayer.cpp	2023-10-04 10:39:44.316569545 +0300
@@ -16,8 +16,8 @@
 #include "qgsmarkersymbollayer.h"
 #include "qgssymbollayerutils.h"
 
-#include "qgsdxfexport.h"
-#include "qgsdxfpaintdevice.h"
+// #include "qgsdxfexport.h"
+// #include "qgsdxfpaintdevice.h"
 #include "qgsfontutils.h"
 #include "qgsimagecache.h"
 #include "qgsimageoperation.h"
@@ -1570,7 +1570,7 @@
     p->drawPath( mPath );
   }
 }
-
+/*
 bool QgsSimpleMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   //data defined size?
@@ -1759,7 +1759,7 @@
 
   return true;
 }
-
+*/
 
 void QgsSimpleMarkerSymbolLayer::setOutputUnit( Qgis::RenderUnit unit )
 {
@@ -2809,7 +2809,7 @@
   m->setOffset( offset );
   return m;
 }
-
+/*
 bool QgsSvgMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   //size
@@ -2930,7 +2930,7 @@
   p.end();
   return true;
 }
-
+*/
 QRectF QgsSvgMarkerSymbolLayer::bounds( QPointF point, QgsSymbolRenderContext &context )
 {
   bool hasDataDefinedSize = false;
