--- a/src/core/auth/qgsauthmanager.cpp	2025-04-25 19:26:53.733230345 +0300
+++ b/src/core/auth/qgsauthmanager.cpp	2025-04-02 18:34:56.458009803 +0300
@@ -75,8 +75,8 @@
 const QString QgsAuthManager::AUTH_CFG_REGEX = QStringLiteral( "authcfg=([a-z]|[A-Z]|[0-9]){7}" );
 
 
-const QLatin1String QgsAuthManager::AUTH_PASSWORD_HELPER_KEY_NAME( "QGIS-Master-Password" );
-const QLatin1String QgsAuthManager::AUTH_PASSWORD_HELPER_FOLDER_NAME( "QGIS" );
+const QLatin1String QgsAuthManager::AUTH_PASSWORD_HELPER_KEY_NAME( "NextGIS-Master-Password" );
+const QLatin1String QgsAuthManager::AUTH_PASSWORD_HELPER_FOLDER_NAME( "NextGIS" );
 
 
 
@@ -90,17 +90,6 @@
 const QString QgsAuthManager::AUTH_PASSWORD_HELPER_DISPLAY_NAME( "Password Manager" );
 #endif
 
-QgsAuthManager *QgsAuthManager::instance()
-{
-  static QMutex sMutex;
-  QMutexLocker locker( &sMutex );
-  if ( !sInstance )
-  {
-    sInstance = new QgsAuthManager( );
-  }
-  return sInstance;
-}
-
 
 QgsAuthManager::QgsAuthManager()
 {
