--- a/src/core/symbology/qgsfillsymbollayer.h	2025-04-25 19:26:53.999236184 +0300
+++ b/src/core/symbology/qgsfillsymbollayer.h	2025-04-02 18:34:56.628177317 +0300
@@ -172,13 +172,13 @@
 
     double estimateMaxBleed( const QgsRenderContext &context ) const override;
 
-    double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    QColor dxfColor( QgsSymbolRenderContext &context ) const override;
-    double dxfAngle( QgsSymbolRenderContext &context ) const override;
-
-    Qt::PenStyle dxfPenStyle() const override;
-    QColor dxfBrushColor( QgsSymbolRenderContext &context ) const override;
-    Qt::BrushStyle dxfBrushStyle() const override;
+    // double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // QColor dxfColor( QgsSymbolRenderContext &context ) const override;
+    // double dxfAngle( QgsSymbolRenderContext &context ) const override;
+
+    // Qt::PenStyle dxfPenStyle() const override;
+    // QColor dxfBrushColor( QgsSymbolRenderContext &context ) const override;
+    // Qt::BrushStyle dxfBrushStyle() const override;
     QImage toTiledPatternImage( ) const override;
 
   protected:
@@ -839,8 +839,8 @@
     Qgis::RenderUnit outputUnit() const override;
     void setMapUnitScale( const QgsMapUnitScale &scale ) override;
     QgsMapUnitScale mapUnitScale() const override;
-    double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    Qt::PenStyle dxfPenStyle() const override;
+    // double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // Qt::PenStyle dxfPenStyle() const override;
     QVariantMap properties() const override;
 
   protected:
@@ -1155,7 +1155,7 @@
     QgsSymbol *subSymbol() override;
     bool setSubSymbol( QgsSymbol *symbol SIP_TRANSFER ) override;
     double estimateMaxBleed( const QgsRenderContext &context ) const override;
-    QColor dxfColor( QgsSymbolRenderContext &context ) const override;
+    // QColor dxfColor( QgsSymbolRenderContext &context ) const override;
     QSet<QString> usedAttributes( const QgsRenderContext &context ) const override;
     bool hasDataDefinedProperties() const override;
 
