--- a/src/core/geometry/qgsabstractgeometry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/geometry/qgsabstractgeometry.cpp	2022-04-06 18:40:09.000000000 +0300
@@ -22,10 +22,12 @@
 #include "qgsgeometrycollection.h"
 #include "qgsvertexid.h"
 
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 #include <limits>
 #include <QTransform>
 
+#include "qgsjsonutils.h"
+
 QgsAbstractGeometry::QgsAbstractGeometry( const QgsAbstractGeometry &geom )
 {
   mWkbType = geom.mWkbType;
@@ -190,12 +192,12 @@
 
 QString QgsAbstractGeometry::asJson( int precision )
 {
-  return QString::fromStdString( asJsonObject( precision ).dump() );
+  return QgsJsonUtils::dump( asJsonObject( precision ) );
 }
 
 json QgsAbstractGeometry::asJsonObject( int precision ) const
 {
-  Q_UNUSED( precision ) return nullptr;
+  Q_UNUSED( precision ) return json();
 }
 
 QgsPoint QgsAbstractGeometry::centroid() const
