--- a/src/core/symbology/qgsfillsymbollayer.h	2023-08-21 19:39:45.747045550 +0300
+++ b/src/core/symbology/qgsfillsymbollayer.h	2023-10-04 11:49:32.015303279 +0300
@@ -172,14 +172,14 @@
 
     double estimateMaxBleed( const QgsRenderContext &context ) const override;
 
-    double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    QColor dxfColor( QgsSymbolRenderContext &context ) const override;
-    double dxfAngle( QgsSymbolRenderContext &context ) const override;
-
-    Qt::PenStyle dxfPenStyle() const override;
-    QColor dxfBrushColor( QgsSymbolRenderContext &context ) const override;
-    Qt::BrushStyle dxfBrushStyle() const override;
-    QImage toTiledPatternImage( ) const override;
+    // double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // QColor dxfColor( QgsSymbolRenderContext &context ) const override;
+    // double dxfAngle( QgsSymbolRenderContext &context ) const override;
+
+    // Qt::PenStyle dxfPenStyle() const override;
+    // QColor dxfBrushColor( QgsSymbolRenderContext &context ) const override;
+    // Qt::BrushStyle dxfBrushStyle() const override;
+    // QImage toTiledPatternImage( ) const override;
 
   protected:
     QBrush mBrush;
@@ -839,8 +839,8 @@
     Qgis::RenderUnit outputUnit() const override;
     void setMapUnitScale( const QgsMapUnitScale &scale ) override;
     QgsMapUnitScale mapUnitScale() const override;
-    double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    Qt::PenStyle dxfPenStyle() const override;
+    // double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // Qt::PenStyle dxfPenStyle() const override;
     QVariantMap properties() const override;
 
   protected:
@@ -1155,7 +1155,7 @@
     QgsSymbol *subSymbol() override;
     bool setSubSymbol( QgsSymbol *symbol SIP_TRANSFER ) override;
     double estimateMaxBleed( const QgsRenderContext &context ) const override;
-    QColor dxfColor( QgsSymbolRenderContext &context ) const override;
+    // QColor dxfColor( QgsSymbolRenderContext &context ) const override;
     QSet<QString> usedAttributes( const QgsRenderContext &context ) const override;
     bool hasDataDefinedProperties() const override;
 
