--- a/src/core/project/qgsproject.cpp	2025-07-15 20:09:37.659543317 +0300
+++ b/src/core/project/qgsproject.cpp	2025-07-30 19:58:18.118961299 +0300
@@ -58,7 +58,7 @@
 #include "qgsprojectviewsettings.h"
 #include "qgsprojectstylesettings.h"
 #include "qgsprojecttimesettings.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsruntimeprofiler.h"
 #include "qgsannotationlayer.h"
 #include "qgspointcloudlayer.h"
@@ -1601,7 +1601,7 @@
       break;
 
     case Qgis::LayerType::VectorTile:
-      mapLayer = std::make_unique<QgsVectorTileLayer>();
+      // mapLayer = std::make_unique<QgsVectorTileLayer>();
       break;
 
     case Qgis::LayerType::PointCloud:
