--- a/src/core/qgsfontutils.cpp	2025-05-15 15:26:35.467372788 +0300
+++ b/src/core/qgsfontutils.cpp	2025-05-15 14:47:00.777294754 +0300
@@ -284,7 +284,7 @@
     else
     {
       bool loaded = false;
-      if ( QgsApplication::isRunningFromBuildDir() )
+      if ( false && QgsApplication::isRunningFromBuildDir() )
       {
         // workaround for bugs with Qt 4.8.5 (other versions?) on Mac 10.9, where fonts
         // from qrc resources load but fail to work and default font is substituted [LS]:
