--- a/src/core/symbology/qgssymbollayer.cpp	2025-05-15 15:26:35.515573152 +0300
+++ b/src/core/symbology/qgssymbollayer.cpp	2025-05-15 14:47:00.834296742 +0300
@@ -15,7 +15,7 @@
 
 #include "qgssymbollayer.h"
 #include "qgsrendercontext.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgsgeometrysimplifier.h"
 #include "qgspainteffect.h"
 #include "qgspainteffectregistry.h"
@@ -153,6 +153,7 @@
   return false;
 }
 
+/*
 bool QgsSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   Q_UNUSED( e )
@@ -217,6 +218,7 @@
 {
   return Qt::NoBrush;
 }
+*/
 
 QgsPaintEffect *QgsSymbolLayer::paintEffect() const
 {
@@ -821,12 +823,13 @@
   return context.convertToPainterUnits( mWidth, mWidthUnit, mWidthMapUnitScale );
 }
 
+/*
 double QgsLineSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   Q_UNUSED( context )
   return width() * QgsDxfExport::mapUnitScaleFactor( e.symbologyScale(), widthUnit(), e.mapUnits(), context.renderContext().mapToPixel().mapUnitsPerPixel() );
 }
-
+*/
 
 void QgsFillSymbolLayer::drawPreviewIcon( QgsSymbolRenderContext &context, QSize size )
 {
@@ -921,6 +924,7 @@
   return {};
 }
 
+/*
 double QgsMarkerSymbolLayer::dxfSize( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double size = mSize;
@@ -954,6 +958,7 @@
   }
   return angle;
 }
+*/
 
 QPainterPath generateClipPath( const QgsRenderContext &renderContext, const QString &id, const QRectF *rect, bool &foundGeometries )
 {
