--- a/src/core/network/qgsnetworkaccessmanager.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/network/qgsnetworkaccessmanager.h	2022-03-23 18:27:08.000000000 +0300
@@ -609,7 +609,7 @@
 
 #ifndef SIP_RUN
     //! Settings entry network timeout
-    static const inline QgsSettingsEntryInteger settingsNetworkTimeout = QgsSettingsEntryInteger( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), QgsSettings::NoSection, 60000, QObject::tr( "Network timeout" ) );
+    static const inline QgsSettingsEntryInteger settingsNetworkTimeout = QgsSettingsEntryInteger( QStringLiteral( "/qgis/networkAndProxy/networkTimeout" ), QgsSettings::NoSection, 300000, QObject::tr( "Network timeout" ) );
 #endif
 
     /**
