--- a/src/core/qgscplhttpfetchoverrider.h	2025-07-14 17:28:14.873544431 +0300
+++ b/src/core/qgscplhttpfetchoverrider.h	2025-07-14 13:59:40.758926827 +0300
@@ -42,7 +42,7 @@
  * \note not available in Python bindings
  * \since QGIS 3.18
  */
-class QgsCPLHTTPFetchOverrider
+class CORE_EXPORT QgsCPLHTTPFetchOverrider
 {
   public:
     //! Installs the redirection for the current thread
@@ -68,7 +68,7 @@
      */
     QThread *thread() const;
 
-  private:
+  protected:
 
 #if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,2,0)
     static CPLHTTPResult *callback( const char *pszURL,
