--- a/src/core/network/qgsnetworkaccessmanager.cpp	2025-05-15 15:26:35.436461450 +0300
+++ b/src/core/network/qgsnetworkaccessmanager.cpp	2025-05-15 14:47:00.726292976 +0300
@@ -45,7 +45,7 @@
 #include <QStandardPaths>
 #include <QUuid>
 
-const QgsSettingsEntryInteger *QgsNetworkAccessManager::settingsNetworkTimeout = new QgsSettingsEntryInteger( QStringLiteral( "network-timeout" ), QgsSettingsTree::sTreeNetwork, 60000, QObject::tr( "Network timeout" ) );
+const QgsSettingsEntryInteger *QgsNetworkAccessManager::settingsNetworkTimeout = new QgsSettingsEntryInteger( QStringLiteral( "network-timeout" ), QgsSettingsTree::sTreeNetwork, 300000, QObject::tr( "Network timeout" ) );
 
 #ifndef QT_NO_SSL
 #include <QSslConfiguration>
