--- a/src/core/settings/qgssettingsregistrycore.cpp	2025-07-15 20:09:37.765545580 +0300
+++ b/src/core/settings/qgssettingsregistrycore.cpp	2025-07-30 19:58:18.120961370 +0300
@@ -22,14 +22,16 @@
 #include "qgssettingsentryenumflag.h"
 #include "qgssettings.h"
 
-#include "qgsbabelformatregistry.h"
+// #include "qgsbabelformatregistry.h"
 #include "qgslayout.h"
 #include "qgslocator.h"
 #include "qgsnetworkaccessmanager.h"
 #include "qgsowsconnection.h"
 #include "qgsprocessing.h"
+/*
 #include "qgsvectortileconnection.h"
 #include "qgsgpsdetector.h"
+*/
 #include "qgsrasterlayer.h"
 #include "qgsvectorlayer.h"
 
@@ -140,12 +142,14 @@
 
   settingsLayerTreeShowFeatureCountForNewLayers->copyValueFromKey( QStringLiteral( "core/layer-tree/show_feature_count_for_new_layers" ) );
 
+  /*
 #if defined( HAVE_QTSERIALPORT )
   QgsGpsDetector::settingsGpsStopBits->copyValueFromKey( QStringLiteral( "core/gps/stop_bits" ) );
   QgsGpsDetector::settingsGpsFlowControl->copyValueFromKey( QStringLiteral( "core/gps/flow_control" ) );
   QgsGpsDetector::settingsGpsDataBits->copyValueFromKey( QStringLiteral( "core/gps/data_bits" ) );
   QgsGpsDetector::settingsGpsParity->copyValueFromKey( QStringLiteral( "core/gps/parity" ) );
 #endif
+  */
 
   QgsRasterLayer::settingsRasterDefaultOversampling->copyValueFromKey( QStringLiteral( "Raster/defaultOversampling" ), true );
   QgsRasterLayer::settingsRasterDefaultEarlyResampling->copyValueFromKey( QStringLiteral( "Raster/defaultEarlyResampling" ), true );
@@ -276,6 +280,7 @@
   }
 
   // Vector tile - added in 3.30
+  /*
   {
     QgsSettings settings;
     settings.beginGroup( QStringLiteral( "qgis/connections-vector-tile" ) );
@@ -297,6 +302,7 @@
       Q_NOWARN_DEPRECATED_POP
     }
   }
+  */
 
   // xyz - added in 3.30
   {
@@ -350,6 +356,7 @@
   }
 
   // babel devices settings - added in 3.30
+  /*
   {
     if ( QgsBabelFormatRegistry::sTreeBabelDevices->items().count() == 0 )
     {
@@ -366,6 +373,7 @@
       }
     }
   }
+  */
 }
 
 // TODO QGIS 4.0: Remove
@@ -480,6 +488,7 @@
   }
 
   // Vector tile - added in 3.30
+  /*
   {
     QgsSettings settings;
     settings.beginGroup( QStringLiteral( "qgis/connections-vector-tile" ) );
@@ -504,6 +513,7 @@
       Q_NOWARN_DEPRECATED_POP
     }
   }
+  */
 
   // xyz - added in 3.30
   {
@@ -555,6 +565,7 @@
   }
 
   // babel devices settings - added in 3.30
+  /*
   {
     const QStringList devices = QgsBabelFormatRegistry::sTreeBabelDevices->items();
     QgsSettings().setValue( QStringLiteral( "/Plugin-GPS/devices/deviceList" ), devices );
@@ -568,4 +579,5 @@
       QgsBabelFormatRegistry::settingsBabelTrkUpload->copyValueToKey( QStringLiteral( "/Plugin-GPS/devices/%1/trkupload" ), {device} );
     }
   }
+  */
 }
