--- a/src/core/project/qgsproject.cpp	2025-05-15 15:26:35.452666065 +0300
+++ b/src/core/project/qgsproject.cpp	2025-05-15 14:47:00.749293778 +0300
@@ -58,7 +58,7 @@
 #include "qgsprojectviewsettings.h"
 #include "qgsprojectstylesettings.h"
 #include "qgsprojecttimesettings.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgstiledscenelayer.h"
 #include "qgsruntimeprofiler.h"
 #include "qgsannotationlayer.h"
@@ -1841,7 +1841,7 @@
       break;
 
     case Qgis::LayerType::VectorTile:
-      mapLayer = std::make_unique<QgsVectorTileLayer>();
+      // mapLayer = std::make_unique<QgsVectorTileLayer>();
       break;
 
     case Qgis::LayerType::PointCloud:
