--- a/src/core/textrenderer/qgsfontmanager.cpp	2025-05-15 15:26:35.523448145 +0300
+++ b/src/core/textrenderer/qgsfontmanager.cpp	2025-05-15 14:47:00.839296917 +0300
@@ -246,6 +246,7 @@
 QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &family, QString &matchedFamily ) const
 {
   // this list is built using scripts/process_google_fonts.py
+  /*
   const thread_local std::vector< QgsFontDownloadDetails > sGoogleFonts
   {
     GoogleFontDetails( QStringLiteral( "ABeeZee" ), { QStringLiteral( "ofl/abeezee/ABeeZee-Regular.ttf" ), QStringLiteral( "ofl/abeezee/ABeeZee-Italic.ttf" ) }, QStringLiteral( "ofl/abeezee/OFL.txt" ) ),
@@ -1809,6 +1810,7 @@
       return candidate;
     }
   }
+  */
 
   return QgsFontDownloadDetails();
 }
