--- a/src/core/qgslayerdefinition.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/qgslayerdefinition.cpp	2022-03-21 09:56:36.000000000 +0300
@@ -28,7 +28,7 @@
 #include "qgsrasterlayer.h"
 #include "qgsreadwritecontext.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsapplication.h"
 #include "qgsmaplayerfactory.h"
 #include "qgsmeshlayer.h"
@@ -331,9 +331,9 @@
           layer = new QgsMeshLayer();
           break;
 
-        case QgsMapLayerType::VectorTileLayer:
-          layer = new QgsVectorTileLayer;
-          break;
+        // case QgsMapLayerType::VectorTileLayer:
+        //   layer = new QgsVectorTileLayer;
+        //   break;
 
         case QgsMapLayerType::PointCloudLayer:
           layer = new QgsPointCloudLayer();
