--- a/src/core/symbology/qgslinesymbollayer.h	2025-05-15 15:26:35.505578818 +0300
+++ b/src/core/symbology/qgslinesymbollayer.h	2025-05-15 14:47:00.826428895 +0300
@@ -87,11 +87,11 @@
     void setMapUnitScale( const QgsMapUnitScale &scale ) override;
     QgsMapUnitScale mapUnitScale() const override;
     double estimateMaxBleed( const QgsRenderContext &context ) const override;
-    QVector<qreal> dxfCustomDashPattern( Qgis::RenderUnit &unit ) const override;
-    Qt::PenStyle dxfPenStyle() const override;
-    double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    double dxfOffset( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
-    QColor dxfColor( QgsSymbolRenderContext &context ) const override;
+    // QVector<qreal> dxfCustomDashPattern( Qgis::RenderUnit &unit ) const override;
+    // Qt::PenStyle dxfPenStyle() const override;
+    // double dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // double dxfOffset( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const override;
+    // QColor dxfColor( QgsSymbolRenderContext &context ) const override;
     bool canCauseArtifactsBetweenAdjacentTiles() const override;
 
     /**
