--- a/src/core/qgsapplication.cpp	2025-07-15 20:09:37.677543701 +0300
+++ b/src/core/qgsapplication.cpp	2025-08-18 15:05:38.216386628 +0300
@@ -68,7 +68,7 @@
 #include "qgsstyle.h"
 #include "qgsprojutils.h"
 #include "qgsvaliditycheckregistry.h"
-#include "qgsnewsfeedparser.h"
+// #include "qgsnewsfeedparser.h"
 #include "qgsbookmarkmanager.h"
 #include "qgsstylemodel.h"
 #include "qgsconnectionregistry.h"
@@ -77,16 +77,16 @@
 #include "qgsfeaturestore.h"
 #include "qgslocator.h"
 #include "qgsreadwritelocker.h"
-#include "qgsbabelformatregistry.h"
+// #include "qgsbabelformatregistry.h"
 #include "qgsdbquerylog.h"
 #include "qgsfontmanager.h"
 #include "qgsunsetattributevalue.h"
 #include "qgscolorrampimpl.h"
 #include "qgsinterval.h"
-#include "qgsgpsconnection.h"
+// #include "qgsgpsconnection.h"
 #include "qgssensorregistry.h"
 
-#include "gps/qgsgpsconnectionregistry.h"
+// #include "gps/qgsgpsconnectionregistry.h"
 #include "processing/qgsprocessingregistry.h"
 #include "processing/models/qgsprocessingmodelchildparametersource.h"
 #include "processing/models/qgsprocessingmodelchilddependency.h"
@@ -150,6 +150,24 @@
 
 #include <proj.h>
 
+#include "ngauthmanager.h"
+
+#ifdef Q_OS_MAC
+static void translationPath(const QString &basePath,
+                               QList<QString> &localePaths)
+{
+    QDir baseDir(basePath);
+    QStringList filters;
+    filters << QString("ngstd_*.framework");
+    QStringList list = baseDir.entryList(filters);
+    for (const QString &subPath : list) {
+        const QString &libTrPath = basePath + "/" + subPath +
+                "/Resources/translations";
+        localePaths.append(libTrPath);
+    }
+}
+#endif // Q_OS_MAC
+
 #if defined(Q_OS_LINUX)
 #include <sys/sysinfo.h>
 #endif
@@ -162,6 +180,9 @@
 const char *QgsApplication::QGIS_ORGANIZATION_NAME = "QGIS";
 const char *QgsApplication::QGIS_ORGANIZATION_DOMAIN = "qgis.org";
 const char *QgsApplication::QGIS_APPLICATION_NAME = "QGIS3";
+const char *QgsApplication::NGQGIS_ORGANIZATION_NAME = VENDOR;
+const char *QgsApplication::NGQGIS_ORGANIZATION_DOMAIN = VENDOR_DOMAIN;
+const char *QgsApplication::NGQGIS_APPLICATION_NAME = APP_NAME;
 QgsApplication::ApplicationMembers *QgsApplication::sApplicationMembers = nullptr;
 QgsAuthManager *QgsApplication::sAuthManager = nullptr;
 int ABISYM( QgsApplication::sMaxThreads ) = -1;
@@ -228,6 +249,7 @@
 
   if ( profileFolder.isEmpty() )
   {
+    /*
     if ( getenv( "QGIS_CUSTOM_CONFIG_PATH" ) )
     {
       profileFolder = getenv( "QGIS_CUSTOM_CONFIG_PATH" );
@@ -236,6 +258,24 @@
     {
       profileFolder = QStandardPaths::standardLocations( QStandardPaths::AppDataLocation ).value( 0 );
     }
+    */
+    profileFolder = QString::fromLocal8Bit( qgetenv( "QGIS_CUSTOM_CONFIG_PATH" ) );
+
+    if( profileFolder.isEmpty() )
+    {
+        QString config;
+      #ifdef Q_OS_MAC
+        config = QStringLiteral("Library/Application Support");
+      #else
+        config = QStringLiteral(".config");
+      #endif
+        profileFolder = QString( "%1%2qgis%3" )
+                .arg( QDir::homePath() + QDir::separator() + config + QDir::separator())
+                .arg( QStringLiteral(VENDOR) + QDir::separator() )
+                .arg( VERSION_INT / 10000 );
+        qputenv("QGIS_CUSTOM_CONFIG_PATH", profileFolder.toUtf8());
+    }
+
     // This will normally get here for custom scripts that use QgsApplication.
     // This doesn't get this hit for QGIS Desktop because we setup the profile via main
     QString rootProfileFolder = QgsUserProfileManager::resolveProfilesFolder( profileFolder );
@@ -281,7 +321,7 @@
     qRegisterMetaType<QgsPoint>( "QgsPoint" );
     qRegisterMetaType<QgsDatumTransform::GridDetails>( "QgsDatumTransform::GridDetails" );
     qRegisterMetaType<QgsDatumTransform::TransformDetails>( "QgsDatumTransform::TransformDetails" );
-    qRegisterMetaType<QgsNewsFeedParser::Entry>( "QgsNewsFeedParser::Entry" );
+    // qRegisterMetaType<QgsNewsFeedParser::Entry>( "QgsNewsFeedParser::Entry" );
     qRegisterMetaType<QgsRectangle>( "QgsRectangle" );
     qRegisterMetaType<QgsLocatorResult>( "QgsLocatorResult" );
     qRegisterMetaType<QgsGradientColorRamp>( "QgsGradientColorRamp" );
@@ -309,7 +349,7 @@
     qRegisterMetaType<QMap<QNetworkRequest::KnownHeaders, QVariant>>( "QMap<QNetworkRequest::KnownHeaders,QVariant>" );
     qRegisterMetaType<QList<QNetworkReply::RawHeaderPair>>( "QList<QNetworkReply::RawHeaderPair>" );
     qRegisterMetaType< QAuthenticator * >( "QAuthenticator*" );
-    qRegisterMetaType< QgsGpsInformation >( "QgsGpsInformation" );
+    // qRegisterMetaType< QgsGpsInformation >( "QgsGpsInformation" );
   } );
 
   ( void ) resolvePkgPath();
@@ -372,9 +412,14 @@
   *sDefaultSvgPaths() << qgisSettingsDirPath() + QStringLiteral( "svg/" );
 
   *sAuthDbDirPath() = qgisSettingsDirPath();
-  if ( getenv( "QGIS_AUTH_DB_DIR_PATH" ) )
+  QString authDBDirPath = QString::fromLocal8Bit( qgetenv( "QGIS_AUTH_DB_DIR_PATH" ) );
+  if (!authDBDirPath.isEmpty())
   {
-    setAuthDatabaseDirPath( getenv( "QGIS_AUTH_DB_DIR_PATH" ) );
+    setAuthDatabaseDirPath( authDBDirPath );
+  }
+  else
+  {
+    setAuthDatabaseDirPath( qgisSettingsDirPath() );
   }
 
   // force use of OpenGL renderer for Qt3d.
@@ -400,6 +445,7 @@
   *sSystemEnvVars() = systemEnvVarMap;
 
   // append local user-writable folder as a proj search path
+  /*
   QStringList currentProjSearchPaths = QgsProjUtils::searchPaths();
   currentProjSearchPaths.append( qgisSettingsDirPath() + QStringLiteral( "proj" ) );
 #ifdef Q_OS_MACX
@@ -422,6 +468,7 @@
     CPLFree( newPaths[i] );
   }
   delete [] newPaths;
+  */
 
   // allow Qt to search for Qt plugins (e.g. sqldrivers) in our plugin directory
   QCoreApplication::addLibraryPath( pluginPath() );
@@ -458,6 +505,55 @@
 
 void QgsApplication::installTranslators()
 {
+  if ( *sTranslation() != QStringLiteral( "C" ) )
+  {
+      QList<QString> localePaths;
+      localePaths.append(i18nPath());
+      #ifdef Q_OS_MAC
+        translationPath(QApplication::applicationDirPath() +
+                             "/Contents/Frameworks/", localePaths);
+        translationPath(QApplication::applicationDirPath() +
+                             "/../../Contents/Frameworks/", localePaths);
+        translationPath(QApplication::applicationDirPath() +
+                             "/../Library/Frameworks/", localePaths);
+        translationPath(QApplication::applicationDirPath() +
+                             "/../../../../Library/Frameworks/", localePaths);
+      #else
+        const QString &libTrPath = QApplication::applicationDirPath()
+                  + QStringLiteral("/../share/translations");
+        localePaths.append(libTrPath);
+      #endif // Q_OS_MAC
+
+        localePaths.append(QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+
+    QStringList translationFilters;
+    translationFilters << QString("ngstd_core_%1").arg(*sTranslation());
+    translationFilters << QString("ngstd_framework_%1").arg(*sTranslation());
+    translationFilters << QString("qt_%1").arg(*sTranslation());
+    translationFilters << QString("qtbase_%1").arg(*sTranslation());
+    translationFilters << QString("qgis_%1").arg(*sTranslation());
+    translationFilters << QString("ngqgis_%1").arg(*sTranslation());
+    translationFilters << QString("qscintilla_%1").arg(*sTranslation());
+
+    for(const QString &localePath : localePaths) {
+        QgsDebugMsgLevel( QStringLiteral( "Translation path ") + localePath + QStringLiteral( " -- get translation files: "), 2);
+        for (const QString &fileName : translationFilters) {
+            QTranslator *translator = new QTranslator;
+            QString loadFile = QFileInfo(localePath, fileName).filePath();
+            if (translator->load(fileName, localePath)) {
+                QgsDebugMsgLevel( QStringLiteral( "Loaded translation file ") +  loadFile, 2);
+                installTranslator(translator);
+                mTranslators.push_back(translator);
+            }
+            else {
+                QgsDebugMsgLevel(QStringLiteral("Loading of translation failed [") + loadFile + QStringLiteral("]"), 2);
+                delete translator;
+            }
+        }
+    }
+  }
+
+  /*
   if ( *sTranslation() != QLatin1String( "C" ) )
   {
     mQgisTranslator = new QTranslator( this );
@@ -469,12 +565,14 @@
     {
       QgsDebugMsgLevel( QStringLiteral( "loading of qgis translation failed %1/qgis_%2" ).arg( i18nPath(), *sTranslation() ), 2 );
     }
+    */
 
     /* Translation file for Qt.
      * The strings from the QMenuBar context section are used by Qt/Mac to shift
      * the About, Preferences and Quit items to the Mac Application menu.
      * These items must be translated identically in both qt_ and qgis_ files.
      */
+    /*
     QString qtTranslationsPath = QLibraryInfo::location( QLibraryInfo::TranslationsPath );
 #ifdef __MINGW32__
     QString prefix = QDir( QString( "%1/../" ).arg( QApplication::applicationDirPath() ) ).absolutePath();
@@ -501,6 +599,7 @@
       QgsDebugMsgLevel( QStringLiteral( "loading of qtbase translation failed %1/qt_%2" ).arg( qtTranslationsPath, *sTranslation() ), 2 );
     }
   }
+  */
 }
 
 QgsApplication::~QgsApplication()
@@ -510,9 +609,14 @@
 
   delete mDataItemProviderRegistry;
   delete mApplicationMembers;
+  /*
   delete mQgisTranslator;
   delete mQtTranslator;
   delete mQtBaseTranslator;
+  */
+  for (QTranslator *translator : mTranslators) {
+      delete translator;
+  }
 
   // we do this here as well as in exitQgis() -- it's safe to call as often as we want,
   // and there's just a *chance* that someone hasn't properly called exitQgis prior to
@@ -626,12 +730,14 @@
 void QgsApplication::setPrefixPath( const QString &prefixPath, bool useDefaultPaths )
 {
   *sPrefixPath() = prefixPath;
+  /*
 #if defined(Q_OS_WIN)
   if ( sPrefixPath()->endsWith( "/bin" ) )
   {
     sPrefixPath()->chop( 4 );
   }
 #endif
+  */
   if ( useDefaultPaths && !ABISYM( mRunningFromBuildDir ) )
   {
     setPluginPath( *sPrefixPath() + '/' + QStringLiteral( QGIS_PLUGIN_SUBDIR ) );
@@ -947,6 +1053,29 @@
     }
   }
 
+#ifdef Q_OS_WIN
+    QDir defaultPrefixDir(applicationDirPath() + QDir::separator() + "..");
+
+    // Add search gdal utilities paths
+    QString path = defaultPrefixDir.absolutePath();
+    qputenv("PATH", QString( "%1;%2" ).arg(path).arg(applicationDirPath()).toUtf8());
+
+#elif defined(Q_OS_MAC)
+    QDir defaultPrefixDir(applicationDirPath() + QDir::separator() + ".."
+                           + QDir::separator() + ".."
+                           + QDir::separator() + ".."
+                           + QDir::separator() + "..");
+#else
+    QDir defaultPrefixDir("/usr");
+#endif
+    QString defaultPrefixPath =  defaultPrefixDir.absolutePath();
+
+    QString envPrefixPath = QString::fromLocal8Bit( qgetenv( "QGIS_PREFIX_PATH" ) );
+    QString prefixPath( envPrefixPath.isEmpty() ?
+                            defaultPrefixPath : envPrefixPath );
+    QgsDebugMsgLevel( QString( "prefixPath(): %1" ).arg( prefixPath ), 2 );
+
+  /*
   QString prefixPath;
   if ( getenv( "QGIS_PREFIX_PATH" ) )
     prefixPath = getenv( "QGIS_PREFIX_PATH" );
@@ -977,10 +1106,16 @@
 #endif
 #endif
   }
+  */
+
+  setPrefixPath( prefixPath, true );
+  qputenv("QGIS_PREFIX_PATH", prefixPath.toUtf8());
 
+  /*
   if ( ABISYM( mRunningFromBuildDir ) )
     return *sBuildOutputPath() + QStringLiteral( "/data" );
   else
+  */
     return prefixPath + '/' + QStringLiteral( QGIS_DATA_SUBDIR );
 }
 
@@ -1139,9 +1274,11 @@
 
 QString QgsApplication::i18nPath()
 {
+  /*
   if ( ABISYM( mRunningFromBuildDir ) )
     return *sBuildOutputPath() + QStringLiteral( "/i18n/" );
   else
+  */
     return pkgDataPath() + QStringLiteral( "/i18n/" );
 }
 
@@ -1182,6 +1319,7 @@
 
 QString QgsApplication::srsDatabaseFilePath()
 {
+  /*
   if ( ABISYM( mRunningFromBuildDir ) )
   {
     QString tempCopy = QDir::tempPath() + "/srs6.db";
@@ -1199,8 +1337,11 @@
   }
   else
   {
+  */
     return pkgDataPath() + QStringLiteral( "/resources/srs.db" );
+  /*
   }
+  */
 }
 
 void QgsApplication::setSvgPaths( const QStringList &svgPaths )
@@ -1298,12 +1439,12 @@
     return *sUserName();
 
   //backup plan - use environment variables
-  *sUserName() = qgetenv( "USER" );
+  *sUserName() = QString::fromLocal8Bit(qgetenv( "USER" ));
   if ( !sUserName()->isEmpty() )
     return *sUserName();
 
   //last resort
-  *sUserName() = qgetenv( "USERNAME" );
+  *sUserName() = QString::fromLocal8Bit(qgetenv( "USERNAME" ));
   return *sUserName();
 }
 
@@ -1329,8 +1470,10 @@
   //fall back to login name
   if ( sUserFullName()->isEmpty() )
     *sUserFullName() = userLoginName();
+  /*
 #elif defined(Q_OS_ANDROID) || defined(__MINGW32__)
   *sUserFullName() = QStringLiteral( "Not available" );
+  */
 #else
   struct passwd *p = getpwuid( getuid() );
 
@@ -1526,6 +1669,7 @@
   {
     if ( !lInstance->mAuthManager )
     {
+      NGAuthManager::initInstance();
       lInstance->mAuthManager = QgsAuthManager::instance();
     }
     return lInstance->mAuthManager;
@@ -1534,7 +1678,10 @@
   {
     // no QgsApplication instance
     if ( !sAuthManager )
+    {
+      NGAuthManager::initInstance();
       sAuthManager = QgsAuthManager::instance();
+    }
     return sAuthManager;
   }
 }
@@ -1583,28 +1730,36 @@
 
 QString QgsApplication::showSettings()
 {
-  QString myEnvironmentVar( getenv( "QGIS_PREFIX_PATH" ) );
+  QString myEnvironmentVar = QString::fromLocal8Bit(qgetenv( "QGIS_PREFIX_PATH" ) );
   QString myState = tr( "Application state:\n"
                         "QGIS_PREFIX_PATH env var:\t\t%1\n"
                         "Prefix:\t\t%2\n"
-                        "Plugin Path:\t\t%3\n"
-                        "Package Data Path:\t%4\n"
-                        "Active Theme Name:\t%5\n"
-                        "Active Theme Path:\t%6\n"
-                        "Default Theme Path:\t%7\n"
-                        "SVG Search Paths:\t%8\n"
-                        "User DB Path:\t%9\n"
-                        "Auth DB Path:\t%10\n" )
+                        "Library Path:\t\t%3\n"
+                        "Plugin Path:\t\t%4\n"
+                        "Package Data Path:\t%5\n"
+                        "Active Theme Name:\t%6\n"
+                        "Active Theme Path:\t%7\n"
+                        "Default Theme Path:\t%8\n"
+                        "SVG Search Paths:\t%9\n"
+                        "User DB Path:\t%10\n"
+                        "Auth DB Path:\t%11\n"
+                        "Translations Path:\t%12\n"
+                        "SRS DB Path:\t%13\n"
+                        "Icons Path:\t%14\n"  )
                     .arg( myEnvironmentVar,
                           prefixPath(),
+                          libraryPath(),
                           pluginPath(),
                           pkgDataPath(),
                           themeName(),
                           activeThemePath(),
                           defaultThemePath(),
-                          svgPaths().join( tr( "\n\t\t", "match indentation of application state" ) ),
-                          qgisMasterDatabaseFilePath() )
-                    .arg( qgisAuthDatabaseFilePath() );
+                          svgPaths().join( tr( "\n\t\t", "match indentation of application state" ) ) )
+                    .arg( qgisMasterDatabaseFilePath(),
+                          qgisAuthDatabaseFilePath(),
+                          i18nPath(),
+                          srsDatabaseFilePath(),
+                          iconsPath() );
   return myState;
 }
 
@@ -2113,7 +2268,7 @@
   ApplicationMembers *appMembers = members();
   if ( appMembers->mNullRepresentation.isNull() )
   {
-    appMembers->mNullRepresentation = QgsSettings().value( QStringLiteral( "qgis/nullValue" ), QStringLiteral( "NULL" ) ).toString();
+    appMembers->mNullRepresentation = QgsSettings().value( QStringLiteral( "qgis/nullValue" ), QStringLiteral( "<NULL>" ) ).toString();
   }
   return appMembers->mNullRepresentation;
 }
@@ -2154,8 +2309,7 @@
 #else
   //under other OS's we use an environment var so the user can
   //override the path if he likes
-  int myChangeFlag = 0; //whether we want to force the env var to change
-  setenv( "GDAL_PAM_PROXY_DIR", myPamPath.toUtf8(), myChangeFlag );
+  qputenv( "GDAL_PAM_PROXY_DIR", myPamPath.toUtf8() );
 #endif
 
   // Check qgis.db and make private copy if necessary
@@ -2503,15 +2657,19 @@
   return members()->mSensorRegistry;
 }
 
+/*
 QgsGpsConnectionRegistry *QgsApplication::gpsConnectionRegistry()
 {
   return members()->mGpsConnectionRegistry;
 }
+*/
 
+/*
 QgsBabelFormatRegistry *QgsApplication::gpsBabelFormatRegistry()
 {
   return members()->mGpsBabelFormatRegistry;
 }
+*/
 
 QgsPluginLayerRegistry *QgsApplication::pluginLayerRegistry()
 {
@@ -2737,6 +2895,7 @@
     mPointCloudRendererRegistry = new QgsPointCloudRendererRegistry();
     profiler->end();
   }
+  /*
   {
     profiler->start( tr( "Setup GPS registry" ) );
     mGpsConnectionRegistry = new QgsGpsConnectionRegistry();
@@ -2747,6 +2906,7 @@
     mGpsBabelFormatRegistry = new QgsBabelFormatRegistry();
     profiler->end();
   }
+  */
   {
     profiler->start( tr( "Setup plugin layer registry" ) );
     mPluginLayerRegistry = new QgsPluginLayerRegistry();
@@ -2840,8 +3000,10 @@
   delete mAnnotationRegistry;
   delete mColorSchemeRegistry;
   delete mFieldFormatterRegistry;
+  /*
   delete mGpsConnectionRegistry;
   delete mGpsBabelFormatRegistry;
+  */
   delete mMessageLog;
   delete mPaintEffectRegistry;
   delete mPluginLayerRegistry;
@@ -2890,3 +3052,8 @@
     return sApplicationMembers;
   }
 }
+
+QString QgsApplication::fontsPath()
+{
+    return pkgDataPath() + QDir::separator() + QStringLiteral( "fonts" );
+}
