--- a/python/core/auto_additions/qgsprocessingutils.py	2025-07-15 20:09:37.184533172 +0300
+++ b/python/core/auto_additions/qgsprocessingutils.py	2025-07-30 19:58:18.104501779 +0300
@@ -18,8 +18,10 @@
 QgsProcessingUtils.Annotation = QgsProcessingUtils.LayerHint.Annotation
 QgsProcessingUtils.Annotation.is_monkey_patched = True
 QgsProcessingUtils.LayerHint.Annotation.__doc__ = "Annotation layer type, since QGIS 3.22"
+"""
 QgsProcessingUtils.VectorTile = QgsProcessingUtils.LayerHint.VectorTile
 QgsProcessingUtils.VectorTile.is_monkey_patched = True
 QgsProcessingUtils.LayerHint.VectorTile.__doc__ = "Vector tile layer type, since QGIS 3.32"
 QgsProcessingUtils.LayerHint.__doc__ = 'Layer type hints.\n\n.. versionadded:: 3.4\n\n' + '* ``UnknownType``: ' + QgsProcessingUtils.LayerHint.UnknownType.__doc__ + '\n' + '* ``Vector``: ' + QgsProcessingUtils.LayerHint.Vector.__doc__ + '\n' + '* ``Raster``: ' + QgsProcessingUtils.LayerHint.Raster.__doc__ + '\n' + '* ``Mesh``: ' + QgsProcessingUtils.LayerHint.Mesh.__doc__ + '\n' + '* ``PointCloud``: ' + QgsProcessingUtils.LayerHint.PointCloud.__doc__ + '\n' + '* ``Annotation``: ' + QgsProcessingUtils.LayerHint.Annotation.__doc__ + '\n' + '* ``VectorTile``: ' + QgsProcessingUtils.LayerHint.VectorTile.__doc__
+"""
 # --
