--- a/src/core/qgslayerdefinition.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/qgslayerdefinition.cpp	2022-04-25 18:33:18.000000000 +0300
@@ -27,7 +27,7 @@
 #include "qgsrasterlayer.h"
 #include "qgsreadwritecontext.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsapplication.h"
 #include "qgsmaplayerfactory.h"
 #include "qgsmeshlayer.h"
@@ -331,9 +331,9 @@
           layer = new QgsMeshLayer();
           break;
 
-        case Qgis::LayerType::VectorTile:
-          layer = new QgsVectorTileLayer;
-          break;
+        // case Qgis::LayerType::VectorTile:
+        //   layer = new QgsVectorTileLayer;
+        //   break;
 
         case Qgis::LayerType::PointCloud:
           layer = new QgsPointCloudLayer();
