--- a/src/core/qgsjsonutils.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/qgsjsonutils.h	2022-04-07 00:39:42.000000000 +0300
@@ -27,6 +27,9 @@
 
 class QTextCodec;
 
+#include "cpl_json.h"
+using json = CPLJSONObject;
+
 /**
  * \ingroup core
  * \class QgsJsonExporter
@@ -383,6 +386,12 @@
      */
     static QVariant parseJson( const QString &jsonString ) SIP_SKIP;
 
+    static bool is_array(const json &obj) SIP_SKIP;
+    static bool is_object(const json &obj) SIP_SKIP;
+    static bool is_number_integer(const json &obj) SIP_SKIP;
+    static QString dump(const json &obj) SIP_SKIP;
+    static json parse( const std::string &jsonString, QString &error ) SIP_SKIP;
+
 };
 
 #endif // QGSJSONUTILS_H
