--- a/src/core/qgsfontutils.cpp	2025-07-15 20:09:37.691544000 +0300
+++ b/src/core/qgsfontutils.cpp	2025-07-30 19:58:18.119961334 +0300
@@ -282,6 +282,7 @@
     else
     {
       bool loaded = false;
+      /*
       if ( QgsApplication::isRunningFromBuildDir() )
       {
         // workaround for bugs with Qt 4.8.5 (other versions?) on Mac 10.9, where fonts
@@ -297,6 +298,7 @@
         QgsDebugMsgLevel( QStringLiteral( "font families in %1: %2" ).arg( fontID ).arg( QFontDatabase().applicationFontFamilies( fontID ).join( "," ) ), 2 );
       }
       else
+      */
       {
         QFile fontResource( ":/testdata/font/" + fontpath );
         if ( fontResource.open( QIODevice::ReadOnly ) )
