--- a/src/core/proj/qgsprojutils.cpp	2023-08-21 19:39:45.711048104 +0300
+++ b/src/core/proj/qgsprojutils.cpp	2023-10-03 23:52:17.122826510 +0300
@@ -18,6 +18,7 @@
 #include "qgis.h"
 #include "qgscoordinatetransform.h"
 #include "qgsexception.h"
+#include "qgsapplication.h"
 #include <QString>
 #include <QSet>
 #include <QRegularExpression>
@@ -406,25 +407,30 @@
 
 QStringList QgsProjUtils::searchPaths()
 {
-  const QString path( proj_info().searchpath );
-  QStringList paths;
-#ifdef Q_OS_WIN
-  paths = path.split( ';' );
-#else
-  paths = path.split( ':' );
-#endif
+//   const QString path( proj_info().searchpath );
+//   QStringList paths;
+// #ifdef Q_OS_WIN
+//   paths = path.split( ';' );
+// #else
+//   paths = path.split( ':' );
+// #endif
 
-  QSet<QString> existing;
-  // thin out duplicates from paths -- see https://github.com/OSGeo/proj.4/pull/1498
+//   QSet<QString> existing;
+//   // thin out duplicates from paths -- see https://github.com/OSGeo/proj.4/pull/1498
   QStringList res;
-  res.reserve( paths.count() );
-  for ( const QString &p : std::as_const( paths ) )
-  {
-    if ( existing.contains( p ) )
-      continue;
-
-    existing.insert( p );
-    res << p;
-  }
+#ifdef Q_OS_MACX
+    res << QgsApplication::prefixPath() + QStringLiteral("/Library/Frameworks/proj.framework/Resources/proj");
+#else
+    res << QgsApplication::prefixPath() + QStringLiteral("/share/proj");
+#endif // Q_OS_MACX
+//   res.reserve( paths.count() );
+//   for ( const QString &p : std::as_const( paths ) )
+//   {
+//     if ( existing.contains( p ) )
+//       continue;
+
+//     existing.insert( p );
+//     res << p;
+//   }
   return res;
 }
