--- a/src/core/processing/qgsprocessingregistry.cpp	2025-05-15 15:26:35.439174440 +0300
+++ b/src/core/processing/qgsprocessingregistry.cpp	2025-05-15 14:47:00.744293603 +0300
@@ -20,11 +20,11 @@
 #include "qgsvectorfilewriter.h"
 #include "qgsprocessingparametertypeimpl.h"
 #include "qgsprocessingparametermeshdataset.h"
-#include "qgsprocessingparametervectortilewriterlayers.h"
+// #include "qgsprocessingparametervectortilewriterlayers.h"
 #include "qgsprocessingparametertininputlayers.h"
 #include "qgsprocessingparameterfieldmap.h"
 #include "qgsprocessingparameteraggregate.h"
-#include "qgsprocessingparameterdxflayers.h"
+// #include "qgsprocessingparameterdxflayers.h"
 #include "qgsprocessingparameteralignrasterlayers.h"
 
 QgsProcessingRegistry::QgsProcessingRegistry( QObject *parent SIP_TRANSFERTHIS )
@@ -70,11 +70,11 @@
   addParameterType( new QgsProcessingParameterTypeProviderConnection() );
   addParameterType( new QgsProcessingParameterTypeDatabaseSchema() );
   addParameterType( new QgsProcessingParameterTypeDatabaseTable() );
-  addParameterType( new QgsProcessingParameterTypeVectorTileWriterLayers() );
+  // addParameterType( new QgsProcessingParameterTypeVectorTileWriterLayers() );
   addParameterType( new QgsProcessingParameterTypeFieldMapping() );
   addParameterType( new QgsProcessingParameterTypeAggregate() );
   addParameterType( new QgsProcessingParameterTypeTinInputLayers() );
-  addParameterType( new QgsProcessingParameterTypeDxfLayers() );
+  // addParameterType( new QgsProcessingParameterTypeDxfLayers() );
   addParameterType( new QgsProcessingParameterTypeMeshDatasetGroups() );
   addParameterType( new QgsProcessingParameterTypeMeshDatasetTime() );
   addParameterType( new QgsProcessingParameterTypePointCloudLayer() );
