--- a/src/core/qgslayerdefinition.cpp	2025-07-15 20:09:37.695544086 +0300
+++ b/src/core/qgslayerdefinition.cpp	2025-07-30 19:58:18.119961334 +0300
@@ -27,7 +27,7 @@
 #include "qgsrasterlayer.h"
 #include "qgsreadwritecontext.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsapplication.h"
 #include "qgsmaplayerfactory.h"
 #include "qgsmeshlayer.h"
@@ -332,7 +332,7 @@
           break;
 
         case Qgis::LayerType::VectorTile:
-          layer = new QgsVectorTileLayer;
+          // layer = new QgsVectorTileLayer;
           break;
 
         case Qgis::LayerType::PointCloud:
@@ -534,5 +534,3 @@
 {
   return mDependentLayerIds.contains( layerId );
 }
-
-
