--- a/src/core/symbology/qgsfillsymbollayer.cpp	2025-07-15 20:09:37.771545709 +0300
+++ b/src/core/symbology/qgsfillsymbollayer.cpp	2025-07-30 19:58:18.121961405 +0300
@@ -18,7 +18,7 @@
 #include "qgslinesymbollayer.h"
 #include "qgssldexportcontext.h"
 #include "qgssymbollayerutils.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgsgeometry.h"
 #include "qgsimagecache.h"
 #include "qgsrendercontext.h"
@@ -515,6 +515,7 @@
   return penBleed + offsetBleed;
 }
 
+/*
 double QgsSimpleFillSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double width = mStrokeWidth;
@@ -567,6 +568,7 @@
 {
   return mBrushStyle;
 }
+*/
 
 QImage QgsSimpleFillSymbolLayer::toTiledPatternImage( ) const
 {
@@ -1844,6 +1846,7 @@
   return mStrokeWidthMapUnitScale;
 }
 
+/*
 double QgsImageFillSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double width = mStrokeWidth;
@@ -1869,6 +1872,7 @@
   }
 #endif //0
 }
+*/
 
 QVariantMap QgsImageFillSymbolLayer::properties() const
 {
@@ -2352,6 +2356,7 @@
   return 0;
 }
 
+/*
 QColor QgsSVGFillSymbolLayer::dxfColor( QgsSymbolRenderContext &context ) const
 {
   Q_UNUSED( context )
@@ -2361,6 +2366,7 @@
   }
   return mStroke->color();
 }
+*/
 
 QSet<QString> QgsSVGFillSymbolLayer::usedAttributes( const QgsRenderContext &context ) const
 {
