--- a/src/core/processing/qgsprocessingutils.h	2025-07-15 20:09:37.655543231 +0300
+++ b/src/core/processing/qgsprocessingutils.h	2025-07-30 19:58:18.117961263 +0300
@@ -35,10 +35,10 @@
 class QgsProcessingFeedback;
 class QgsProcessingFeatureSource;
 class QgsProcessingAlgorithm;
-class QgsVectorTileLayer;
+// class QgsVectorTileLayer;
 class QgsPointCloudLayer;
 class QgsAnnotationLayer;
-class QgsVectorTileLayer;
+// class QgsVectorTileLayer;
 
 #include <QString>
 #include <QVariant>
@@ -185,7 +185,7 @@
      *
      * \since QGIS 3.32
      */
-    static QList<QgsVectorTileLayer *> compatibleVectorTileLayers( QgsProject *project, bool sort = true );
+    // static QList<QgsVectorTileLayer *> compatibleVectorTileLayers( QgsProject *project, bool sort = true );
 
     /**
      * Returns a list of map layers from a \a project which are compatible with the processing
@@ -574,7 +574,7 @@
     static bool canUseLayer( const QgsRasterLayer *layer );
     static bool canUseLayer( const QgsMeshLayer *layer );
     static bool canUseLayer( const QgsPluginLayer *layer );
-    static bool canUseLayer( const QgsVectorTileLayer *layer );
+    // static bool canUseLayer( const QgsVectorTileLayer *layer );
     static bool canUseLayer( const QgsPointCloudLayer *layer );
     static bool canUseLayer( const QgsAnnotationLayer *layer );
     static bool canUseLayer( const QgsVectorLayer *layer,
@@ -763,5 +763,3 @@
 #endif
 
 #endif // QGSPROCESSINGUTILS_H
-
-
