--- a/src/core/symbology/qgsellipsesymbollayer.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/symbology/qgsellipsesymbollayer.cpp	2022-03-21 09:56:36.000000000 +0300
@@ -13,7 +13,7 @@
  *                                                                         *
  ***************************************************************************/
 #include "qgsellipsesymbollayer.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgsexpression.h"
 #include "qgsfeature.h"
 #include "qgsrendercontext.h"
@@ -827,7 +827,7 @@
 
   return symbolBounds;
 }
-
+/*
 bool QgsEllipseSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   //width
@@ -1011,7 +1011,7 @@
 
   return false;
 }
-
+*/
 QgsEllipseSymbolLayer::Shape QgsEllipseSymbolLayer::decodeShape( const QString &name, bool *ok )
 {
   if ( ok )
