--- a/src/core/geometry/qgsgeometry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/geometry/qgsgeometry.cpp	2022-04-07 00:39:42.000000000 +0300
@@ -17,7 +17,7 @@
 #include <cstdarg>
 #include <cstdio>
 #include <cmath>
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 #include <QCache>
 
 #include "qgis.h"
@@ -56,6 +56,8 @@
 #include "qgscurve.h"
 #include "qgsreadwritelocker.h"
 
+#include "qgsjsonutils.h"
+
 struct QgsGeometryPrivate
 {
   QgsGeometryPrivate(): ref( 1 ) {}
@@ -1426,14 +1428,14 @@
 
 QString QgsGeometry::asJson( int precision ) const
 {
-  return QString::fromStdString( asJsonObject( precision ).dump() );
+  return QgsJsonUtils::dump( asJsonObject( precision ) );
 }
 
 json QgsGeometry::asJsonObject( int precision ) const
 {
   if ( !d->geometry )
   {
-    return nullptr;
+    return CPLJSONObject();
   }
   return d->geometry->asJsonObject( precision );
 
