--- a/src/core/providers/qgsproviderregistry.cpp	2025-07-15 20:09:37.673543616 +0300
+++ b/src/core/providers/qgsproviderregistry.cpp	2025-07-30 19:58:18.118961299 +0300
@@ -24,7 +24,7 @@
 #include "qgslogger.h"
 #include "qgsmessagelog.h"
 #include "qgsprovidermetadata.h"
-#include "qgsvectortileprovidermetadata.h"
+// #include "qgsvectortileprovidermetadata.h"
 #include "qgsproject.h"
 #include "qgsprovidersublayerdetails.h"
 #include "providers/memory/qgsmemoryprovider.h"
@@ -33,10 +33,12 @@
 #include "providers/ogr/qgsogrprovider.h"
 #include "providers/meshmemory/qgsmeshmemorydataprovider.h"
 
+/*
 #include "qgsmbtilesvectortiledataprovider.h"
 #include "qgsarcgisvectortileservicedataprovider.h"
 #include "qgsxyzvectortiledataprovider.h"
 #include "qgsvtpkvectortiledataprovider.h"
+*/
 
 #ifdef HAVE_EPT
 #include "providers/ept/qgseptprovider.h"
@@ -192,6 +194,7 @@
     const QgsScopedRuntimeProfile profile( QObject::tr( "Create OGR provider" ) );
     mProviders[ QgsOgrProvider::providerKey() ] = new QgsOgrProviderMetadata();
   }
+  /*
   {
     const QgsScopedRuntimeProfile profile( QObject::tr( "Create vector tile providers" ) );
     QgsProviderMetadata *vt = new QgsVectorTileProviderMetadata();
@@ -205,6 +208,7 @@
     vt = new QgsMbTilesVectorTileDataProviderMetadata();
     mProviders[ vt->key() ] = vt;
   }
+  */
 #ifdef HAVE_EPT
   {
     const QgsScopedRuntimeProfile profile( QObject::tr( "Create EPT point cloud provider" ) );
