--- a/src/core/symbology/qgslinesymbollayer.cpp	2025-05-15 15:26:35.502699336 +0300
+++ b/src/core/symbology/qgslinesymbollayer.cpp	2025-05-15 14:47:00.826428895 +0300
@@ -15,7 +15,7 @@
 
 #include "qgslinesymbollayer.h"
 #include "qgscurvepolygon.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgssymbollayerutils.h"
 #include "qgsrendercontext.h"
 #include "qgslogger.h"
@@ -1088,7 +1088,7 @@
            context.convertToPainterUnits( std::fabs( mOffset ), mOffsetUnit, mOffsetMapUnitScale );
   }
 }
-
+/*
 QVector<qreal> QgsSimpleLineSymbolLayer::dxfCustomDashPattern( Qgis::RenderUnit &unit ) const
 {
   unit = mCustomDashPatternUnit;
@@ -1126,7 +1126,7 @@
   }
   return mColor;
 }
-
+*/
 bool QgsSimpleLineSymbolLayer::canCauseArtifactsBetweenAdjacentTiles() const
 {
   return mPenStyle != Qt::SolidLine || mUseCustomDashPattern;
@@ -1151,7 +1151,7 @@
 {
   mPatternCartographicTweakOnSharpCorners = enabled;
 }
-
+/*
 double QgsSimpleLineSymbolLayer::dxfOffset( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double offset = mOffset;
@@ -1169,7 +1169,7 @@
   }
   return -offset; //direction seems to be inverse to symbology offset
 }
-
+*/
 /////////
 
 ///@cond PRIVATE
