--- a/python/core/auto_additions/qgsdxfexport.py	2025-07-15 20:09:37.183533150 +0300
+++ b/python/core/auto_additions/qgsdxfexport.py	2025-07-30 19:58:18.104501779 +0300
@@ -1,4 +1,5 @@
 # The following has been generated automatically from src/core/dxf/qgsdxfexport.h
+"""
 # monkey patching scoped based enum
 QgsDxfExport.ExportResult.Success.__doc__ = "Successful export"
 QgsDxfExport.ExportResult.InvalidDeviceError.__doc__ = "Invalid device error"
@@ -24,3 +25,4 @@
 QgsDxfExport.HAlign.Undefined.__doc__ = "Undefined"
 QgsDxfExport.HAlign.__doc__ = 'Horizontal alignments.\n\n' + '* ``HLeft``: ' + QgsDxfExport.HAlign.HLeft.__doc__ + '\n' + '* ``HCenter``: ' + QgsDxfExport.HAlign.HCenter.__doc__ + '\n' + '* ``HRight``: ' + QgsDxfExport.HAlign.HRight.__doc__ + '\n' + '* ``HAligned``: ' + QgsDxfExport.HAlign.HAligned.__doc__ + '\n' + '* ``HMiddle``: ' + QgsDxfExport.HAlign.HMiddle.__doc__ + '\n' + '* ``HFit``: ' + QgsDxfExport.HAlign.HFit.__doc__ + '\n' + '* ``Undefined``: ' + QgsDxfExport.HAlign.Undefined.__doc__
 # --
+"""
