--- a/src/core/symbology/qgsellipsesymbollayer.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/symbology/qgsellipsesymbollayer.h	2022-03-07 23:14:46.000000000 +0300
@@ -74,7 +74,7 @@
     void toSld( QDomDocument &doc, QDomElement &element, const QVariantMap &props ) const override;
     void writeSldMarker( QDomDocument &doc, QDomElement &element, const QVariantMap &props ) const override;
 
-    bool writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift = QPointF( 0.0, 0.0 ) ) const override;
+    // bool writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift = QPointF( 0.0, 0.0 ) ) const override;
 
     /**
      * Sets the rendered ellipse marker shape using a symbol \a name.
