--- a/src/core/processing/qgsprocessingutils.cpp	2025-07-15 20:09:37.655543231 +0300
+++ b/src/core/processing/qgsprocessingutils.cpp	2025-07-30 19:58:18.117961263 +0300
@@ -33,7 +33,7 @@
 #include "qgspluginlayer.h"
 #include "qgsreferencedgeometry.h"
 #include "qgsrasterfilewriter.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgspointcloudlayer.h"
 #include "qgsannotationlayer.h"
 #include <QRegularExpression>
@@ -101,10 +101,12 @@
   return res;
 }
 
+/*
 QList<QgsVectorTileLayer *> QgsProcessingUtils::compatibleVectorTileLayers( QgsProject *project, bool sort )
 {
   return compatibleMapLayers< QgsVectorTileLayer >( project, sort );
 }
+*/
 
 template<typename T> QList<T *> QgsProcessingUtils::compatibleMapLayers( QgsProject *project, bool sort )
 {
@@ -157,9 +159,11 @@
     layers << al;
   layers << project->mainAnnotationLayer();
 
+  /*
   const auto vectorTileLayers = compatibleMapLayers< QgsVectorTileLayer >( project, false );
   for ( QgsVectorTileLayer *vtl : vectorTileLayers )
     layers << vtl;
+  */
 
   const auto pluginLayers = compatibleMapLayers< QgsPluginLayer >( project, false );
   for ( QgsPluginLayer *pl : pluginLayers )
@@ -214,8 +218,10 @@
         return true;
       case Qgis::LayerType::Mesh:
         return !canUseLayer( qobject_cast< QgsMeshLayer * >( layer ) );
+      /*
       case Qgis::LayerType::VectorTile:
         return !canUseLayer( qobject_cast< QgsVectorTileLayer * >( layer ) );
+      */
       case Qgis::LayerType::PointCloud:
         return !canUseLayer( qobject_cast< QgsPointCloudLayer * >( layer ) );
       case Qgis::LayerType::Annotation:
@@ -391,6 +397,7 @@
       return pointCloudLayer.release();
     }
   }
+  /*
   if ( typeHint == LayerHint::UnknownType || typeHint == LayerHint::VectorTile )
   {
     QgsDataSourceUri dsUri;
@@ -405,6 +412,7 @@
       return tileLayer.release();
     }
   }
+  */
   return nullptr;
 }
 
@@ -594,10 +602,12 @@
   return layer && layer->isValid();
 }
 
+/*
 bool QgsProcessingUtils::canUseLayer( const QgsVectorTileLayer *layer )
 {
   return layer && layer->isValid();
 }
+*/
 
 bool QgsProcessingUtils::canUseLayer( const QgsRasterLayer *layer )
 {
