--- a/src/core/symbology/qgslinesymbollayer.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/symbology/qgslinesymbollayer.cpp	2022-03-07 23:14:46.000000000 +0300
@@ -16,7 +16,7 @@
 #include "qgslinesymbollayer.h"
 #include "qgscurve.h"
 #include "qgscurvepolygon.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgssymbollayerutils.h"
 #include "qgsexpression.h"
 #include "qgsrendercontext.h"
@@ -1069,7 +1069,7 @@
            context.convertToPainterUnits( std::fabs( mOffset ), mOffsetUnit, mOffsetMapUnitScale );
   }
 }
-
+/*
 QVector<qreal> QgsSimpleLineSymbolLayer::dxfCustomDashPattern( QgsUnitTypes::RenderUnit &unit ) const
 {
   unit = mCustomDashPatternUnit;
@@ -1107,7 +1107,7 @@
   }
   return mColor;
 }
-
+*/
 bool QgsSimpleLineSymbolLayer::canCauseArtifactsBetweenAdjacentTiles() const
 {
   return mPenStyle != Qt::SolidLine || mUseCustomDashPattern;
@@ -1132,7 +1132,7 @@
 {
   mPatternCartographicTweakOnSharpCorners = enabled;
 }
-
+/*
 double QgsSimpleLineSymbolLayer::dxfOffset( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double offset = mOffset;
@@ -1150,7 +1150,7 @@
   }
   return -offset; //direction seems to be inverse to symbology offset
 }
-
+*/
 /////////
 
 ///@cond PRIVATE
