--- a/src/core/qgslayerdefinition.cpp	2025-05-15 15:26:35.469724501 +0300
+++ b/src/core/qgslayerdefinition.cpp	2025-05-15 14:47:00.781294894 +0300
@@ -27,7 +27,7 @@
 #include "qgsrasterlayer.h"
 #include "qgsreadwritecontext.h"
 #include "qgsvectorlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgstiledscenelayer.h"
 #include "qgsapplication.h"
 #include "qgsmaplayerfactory.h"
@@ -345,9 +345,9 @@
           layer = new QgsMeshLayer();
           break;
 
-        case Qgis::LayerType::VectorTile:
-          layer = new QgsVectorTileLayer;
-          break;
+        // case Qgis::LayerType::VectorTile:
+        //   layer = new QgsVectorTileLayer;
+        //   break;
 
         case Qgis::LayerType::PointCloud:
           layer = new QgsPointCloudLayer();
@@ -552,5 +552,3 @@
 {
   return mDependentLayerIds.contains( layerId );
 }
-
-
