--- a/src/core/qgsmaplayerfactory.cpp	2025-05-15 15:26:35.473236397 +0300
+++ b/src/core/qgsmaplayerfactory.cpp	2025-05-15 14:47:00.784294999 +0300
@@ -20,7 +20,7 @@
 #include "qgsrasterlayer.h"
 #include "qgsmeshlayer.h"
 #include "qgspointcloudlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsannotationlayer.h"
 #include "qgsgrouplayer.h"
 #include "qgstiledscenelayer.h"
@@ -34,8 +34,8 @@
     return Qgis::LayerType::Raster;
   else if ( string.compare( QLatin1String( "mesh" ), Qt::CaseInsensitive ) == 0 )
     return Qgis::LayerType::Mesh;
-  else if ( string.compare( QLatin1String( "vector-tile" ), Qt::CaseInsensitive ) == 0 )
-    return Qgis::LayerType::VectorTile;
+ else if ( string.compare( QLatin1String( "vector-tile" ), Qt::CaseInsensitive ) == 0 )
+   return Qgis::LayerType::VectorTile;
   else if ( string.compare( QLatin1String( "point-cloud" ), Qt::CaseInsensitive ) == 0 )
     return Qgis::LayerType::PointCloud;
   else if ( string.compare( QLatin1String( "plugin" ), Qt::CaseInsensitive ) == 0 )
@@ -106,11 +106,13 @@
       return new QgsMeshLayer( uri, name, provider, meshOptions );
     }
 
+    /*
     case Qgis::LayerType::VectorTile:
     {
       const QgsVectorTileLayer::LayerOptions vectorTileOptions( options.transformContext );
       return new QgsVectorTileLayer( uri, name, vectorTileOptions );
     }
+    */
 
     case Qgis::LayerType::Annotation:
     {
