--- a/src/core/layout/qgslayoutitemmap.cpp	2025-07-15 20:09:37.615542377 +0300
+++ b/src/core/layout/qgslayoutitemmap.cpp	2025-07-30 19:58:18.116961227 +0300
@@ -37,7 +37,7 @@
 #include "qgscoordinatereferencesystemregistry.h"
 #include "qgsprojoperation.h"
 #include "qgslabelingresults.h"
-#include "qgsvectortileutils.h"
+// #include "qgsvectortileutils.h"
 #include "qgsunittypes.h"
 
 #include <QPainter>
@@ -1768,8 +1768,10 @@
   const double mapScale = scale();
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_scale" ), mapScale, true ) );
 
+  /*
   scope->setVariable( QStringLiteral( "zoom_level" ), QgsVectorTileUtils::scaleToZoomLevel( mapScale, 0, 99999 ), true );
   scope->setVariable( QStringLiteral( "vector_tile_zoom" ), QgsVectorTileUtils::scaleToZoom( mapScale ), true );
+  */
 
   QgsRectangle currentExtent( extent() );
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_extent" ), QVariant::fromValue( QgsGeometry::fromRect( currentExtent ) ), true ) );
