--- a/src/core/mesh/qgstopologicalmesh.h	2025-07-15 20:09:37.630542697 +0300
+++ b/src/core/mesh/qgstopologicalmesh.h	2025-07-30 03:05:27.888802509 +0300
@@ -26,8 +26,6 @@
 template CORE_EXPORT QVector<QVector<int>> SIP_SKIP;
 #endif
 
-SIP_NO_FILE
-
 class QgsMeshEditingError;
 class QgsMeshVertexCirculator;
 
@@ -36,11 +34,11 @@
  *
  * \brief Class that wraps a QgsMesh to ensure the consistency of the mesh during editing and help to access to elements from other elements
  *
- *  A topological face need to:
+ *  A topological face must to:
  *
  * - be convex
- * - counter clock wise
- * - not share an unique vertex with another face
+ * - be oriented counter-clockwise
+ * - not share a unique vertex with another face
  *
  * \since QGIS 3.22
  */
@@ -55,22 +53,28 @@
      *
      * \brief Class that contains independent faces an topological information about this faces
      *
-     * This class supports unique shared vertex between faces
+     * This class supports unique shared vertices between faces
      *
      * \since QGIS 3.22
      */
     class CORE_EXPORT TopologicalFaces
     {
       public:
-
-        //! Returns faces
-        QVector<QgsMeshFace>  meshFaces() const {return mFaces;}
+        /**
+         * Returns faces.
+         * \note Not available in Python bindings.
+         */
+        SIP_SKIP QVector<QgsMeshFace>  meshFaces() const {return mFaces;}
 
         //! Clears all data contained in the instance.
         void clear();
 
-        //! Returns the face neighborhood of the faces, indexing is local
-        QVector<FaceNeighbors> facesNeighborhood() const;
+        /**
+         * Returns the face neighborhood of the faces, indexing is local.
+         *
+         * \note Not available in Python bindings.
+         */
+        SIP_SKIP QVector<FaceNeighbors> facesNeighborhood() const;
 
         //! Returns a face linked to the vertices with index \a vertexIndex
         int vertexToFace( int vertexIndex ) const;
@@ -97,17 +101,29 @@
     {
       public:
 
-        //! Returns the face that are added with this changes
-        QVector<QgsMeshFace> addedFaces() const;
-
-        //! Returns the faces that are removed with this changes
-        QVector<QgsMeshFace> removedFaces() const;
+        /**
+         * Returns the face that are added with this changes.
+         *
+         * \note Not available in Python bindings.
+         */
+        SIP_SKIP QVector<QgsMeshFace> addedFaces() const;
+
+        /**
+         * Returns the faces that are removed with this changes.
+         *
+         * \note Not available in Python bindings.
+         */
+        SIP_SKIP QVector<QgsMeshFace> removedFaces() const;
 
         //! Returns the indexes of the faces that are removed with this changes
         QList<int> removedFaceIndexes() const;
 
         //! Returns the added vertices with this changes
+#ifndef SIP_RUN
         QVector<QgsMeshVertex> addedVertices() const;
+#else
+        QVector<QgsPoint> addedVertices() const;
+#endif
 
         //! Returns the indexes of vertices to remove
         QList<int> verticesToRemoveIndexes() const;
@@ -169,8 +185,12 @@
      */
     static QgsTopologicalMesh createTopologicalMesh( QgsMesh *mesh, int maxVerticesPerFace, QgsMeshEditingError &error );
 
-    //! Creates new topological faces that are not yet included in the mesh
-    static TopologicalFaces  createNewTopologicalFaces( const QVector<QgsMeshFace> &faces, bool uniqueSharedVertexAllowed, QgsMeshEditingError &error );
+    /**
+     * Creates new topological faces that are not yet included in the mesh.
+     *
+     * \note Not available in Python bindings
+     */
+    SIP_SKIP static TopologicalFaces createNewTopologicalFaces( const QVector<QgsMeshFace> &faces, bool uniqueSharedVertexAllowed, QgsMeshEditingError &error );
 
     //----------- access element methods
 
@@ -195,8 +215,12 @@
     //! Returns a list of vertices are not linked to any faces
     QList<int> freeVerticesIndexes() const;
 
-    //! Returns a vertex circulator linked to this mesh around the vertex with index \a vertexIndex
-    QgsMeshVertexCirculator vertexCirculator( int vertexIndex ) const;
+    /**
+     * Returns a vertex circulator linked to this mesh around the vertex with index \a vertexIndex.
+     *
+     * \note Not available in Python bindings
+     */
+    SIP_SKIP QgsMeshVertexCirculator vertexCirculator( int vertexIndex ) const;
 
     //----------- editing methods
 
@@ -312,7 +336,7 @@
      *
      * \since QGIS 3.30
      */
-    static QgsMeshEditingError checkTopologyOfVerticesAsFace( const QVector<QgsMeshVertex> &vertices, bool &clockwise );
+    SIP_SKIP static QgsMeshEditingError checkTopologyOfVerticesAsFace( const QVector<QgsMeshVertex> &vertices, bool &clockwise );
 
     /**
      * Reindexes faces and vertices, after this operation, the topological
@@ -377,11 +401,15 @@
 
 };
 
+#ifndef SIP_RUN
+
 /**
  * \ingroup core
  *
  * \brief  Convenient class that turn around a vertex and provide information about faces and vertices
  *
+ * \note Not available in Python bindings
+ *
  * \since QGIS 3.22
  */
 class CORE_EXPORT QgsMeshVertexCirculator
@@ -449,5 +477,6 @@
 
     int positionInCurrentFace() const;
 };
+#endif
 
 #endif // QGSTOPOLOGICALMESH_H
