--- a/src/core/symbology/qgsmarkersymbollayer.cpp	2025-07-15 20:09:37.776545815 +0300
+++ b/src/core/symbology/qgsmarkersymbollayer.cpp	2025-07-30 19:58:18.123961477 +0300
@@ -16,8 +16,10 @@
 #include "qgsmarkersymbollayer.h"
 #include "qgssymbollayerutils.h"
 
+/*
 #include "qgsdxfexport.h"
 #include "qgsdxfpaintdevice.h"
+*/
 #include "qgsfontutils.h"
 #include "qgsimagecache.h"
 #include "qgsimageoperation.h"
@@ -1571,6 +1573,7 @@
   }
 }
 
+/*
 bool QgsSimpleMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   //data defined size?
@@ -1759,6 +1762,7 @@
 
   return true;
 }
+*/
 
 
 void QgsSimpleMarkerSymbolLayer::setOutputUnit( Qgis::RenderUnit unit )
@@ -2810,6 +2814,7 @@
   return m;
 }
 
+/*
 bool QgsSvgMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   //size
@@ -2930,6 +2935,7 @@
   p.end();
   return true;
 }
+*/
 
 QRectF QgsSvgMarkerSymbolLayer::bounds( QPointF point, QgsSymbolRenderContext &context )
 {
@@ -4095,4 +4101,3 @@
   bool cached = false;
   return QgsApplication::imageCache()->pathAsImage( path, size, preserveAspectRatio, opacity, cached, context.flags() & Qgis::RenderContextFlag::RenderBlocking, 96, movieFrame );
 }
-
