--- a/src/core/symbology/qgsfillsymbollayer.cpp	2025-05-15 15:26:35.495712693 +0300
+++ b/src/core/symbology/qgsfillsymbollayer.cpp	2025-05-15 14:47:00.821296289 +0300
@@ -18,7 +18,7 @@
 #include "qgslinesymbollayer.h"
 #include "qgssldexportcontext.h"
 #include "qgssymbollayerutils.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgsgeometry.h"
 #include "qgsimagecache.h"
 #include "qgsrendercontext.h"
@@ -517,6 +517,7 @@
   return penBleed + offsetBleed;
 }
 
+/*
 double QgsSimpleFillSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double width = mStrokeWidth;
@@ -569,6 +570,7 @@
 {
   return mBrushStyle;
 }
+*/
 
 QImage QgsSimpleFillSymbolLayer::toTiledPatternImage( ) const
 {
@@ -1851,6 +1853,7 @@
   return mStrokeWidthMapUnitScale;
 }
 
+/*
 double QgsImageFillSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double width = mStrokeWidth;
@@ -1876,7 +1879,7 @@
   }
 #endif //0
 }
-
+*/
 QVariantMap QgsImageFillSymbolLayer::properties() const
 {
   QVariantMap map;
@@ -2361,6 +2364,7 @@
   return 0;
 }
 
+/*
 QColor QgsSVGFillSymbolLayer::dxfColor( QgsSymbolRenderContext &context ) const
 {
   Q_UNUSED( context )
@@ -2370,6 +2374,7 @@
   }
   return mStroke->color();
 }
+*/
 
 QSet<QString> QgsSVGFillSymbolLayer::usedAttributes( const QgsRenderContext &context ) const
 {
