--- a/src/core/auth/qgsauthcertutils.cpp	2025-07-15 20:09:37.543540839 +0300
+++ b/src/core/auth/qgsauthcertutils.cpp	2025-07-30 19:58:18.114961156 +0300
@@ -27,10 +27,12 @@
 #include "qgsauthmanager.h"
 #include "qgslogger.h"
 
+/*
 #ifdef Q_OS_MAC
 #include <string.h>
 #include "libtasn1.h"
 #endif
+*/
 
 
 QString QgsAuthCertUtils::getSslProtocolName( QSsl::SslProtocol protocol )
@@ -300,7 +302,7 @@
   return keyPemTxt.contains( pkcs8Header ) && keyPemTxt.contains( pkcs8Footer );
 }
 
-#ifdef Q_OS_MAC
+#ifdef UNUSED_TMP // Q_OS_MAC
 QByteArray QgsAuthCertUtils::pkcs8PrivateKey( QByteArray &pkcs8Der )
 {
   QByteArray pkcs1;
@@ -493,7 +495,7 @@
   }
 
   QString keyPem;
-#ifdef Q_OS_MAC
+#ifdef UNUSED_TMP // Q_OS_MAC
   if ( keyalg == QSsl::Rsa && QgsAuthCertUtils::pemIsPkcs8( bundle.privateKey().toPEM() ) )
   {
     QgsDebugMsgLevel( QStringLiteral( "Private key is PKCS#8: attempting conversion to PKCS#1..." ), 4 );
