--- a/src/core/labeling/qgspallabeling.cpp	2025-05-15 15:26:35.429514493 +0300
+++ b/src/core/labeling/qgspallabeling.cpp	2025-05-15 14:47:00.695291895 +0300
@@ -56,8 +56,8 @@
 #include "qgsgeometrycollection.h"
 #include "callouts/qgscallout.h"
 #include "callouts/qgscalloutsregistry.h"
-#include "qgsvectortilelayer.h"
-#include "qgsvectortilebasiclabeling.h"
+// #include "qgsvectortilelayer.h"
+// #include "qgsvectortilebasiclabeling.h"
 #include "qgsfontmanager.h"
 #include "qgsvariantutils.h"
 #include "qgsmeshlayer.h"
@@ -4081,12 +4081,14 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       const QgsVectorTileLayer *vl = qobject_cast< const QgsVectorTileLayer * >( layer );
       if ( !vl->labeling() )
         return false;
 
       if ( const QgsVectorTileBasicLabeling *labeling = dynamic_cast< const QgsVectorTileBasicLabeling *>( vl->labeling() ) )
         return !labeling->styles().empty();
+      */
 
       return false;
     }
