--- a/src/core/processing/qgsprocessingutils.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/core/processing/qgsprocessingutils.h	2022-04-06 18:40:09.000000000 +0300
@@ -37,7 +37,7 @@
 class QgsProcessingFeedback;
 class QgsProcessingFeatureSource;
 class QgsProcessingAlgorithm;
-class QgsVectorTileLayer;
+// class QgsVectorTileLayer;
 class QgsPointCloudLayer;
 class QgsAnnotationLayer;
 
@@ -499,7 +499,7 @@
     static bool canUseLayer( const QgsRasterLayer *layer );
     static bool canUseLayer( const QgsMeshLayer *layer );
     static bool canUseLayer( const QgsPluginLayer *layer );
-    static bool canUseLayer( const QgsVectorTileLayer *layer );
+    // static bool canUseLayer( const QgsVectorTileLayer *layer );
     static bool canUseLayer( const QgsPointCloudLayer *layer );
     static bool canUseLayer( const QgsAnnotationLayer *layer );
     static bool canUseLayer( const QgsVectorLayer *layer,
