--- a/src/core/auth/qgsauthmanager.h	2025-07-15 20:09:37.545540882 +0300
+++ b/src/core/auth/qgsauthmanager.h	2025-07-30 19:58:18.114961156 +0300
@@ -131,7 +131,7 @@
      * \note Do not emit verification signals when only comparing
      * \param compare Password to compare against
      */
-    bool verifyMasterPassword( const QString &compare = QString() );
+    virtual bool verifyMasterPassword( const QString &compare = QString() );
 
     //! Whether master password has be input and verified, i.e. authentication database is accessible
     bool masterPasswordIsSet() const;
@@ -678,7 +678,7 @@
      * Password helper enabled getter
      * \note Available in Python bindings since QGIS 3.8.0
      */
-    bool passwordHelperEnabled() const;
+    virtual bool passwordHelperEnabled() const;
 
     /**
      * Password helper enabled setter
@@ -782,7 +782,17 @@
      * Enforce singleton pattern
      * \note To set up the manager instance and initialize everything use QgsAuthManager::instance()->init()
      */
-    static QgsAuthManager *instance() SIP_SKIP;
+    template<typename T = QgsAuthManager>
+    static QgsAuthManager *instance() SIP_SKIP
+    {
+        static QMutex sMutex;
+        QMutexLocker locker(&sMutex);
+        if (!sInstance)
+        {
+            sInstance = new T;
+        }
+        return sInstance;
+    }
 
 
 #ifdef Q_OS_WIN
@@ -804,30 +814,36 @@
     //! Print a debug message in QGIS
     void passwordHelperLog( const QString &msg ) const;
 
+  protected:
     //! Read Master password from the wallet
     QString passwordHelperRead();
 
     //! Store Master password in the wallet
     bool passwordHelperWrite( const QString &password );
 
+  private:
     //! Error message setter
     void passwordHelperSetErrorMessage( const QString &errorMessage ) { mPasswordHelperErrorMessage = errorMessage; }
 
     //! Clear error code and message
     void passwordHelperClearErrors();
 
+  protected:
     /**
      * Process the error: show it and/or disable the password helper system in case of
      * access denied or no backend, reset error flags at the end
      */
-    void passwordHelperProcessError();
+    virtual void passwordHelperProcessError();
 
+  private:
     bool createConfigTables();
 
     bool createCertTables();
 
-    bool masterPasswordInput();
+  protected:
+    virtual bool masterPasswordInput();
 
+  private:
     bool masterPasswordRowsInDb( int *rows ) const;
 
     bool masterPasswordCheckAgainstDb( const QString &compare = QString() ) const;
@@ -892,7 +908,10 @@
     QHash<QString, QString> mConfigAuthMethods;
     QHash<QString, QgsAuthMethod *> mAuthMethods;
 
+  protected:
     QString mMasterPass;
+
+  private:
     int mPassTries = 0;
     bool mAuthDisabled = false;
     QString mAuthDisabledMessage;
@@ -929,9 +948,11 @@
     //! Store last error message
     QString mPasswordHelperErrorMessage;
 
+  protected:
     //! Store last error code (enum)
     QKeychain::Error mPasswordHelperErrorCode = QKeychain::NoError;
 
+  private:
     //! Enable logging
     bool mPasswordHelperLoggingEnabled = false;
 
