--- a/src/core/layout/qgslayoutitemmap.cpp	2025-05-15 15:26:35.433666840 +0300
+++ b/src/core/layout/qgslayoutitemmap.cpp	2025-05-15 14:47:00.706292278 +0300
@@ -38,7 +38,7 @@
 #include "qgscoordinatereferencesystemregistry.h"
 #include "qgsprojoperation.h"
 #include "qgslabelingresults.h"
-#include "qgsvectortileutils.h"
+// #include "qgsvectortileutils.h"
 #include "qgsunittypes.h"
 
 #include <QApplication>
@@ -1899,8 +1899,10 @@
   const double mapScale = scale();
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_scale" ), mapScale, true ) );
 
+  /*
   scope->setVariable( QStringLiteral( "zoom_level" ), !qgsDoubleNear( mapScale, 0 ) ? QgsVectorTileUtils::scaleToZoomLevel( mapScale, 0, 99999 ) : 0, true );
   scope->setVariable( QStringLiteral( "vector_tile_zoom" ), !qgsDoubleNear( mapScale, 0 ) ? QgsVectorTileUtils::scaleToZoom( mapScale ) : 0, true );
+  */
 
   QgsRectangle currentExtent( extent() );
   scope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "map_extent" ), QVariant::fromValue( QgsGeometry::fromRect( currentExtent ) ), true ) );
