--- a/src/core/symbology/qgssymbollayer.cpp	2025-07-15 20:09:37.782545943 +0300
+++ b/src/core/symbology/qgssymbollayer.cpp	2025-07-30 19:58:18.123961477 +0300
@@ -15,7 +15,7 @@
 
 #include "qgssymbollayer.h"
 #include "qgsrendercontext.h"
-#include "qgsdxfexport.h"
+// #include "qgsdxfexport.h"
 #include "qgsgeometrysimplifier.h"
 #include "qgspainteffect.h"
 #include "qgspainteffectregistry.h"
@@ -150,6 +150,7 @@
   return false;
 }
 
+/*
 bool QgsSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFactor, const QString &layerName, QgsSymbolRenderContext &context, QPointF shift ) const
 {
   Q_UNUSED( e )
@@ -214,6 +215,7 @@
 {
   return Qt::NoBrush;
 }
+*/
 
 QgsPaintEffect *QgsSymbolLayer::paintEffect() const
 {
@@ -808,11 +810,13 @@
   return context.convertToPainterUnits( mWidth, mWidthUnit, mWidthMapUnitScale );
 }
 
+/*
 double QgsLineSymbolLayer::dxfWidth( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   Q_UNUSED( context )
   return width() * QgsDxfExport::mapUnitScaleFactor( e.symbologyScale(), widthUnit(), e.mapUnits(), context.renderContext().mapToPixel().mapUnitsPerPixel() );
 }
+*/
 
 
 void QgsFillSymbolLayer::drawPreviewIcon( QgsSymbolRenderContext &context, QSize size )
@@ -908,6 +912,7 @@
   return {};
 }
 
+/*
 double QgsMarkerSymbolLayer::dxfSize( const QgsDxfExport &e, QgsSymbolRenderContext &context ) const
 {
   double size = mSize;
@@ -941,6 +946,7 @@
   }
   return angle;
 }
+*/
 
 void QgsSymbolLayer::prepareMasks( const QgsSymbolRenderContext &context )
 {
