--- a/src/core/browser/qgsfilebaseddataitemprovider.cpp	2025-05-15 15:26:35.420476868 +0300
+++ b/src/core/browser/qgsfilebaseddataitemprovider.cpp	2025-05-15 14:47:00.661290709 +0300
@@ -158,7 +158,8 @@
       return Qgis::BrowserLayerType::Mesh;
 
     case Qgis::LayerType::VectorTile:
-      return Qgis::BrowserLayerType::VectorTile;
+      // return Qgis::BrowserLayerType::VectorTile;
+      break;
 
     case Qgis::LayerType::PointCloud:
       return Qgis::BrowserLayerType::PointCloud;
