--- a/src/core/qgsmaplayerfactory.cpp	2025-07-15 20:09:37.698544150 +0300
+++ b/src/core/qgsmaplayerfactory.cpp	2025-07-30 19:58:18.119961334 +0300
@@ -20,9 +20,10 @@
 #include "qgsrasterlayer.h"
 #include "qgsmeshlayer.h"
 #include "qgspointcloudlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsannotationlayer.h"
 #include "qgsgrouplayer.h"
+#include <cstddef>
 
 Qgis::LayerType QgsMapLayerFactory::typeFromString( const QString &string, bool &ok )
 {
@@ -103,8 +104,11 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       const QgsVectorTileLayer::LayerOptions vectorTileOptions( options.transformContext );
       return new QgsVectorTileLayer( uri, name, vectorTileOptions );
+      */
+      break;
     }
 
     case Qgis::LayerType::Annotation:
