--- a/src/gui/editorwidgets/core/qgseditorwidgetregistry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/editorwidgets/core/qgseditorwidgetregistry.cpp	2022-04-06 18:40:09.000000000 +0300
@@ -30,7 +30,7 @@
 #include "qgscolorwidgetfactory.h"
 #include "qgsdatetimeeditfactory.h"
 #include "qgsenumerationwidgetfactory.h"
-#include "qgsexternalresourcewidgetfactory.h"
+// #include "qgsexternalresourcewidgetfactory.h"
 #include "qgshiddenwidgetfactory.h"
 #include "qgskeyvaluewidgetfactory.h"
 #include "qgsjsoneditwidgetfactory.h"
@@ -63,7 +63,7 @@
   registerWidget( QStringLiteral( "Color" ), new QgsColorWidgetFactory( tr( "Color" ) ) );
   registerWidget( QStringLiteral( "RelationReference" ), new QgsRelationReferenceFactory( tr( "Relation Reference" ), mapCanvas, messageBar ) );
   registerWidget( QStringLiteral( "DateTime" ), new QgsDateTimeEditFactory( tr( "Date/Time" ) ) );
-  registerWidget( QStringLiteral( "ExternalResource" ), new QgsExternalResourceWidgetFactory( tr( "Attachment" ), messageBar ) );
+//   registerWidget( QStringLiteral( "ExternalResource" ), new QgsExternalResourceWidgetFactory( tr( "Attachment" ), messageBar ) );
   registerWidget( QStringLiteral( "KeyValue" ), new QgsKeyValueWidgetFactory( tr( "Key/Value" ) ) );
   registerWidget( QStringLiteral( "List" ), new QgsListWidgetFactory( tr( "List" ) ) );
   registerWidget( QStringLiteral( "Binary" ), new QgsBinaryWidgetFactory( tr( "Binary (BLOB)" ), messageBar ) );
