--- a/src/gui/qgsmanageconnectionsdialog.h	2025-07-15 20:09:37.917548827 +0300
+++ b/src/gui/qgsmanageconnectionsdialog.h	2025-07-30 19:58:18.125961548 +0300
@@ -75,8 +75,10 @@
     QDomDocument saveOracleConnections( const QStringList &connections );
     QDomDocument saveHanaConnections( const QStringList &connections );
     QDomDocument saveXyzTilesConnections( const QStringList &connections );
+    /*
     QDomDocument saveArcgisConnections( const QStringList &connections );
     QDomDocument saveVectorTileConnections( const QStringList &connections );
+    */
 
     void loadOWSConnections( const QDomDocument &doc, const QStringList &items, const QString &service );
     void loadWfsConnections( const QDomDocument &doc, const QStringList &items );
@@ -85,8 +87,10 @@
     void loadOracleConnections( const QDomDocument &doc, const QStringList &items );
     void loadHanaConnections( const QDomDocument &doc, const QStringList &items );
     void loadXyzTilesConnections( const QDomDocument &doc, const QStringList &items );
+    /*
     void loadArcgisConnections( const QDomDocument &doc, const QStringList &items, const QString &service );
     void loadVectorTileConnections( const QDomDocument &doc, const QStringList &items );
+    */
 
     QString mFileName;
     Mode mDialogMode;
@@ -96,4 +100,3 @@
 // clazy:excludeall=qstring-allocations
 
 #endif // QGSMANAGECONNECTIONSDIALOG_H
-
