--- a/python/gui/auto_generated/qgisinterface.sip.in	2022-02-18 15:06:10.000000000 +0300
+++ b/python/gui/auto_generated/qgisinterface.sip.in	2022-04-30 21:29:42.713993000 +0300
@@ -582,13 +582,6 @@
 .. versionadded:: 3.14
 %End
 
-    virtual QAction *actionAddVectorTileLayer() = 0;
-%Docstring
-Returns the native Add Vector Tile Layer action.
-
-.. versionadded:: 3.14
-%End
-
     virtual QAction *actionAddPointCloudLayer() = 0;
 %Docstring
 Returns the native Add Point Cloud Layer action.
@@ -821,13 +814,6 @@
 Adds a mesh layer to the current project.
 %End
 
-    virtual QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName ) = 0;
-%Docstring
-Adds a vector tile layer to the current project.
-
-.. versionadded:: 3.14
-%End
-
     virtual QgsPointCloudLayer *addPointCloudLayer( const QString &url, const QString &baseName, const QString &providerKey ) = 0;
 %Docstring
 Adds a point cloud layer to the current project.
@@ -1575,16 +1561,6 @@
 .. versionadded:: 3.4
 %End
 
-    virtual void setGpsPanelConnection( QgsGpsConnection *connection ) = 0;
-%Docstring
-Sets a GPS ``connection`` to use within the GPS Panel widget.
-
-Any existing GPS connection used by the widget will be disconnect and replaced with this connection. The connection
-is automatically registered within the :py:func:`QgsApplication.gpsConnectionRegistry()`.
-
-.. versionadded:: 3.16
-%End
-
   signals:
 
     void currentLayerChanged( QgsMapLayer *layer );
