--- a/src/gui/qgsgui.h	2025-05-15 15:26:35.554165935 +0300
+++ b/src/gui/qgsgui.h	2025-05-15 14:47:00.960301136 +0300
@@ -50,7 +50,7 @@
 class QgsProviderSourceWidgetProviderRegistry;
 class QgsRelationWidgetRegistry;
 class QgsMapToolShapeRegistry;
-class QgsHistoryProviderRegistry;
+// class QgsHistoryProviderRegistry;
 class QgsSensorGuiRegistry;
 class QgsSettingsEditorWidgetRegistry;
 class QgsInputControllerManager;
@@ -226,7 +226,7 @@
      * Returns the global history provider registry, used for tracking history providers.
      * \since QGIS 3.24
      */
-    static QgsHistoryProviderRegistry *historyProviderRegistry() SIP_KEEPREFERENCE;
+    // static QgsHistoryProviderRegistry *historyProviderRegistry() SIP_KEEPREFERENCE;
 
     /**
      * Returns the registry of settings editors.
@@ -361,7 +361,7 @@
     QgsProviderSourceWidgetProviderRegistry *mProviderSourceWidgetProviderRegistry = nullptr;
     QgsRelationWidgetRegistry *mRelationEditorRegistry = nullptr;
     QgsMapToolShapeRegistry *mShapeMapToolRegistry = nullptr;
-    QgsHistoryProviderRegistry *mHistoryProviderRegistry = nullptr;
+    // QgsHistoryProviderRegistry *mHistoryProviderRegistry = nullptr;
     QgsSensorGuiRegistry *mSensorGuiRegistry = nullptr;
     QgsSettingsEditorWidgetRegistry *mSettingsEditorRegistry = nullptr;
     QgsInputControllerManager *mInputControllerManager = nullptr;
