--- a/src/gui/qgisinterface.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/qgisinterface.h	2022-04-06 18:40:09.000000000 +0300
@@ -59,7 +59,7 @@
 class QgsRasterLayer;
 class QgsVectorLayer;
 class QgsVectorLayerTools;
-class QgsVectorTileLayer;
+// class QgsVectorTileLayer;
 class QgsPointCloudLayer;
 class QgsOptionsWidgetFactory;
 class QgsLocatorFilter;
@@ -67,7 +67,7 @@
 class QgsMeshLayer;
 class QgsBrowserGuiModel;
 class QgsDevToolWidgetFactory;
-class QgsGpsConnection;
+// class QgsGpsConnection;
 class QgsApplicationExitBlockerInterface;
 class QgsAbstractMapToolHandler;
 
@@ -530,7 +530,7 @@
      * Returns the native Add Vector Tile Layer action.
      * \since QGIS 3.14
      */
-    virtual QAction *actionAddVectorTileLayer() = 0;
+    // virtual QAction *actionAddVectorTileLayer() = 0;
 
     /**
      * Returns the native Add Point Cloud Layer action.
@@ -727,7 +727,7 @@
      * Adds a vector tile layer to the current project.
      * \since QGIS 3.14
      */
-    virtual QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName ) = 0;
+    // virtual QgsVectorTileLayer *addVectorTileLayer( const QString &url, const QString &baseName ) = 0;
 
     /**
      * Adds a point cloud layer to the current project.
@@ -1323,7 +1323,7 @@
      *
      * \since QGIS 3.16
      */
-    virtual void setGpsPanelConnection( QgsGpsConnection *connection ) = 0;
+    // virtual void setGpsPanelConnection( QgsGpsConnection *connection ) = 0;
 
   signals:
 
