--- a/src/gui/auth/qgsautheditorwidgets.cpp	2025-07-15 20:09:37.810546542 +0300
+++ b/src/gui/auth/qgsautheditorwidgets.cpp	2025-07-30 19:58:18.124961513 +0300
@@ -115,8 +115,11 @@
   if ( !QgsApplication::authManager()->isDisabled() )
   {
     wdgtConfigEditor->setRelayMessages( false );
+    /*
     wdgtConfigEditor->setShowUtilitiesButton( false );
     setupUtilitiesMenu();
+    */
+    btnAuthUtilities->hide();
   }
   else
   {
@@ -298,4 +301,3 @@
 {
   return mMsgBar;
 }
-
