--- a/src/gui/qgsproviderguiregistry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/qgsproviderguiregistry.cpp	2022-04-07 00:39:43.000000000 +0300
@@ -25,7 +25,7 @@
 #include "qgslogger.h"
 #include "qgsgdalguiprovider.h"
 #include "qgsogrguiprovider.h"
-#include "qgsvectortileproviderguimetadata.h"
+// #include "qgsvectortileproviderguimetadata.h"
 #include "qgspointcloudproviderguimetadata.h"
 #include "qgsmaplayerconfigwidgetfactory.h"
 
@@ -84,8 +84,8 @@
   QgsProviderGuiMetadata *ogr = new QgsOgrGuiProviderMetadata();
   mProviders[ ogr->key() ] = ogr;
 
-  QgsProviderGuiMetadata *vt = new QgsVectorTileProviderGuiMetadata();
-  mProviders[ vt->key() ] = vt;
+//   QgsProviderGuiMetadata *vt = new QgsVectorTileProviderGuiMetadata();
+//   mProviders[ vt->key() ] = vt;
 
 #ifdef HAVE_EPT
   QgsProviderGuiMetadata *ept = new QgsEptProviderGuiMetadata();
