--- a/src/gui/raster/qgsrasterlayerproperties.cpp	2025-05-15 15:26:35.567575206 +0300
+++ b/src/gui/raster/qgsrasterlayerproperties.cpp	2025-05-15 14:47:00.997302427 +0300
@@ -573,6 +573,10 @@
   mOptsPage_Legend->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#legend-properties" ) );
   mOptsPage_Server->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#server-properties" ) );
 
+  auto qgisServerItems = mOptionsListWidget->findItems("QGIS Server", Qt::MatchExactly);
+  for (auto item : qgisServerItems)
+    item->setHidden(true);
+
   initialize();
 }
 
