--- a/src/gui/raster/qgsrasterlayerproperties.cpp	2025-07-14 17:28:14.963571397 +0300
+++ b/src/gui/raster/qgsrasterlayerproperties.cpp	2025-07-14 13:59:40.922930534 +0300
@@ -584,6 +584,10 @@
   mOptsPage_Metadata->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#metadata-properties" ) );
   mOptsPage_Legend->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#legend-properties" ) );
   mOptsPage_Server->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#server-properties" ) );
+
+  auto qgisServerItems = mOptionsListWidget->findItems("QGIS Server", Qt::MatchExactly);
+  for (auto item : qgisServerItems)
+    item->setHidden(true);
 }
 
 void QgsRasterLayerProperties::addPropertiesPageFactory( const QgsMapLayerConfigWidgetFactory *factory )
