--- a/src/gui/vector/qgsvectorlayerproperties.cpp	2025-05-15 15:26:35.570890910 +0300
+++ b/src/gui/vector/qgsvectorlayerproperties.cpp	2025-05-15 14:47:01.016303089 +0300
@@ -499,6 +499,10 @@
 
   optionsStackedWidget_CurrentChanged( mOptStackedWidget->currentIndex() );
 
+  auto qgisServerItems = mOptionsListWidget->findItems("QGIS Server", Qt::MatchExactly);
+  for (auto item : qgisServerItems)
+    item->setHidden(true);
+
   initialize();
 }
 
