--- a/src/gui/processing/qgsprocessingmultipleselectiondialog.cpp	2025-07-15 20:09:37.864547695 +0300
+++ b/src/gui/processing/qgsprocessingmultipleselectiondialog.cpp	2025-07-30 19:58:18.124961513 +0300
@@ -23,7 +23,7 @@
 #include "qgspluginlayer.h"
 #include "qgspointcloudlayer.h"
 #include "qgsannotationlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsproject.h"
 #include "processing/models/qgsprocessingmodelchildparametersource.h"
 #include <QStandardItemModel>
@@ -572,11 +572,13 @@
 
     case QgsProcessing::TypeVectorTile:
     {
+      /*
       const QList<QgsVectorTileLayer *> options = QgsProcessingUtils::compatibleVectorTileLayers( project, false );
       for ( const QgsVectorTileLayer *layer : options )
       {
         addLayer( layer );
       }
+      */
 
       break;
     }
