--- a/src/gui/qgsgui.h	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/qgsgui.h	2022-04-25 18:33:17.000000000 +0300
@@ -46,7 +46,7 @@
 class QgsProviderSourceWidgetProviderRegistry;
 class QgsRelationWidgetRegistry;
 class QgsMapToolShapeRegistry;
-class QgsHistoryProviderRegistry;
+// class QgsHistoryProviderRegistry;
 class QgsSensorGuiRegistry;
 class QgsSettingsEditorWidgetRegistry;
 
@@ -211,7 +211,7 @@
      * Returns the global history provider registry, used for tracking history providers.
      * \since QGIS 3.24
      */
-    static QgsHistoryProviderRegistry *historyProviderRegistry() SIP_KEEPREFERENCE;
+    // static QgsHistoryProviderRegistry *historyProviderRegistry() SIP_KEEPREFERENCE;
 
     /**
      * Returns the registry of settings editors.
@@ -326,7 +326,7 @@
     QgsProviderSourceWidgetProviderRegistry *mProviderSourceWidgetProviderRegistry = nullptr;
     QgsRelationWidgetRegistry *mRelationEditorRegistry = nullptr;
     QgsMapToolShapeRegistry *mShapeMapToolRegistry = nullptr;
-    QgsHistoryProviderRegistry *mHistoryProviderRegistry = nullptr;
+    // QgsHistoryProviderRegistry *mHistoryProviderRegistry = nullptr;
     QgsSensorGuiRegistry *mSensorGuiRegistry = nullptr;
     QgsSettingsEditorWidgetRegistry *mSettingsEditorRegistry = nullptr;
     std::unique_ptr< QgsWindowManagerInterface > mWindowManager;
