--- a/src/gui/processing/qgsprocessingmultipleselectiondialog.cpp	2025-05-15 15:26:35.542713397 +0300
+++ b/src/gui/processing/qgsprocessingmultipleselectiondialog.cpp	2025-05-15 14:47:00.925299916 +0300
@@ -24,7 +24,7 @@
 #include "qgspluginlayer.h"
 #include "qgspointcloudlayer.h"
 #include "qgsannotationlayer.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsproject.h"
 #include "processing/models/qgsprocessingmodelchildparametersource.h"
 #include <QStandardItemModel>
@@ -709,12 +709,14 @@
 
     case Qgis::ProcessingSourceType::VectorTile:
     {
+      /*
       const QList<QgsVectorTileLayer *> options = QgsProcessingUtils::compatibleVectorTileLayers( project, false );
       for ( const QgsVectorTileLayer *layer : options )
       {
         addLayer( layer );
       }
 
+      */
       break;
     }
 
