--- a/src/gui/raster/qgsrasterlayerproperties.cpp	2023-08-22 11:32:53.431358491 +0300
+++ b/src/gui/raster/qgsrasterlayerproperties.cpp	2023-10-26 01:46:54.262333015 +0300
@@ -532,6 +532,10 @@
   mOptsPage_Metadata->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#metadata-properties" ) );
   mOptsPage_Legend->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#legend-properties" ) );
   mOptsPage_Server->setProperty( "helpPage", QStringLiteral( "working_with_raster/raster_properties.html#server-properties" ) );
+
+  auto qgisServerItems = mOptionsListWidget->findItems("QGIS Server", Qt::MatchExactly);
+  for (auto item : qgisServerItems)
+    item->setHidden(true);
 }
 
 void QgsRasterLayerProperties::addPropertiesPageFactory( const QgsMapLayerConfigWidgetFactory *factory )
