--- a/src/gui/auth/qgsautheditorwidgets.cpp	2025-05-15 15:26:35.526249228 +0300
+++ b/src/gui/auth/qgsautheditorwidgets.cpp	2025-05-15 14:47:00.864297788 +0300
@@ -116,8 +116,9 @@
   if ( !QgsApplication::authManager()->isDisabled() )
   {
     wdgtConfigEditor->setRelayMessages( false );
-    wdgtConfigEditor->setShowUtilitiesButton( false );
-    setupUtilitiesMenu();
+    // wdgtConfigEditor->setShowUtilitiesButton( false );
+    // setupUtilitiesMenu();
+    btnAuthUtilities->hide();
   }
   else
   {
