--- a/src/gui/qgsmanageconnectionsdialog.cpp	2025-07-15 20:09:37.917548827 +0300
+++ b/src/gui/qgsmanageconnectionsdialog.cpp	2025-07-30 19:58:18.125961548 +0300
@@ -24,7 +24,7 @@
 #include "qgsmanageconnectionsdialog.h"
 #include "qgshttpheaders.h"
 #include "qgsowsconnection.h"
-#include "qgsvectortileconnection.h"
+// #include "qgsvectortileconnection.h"
 #include "qgssettingsentryimpl.h"
 #include "qgssettingsentryenumflag.h"
 
@@ -139,6 +139,7 @@
       case XyzTiles:
         doc = saveXyzTilesConnections( items );
         break;
+      /*
       case ArcgisMapServer:
       case ArcgisFeatureServer:
         doc = saveArcgisConnections( items );
@@ -146,6 +147,7 @@
       case VectorTile:
         doc = saveVectorTileConnections( items );
         break;
+      */
     }
 
     QFile file( mFileName );
@@ -214,6 +216,7 @@
       case XyzTiles:
         loadXyzTilesConnections( doc, items );
         break;
+      /*
       case ArcgisMapServer:
         loadArcgisConnections( doc, items, QStringLiteral( "ARCGISMAPSERVER" ) );
         break;
@@ -223,6 +226,7 @@
       case VectorTile:
         loadVectorTileConnections( doc, items );
         break;
+      */
     }
     // clear connections list and close window
     listConnections->clear();
@@ -269,6 +273,7 @@
       case XyzTiles:
         connections = QgsXyzConnectionSettings::sTreeXyzConnections->items();
         break;
+      /*
       case ArcgisMapServer:
       case ArcgisFeatureServer:
         connections = QgsArcGisConnectionSettings::sTreeConnectionArcgis->items();
@@ -276,6 +281,7 @@
       case VectorTile:
         connections = QgsVectorTileProviderConnection::sTreeConnectionVectorTile->items();
         break;
+      */
     }
     for ( const QString &connection : std::as_const( connections ) )
     {
@@ -693,6 +699,7 @@
   return doc;
 }
 
+/*
 QDomDocument QgsManageConnectionsDialog::saveArcgisConnections( const QStringList &connections )
 {
   QDomDocument doc( QStringLiteral( "connections" ) );
@@ -718,7 +725,9 @@
 
   return doc;
 }
+*/
 
+/*
 QDomDocument QgsManageConnectionsDialog::saveVectorTileConnections( const QStringList &connections )
 {
   QDomDocument doc( QStringLiteral( "connections" ) );
@@ -748,6 +757,7 @@
 
   return doc;
 }
+*/
 
 void QgsManageConnectionsDialog::loadOWSConnections( const QDomDocument &doc, const QStringList &items, const QString &service )
 {
@@ -1391,6 +1401,7 @@
   }
 }
 
+/*
 void QgsManageConnectionsDialog::loadArcgisConnections( const QDomDocument &doc, const QStringList &items, const QString &service )
 {
   const QDomElement root = doc.documentElement();
@@ -1472,7 +1483,9 @@
     child = child.nextSiblingElement();
   }
 }
+*/
 
+/*
 void QgsManageConnectionsDialog::loadVectorTileConnections( const QDomDocument &doc, const QStringList &items )
 {
   const QDomElement root = doc.documentElement();
@@ -1559,6 +1572,7 @@
     child = child.nextSiblingElement();
   }
 }
+*/
 
 void QgsManageConnectionsDialog::selectAll()
 {
