--- a/src/gui/processing/qgsprocessingguiregistry.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/processing/qgsprocessingguiregistry.cpp	2022-04-07 00:39:45.000000000 +0300
@@ -18,10 +18,10 @@
 #include "qgsprocessingguiregistry.h"
 #include "qgsprocessingalgorithmconfigurationwidget.h"
 #include "qgsprocessingconfigurationwidgets.h"
-#include "qgsprocessingvectortilewriterlayerswidgetwrapper.h"
+// #include "qgsprocessingvectortilewriterlayerswidgetwrapper.h"
 #include "qgsprocessingfieldmapwidgetwrapper.h"
 #include "qgsprocessingaggregatewidgetwrapper.h"
-#include "qgsprocessingdxflayerswidgetwrapper.h"
+// #include "qgsprocessingdxflayerswidgetwrapper.h"
 #include "qgsprocessingwidgetwrapperimpl.h"
 #include "qgsprocessingtininputlayerswidget.h"
 #include "qgsprocessingmeshdatasetwidget.h"
@@ -67,7 +67,7 @@
   addParameterWidgetFactory( new QgsProcessingMeshLayerWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingBandWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingMultipleLayerWidgetWrapper() );
-  addParameterWidgetFactory( new QgsProcessingVectorTileWriterLayersWidgetWrapper() );
+//   addParameterWidgetFactory( new QgsProcessingVectorTileWriterLayersWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingFeatureSinkWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingVectorDestinationWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingRasterDestinationWidgetWrapper() );
@@ -76,7 +76,7 @@
   addParameterWidgetFactory( new QgsProcessingFieldMapWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingAggregateWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingTinInputLayersWidgetWrapper() );
-  addParameterWidgetFactory( new QgsProcessingDxfLayersWidgetWrapper() );
+//   addParameterWidgetFactory( new QgsProcessingDxfLayersWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingMeshDatasetGroupsWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingMeshDatasetTimeWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingPointCloudLayerWidgetWrapper() );
