--- a/src/gui/qgsbrowserdockwidget_p.cpp	2025-05-15 15:26:35.548137709 +0300
+++ b/src/gui/qgsbrowserdockwidget_p.cpp	2025-05-15 14:47:00.940300439 +0300
@@ -44,7 +44,7 @@
 #include "qgsnative.h"
 #include "qgsmaptoolpan.h"
 #include "qgsvectorlayercache.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsattributetablemodel.h"
 #include "qgsattributetablefiltermodel.h"
 #include "qgsapplication.h"
@@ -217,9 +217,12 @@
 
     case Qgis::LayerType::VectorTile:
     {
+      /*
       QgsDebugMsgLevel( QStringLiteral( "creating vector tile layer" ), 2 );
       mLayer = std::make_unique<QgsVectorTileLayer>( layerItem->uri(), layerItem->name() );
       break;
+      */
+      return;
     }
 
     case Qgis::LayerType::PointCloud:
