--- a/src/gui/qgsbrowserdockwidget_p.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/qgsbrowserdockwidget_p.cpp	2022-04-07 00:39:45.000000000 +0300
@@ -44,7 +44,7 @@
 #include "qgsnative.h"
 #include "qgsmaptoolpan.h"
 #include "qgsvectorlayercache.h"
-#include "qgsvectortilelayer.h"
+// #include "qgsvectortilelayer.h"
 #include "qgsattributetablemodel.h"
 #include "qgsattributetablefiltermodel.h"
 #include "qgsapplication.h"
@@ -213,12 +213,12 @@
       break;
     }
 
-    case QgsMapLayerType::VectorTileLayer:
-    {
-      QgsDebugMsgLevel( QStringLiteral( "creating vector tile layer" ), 2 );
-      mLayer = std::make_unique< QgsVectorTileLayer >( layerItem->uri(), layerItem->name() );
-      break;
-    }
+    // case QgsMapLayerType::VectorTileLayer:
+    // {
+    //   QgsDebugMsgLevel( QStringLiteral( "creating vector tile layer" ), 2 );
+    //   mLayer = std::make_unique< QgsVectorTileLayer >( layerItem->uri(), layerItem->name() );
+    //   break;
+    // }
 
     case QgsMapLayerType::PointCloudLayer:
     {
