--- a/src/gui/qgsgui.cpp	2025-07-15 20:09:37.912548720 +0300
+++ b/src/gui/qgsgui.cpp	2025-07-30 19:58:18.125961548 +0300
@@ -58,10 +58,12 @@
 #include "qgsrelationwidgetregistry.h"
 #include "qgsmaptoolshaperegistry.h"
 #include "qgssettingsregistrygui.h"
-#include "qgshistoryproviderregistry.h"
+// #include "qgshistoryproviderregistry.h"
 #include "qgslayermetadatasourceselectprovider.h"
+/*
 #include "qgssensorguiregistry.h"
 #include "qgshistoryentry.h"
+*/
 
 #include "qgssettingseditorwidgetregistry.h"
 
@@ -180,10 +182,12 @@
   return instance()->mSensorGuiRegistry;
 }
 
+/*
 QgsHistoryProviderRegistry *QgsGui::historyProviderRegistry()
 {
   return instance()->mHistoryProviderRegistry;
 }
+*/
 
 QgsSettingsEditorWidgetRegistry *QgsGui::settingsEditorWidgetRegistry()
 {
@@ -232,7 +236,7 @@
   delete mEditorWidgetRegistry;
   delete mMapLayerActionRegistry;
   delete mSourceSelectProviderRegistry;
-  delete mHistoryProviderRegistry;
+  // delete mHistoryProviderRegistry;
   delete mShortcutsManager;
   delete mNative;
   delete mNumericFormatGuiRegistry;
@@ -302,11 +306,15 @@
   mCodeEditorColorSchemeRegistry = new QgsCodeEditorColorSchemeRegistry();
 
   // provider gui registry initialize QgsProviderRegistry too
+  /*
   mSensorGuiRegistry = new QgsSensorGuiRegistry();
   mSensorGuiRegistry->populate();
+  */
 
+  /*
   mHistoryProviderRegistry = new QgsHistoryProviderRegistry();
   mHistoryProviderRegistry->addDefaultProviders();
+  */
 
   mProviderGuiRegistry = new QgsProviderGuiRegistry( QgsApplication::pluginPath() );
   mProjectStorageGuiRegistry = new QgsProjectStorageGuiRegistry();
@@ -338,7 +346,7 @@
   mProcessingRecentAlgorithmLog = new QgsProcessingRecentAlgorithmLog();
   mProcessingGuiRegistry = new QgsProcessingGuiRegistry();
 
-  qRegisterMetaType< QgsHistoryEntry >( "QgsHistoryEntry" );
+  // qRegisterMetaType< QgsHistoryEntry >( "QgsHistoryEntry" );
 }
 
 bool QgsGui::pythonMacroAllowed( void ( *lambda )(), QgsMessageBar *messageBar )
