--- a/src/gui/processing/qgsprocessinghistorywidget.cpp	2025-05-15 15:26:35.540007252 +0300
+++ b/src/gui/processing/qgsprocessinghistorywidget.cpp	2025-05-15 14:47:00.923299846 +0300
@@ -43,7 +43,7 @@
 {
   if ( QMessageBox::question( this, tr( "Clear History" ), tr( "Are you sure you want to clear the Processing history?" ), QMessageBox::Yes | QMessageBox::No, QMessageBox::No ) == QMessageBox::Yes )
   {
-    QgsGui::historyProviderRegistry()->clearHistory( Qgis::HistoryProviderBackend::LocalProfile, QStringLiteral( "processing" ) );
+    // QgsGui::historyProviderRegistry()->clearHistory( Qgis::HistoryProviderBackend::LocalProfile, QStringLiteral( "processing" ) );
   }
 }
 
@@ -64,7 +64,7 @@
 
   fileName = QgsFileUtils::ensureFileNameHasExtension( fileName, { QStringLiteral( "log" ) } );
 
-  const QList<QgsHistoryEntry> entries = QgsGui::historyProviderRegistry()->queryEntries( QDateTime(), QDateTime(), QStringLiteral( "processing" ) );
+  const QList<QgsHistoryEntry> entries; // = QgsGui::historyProviderRegistry()->queryEntries( QDateTime(), QDateTime(), QStringLiteral( "processing" ) );
 
   const QString logSeparator = QStringLiteral( "|~|" );
   QFile logFile( fileName );
