--- a/src/gui/qgsmanageconnectionsdialog.cpp	2025-05-15 15:26:35.556883026 +0300
+++ b/src/gui/qgsmanageconnectionsdialog.cpp	2025-05-15 14:47:00.966301345 +0300
@@ -25,7 +25,7 @@
 #include "moc_qgsmanageconnectionsdialog.cpp"
 #include "qgshttpheaders.h"
 #include "qgsowsconnection.h"
-#include "qgsvectortileconnection.h"
+// #include "qgsvectortileconnection.h"
 #include "qgssettingsentryimpl.h"
 #include "qgssettingsentryenumflag.h"
 #include "qgstiledsceneconnection.h"
@@ -145,10 +145,10 @@
         break;
       case ArcgisMapServer:
       case ArcgisFeatureServer:
-        doc = saveArcgisConnections( items );
+        // doc = saveArcgisConnections( items );
         break;
       case VectorTile:
-        doc = saveVectorTileConnections( items );
+        // doc = saveVectorTileConnections( items );
         break;
       case TiledScene:
         doc = saveTiledSceneConnections( items );
@@ -221,13 +221,13 @@
         loadXyzTilesConnections( doc, items );
         break;
       case ArcgisMapServer:
-        loadArcgisConnections( doc, items, QStringLiteral( "ARCGISMAPSERVER" ) );
+        // loadArcgisConnections( doc, items, QStringLiteral( "ARCGISMAPSERVER" ) );
         break;
       case ArcgisFeatureServer:
-        loadArcgisConnections( doc, items, QStringLiteral( "ARCGISFEATURESERVER" ) );
+        // loadArcgisConnections( doc, items, QStringLiteral( "ARCGISFEATURESERVER" ) );
         break;
       case VectorTile:
-        loadVectorTileConnections( doc, items );
+        // loadVectorTileConnections( doc, items );
         break;
       case TiledScene:
         loadTiledSceneConnections( doc, items );
@@ -289,10 +289,10 @@
         break;
       case ArcgisMapServer:
       case ArcgisFeatureServer:
-        connections = QgsArcGisConnectionSettings::sTreeConnectionArcgis->items();
+        // connections = QgsArcGisConnectionSettings::sTreeConnectionArcgis->items();
         break;
       case VectorTile:
-        connections = QgsVectorTileProviderConnection::sTreeConnectionVectorTile->items();
+        // connections = QgsVectorTileProviderConnection::sTreeConnectionVectorTile->items();
         break;
       case TiledScene:
         connections = QgsTiledSceneProviderConnection::sTreeConnectionTiledScene->items();
@@ -745,7 +745,7 @@
 
   return doc;
 }
-
+/*
 QDomDocument QgsManageConnectionsDialog::saveArcgisConnections( const QStringList &connections )
 {
   QDomDocument doc( QStringLiteral( "connections" ) );
@@ -774,7 +774,8 @@
 
   return doc;
 }
-
+*/
+/*
 QDomDocument QgsManageConnectionsDialog::saveVectorTileConnections( const QStringList &connections )
 {
   QDomDocument doc( QStringLiteral( "connections" ) );
@@ -929,6 +930,7 @@
 
   return doc;
 }
+*/
 
 void QgsManageConnectionsDialog::loadOWSConnections( const QDomDocument &doc, const QStringList &items, const QString &service )
 {
@@ -1530,7 +1532,7 @@
     child = child.nextSiblingElement();
   }
 }
-
+/*
 void QgsManageConnectionsDialog::loadArcgisConnections( const QDomDocument &doc, const QStringList &items, const QString &service )
 {
   const QDomElement root = doc.documentElement();
@@ -1607,7 +1609,8 @@
     child = child.nextSiblingElement();
   }
 }
-
+*/
+/*
 void QgsManageConnectionsDialog::loadVectorTileConnections( const QDomDocument &doc, const QStringList &items )
 {
   const QDomElement root = doc.documentElement();
@@ -1689,6 +1692,7 @@
     child = child.nextSiblingElement();
   }
 }
+*/
 
 void QgsManageConnectionsDialog::loadTiledSceneConnections( const QDomDocument &doc, const QStringList &items )
 {
