--- a/src/gui/processing/qgsprocessingalgorithmdialogbase.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/gui/processing/qgsprocessingalgorithmdialogbase.cpp	2022-04-07 00:39:43.000000000 +0300
@@ -34,7 +34,7 @@
 #include <QFileDialog>
 #include <QMimeData>
 #include <QMenu>
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 
 
 ///@cond NOT_STABLE
@@ -207,7 +207,7 @@
             return;
 
           const QVariantMap properties = alg->asMap( createProcessingParameters(), *context );
-          const QString json = QString::fromStdString( QgsJsonUtils::jsonFromVariant( properties ).dump( 2 ) );
+          const QString json = QgsJsonUtils::dump(QgsJsonUtils::jsonFromVariant( properties ));
 
           QMimeData *m = new QMimeData();
           m->setText( json );
