--- a/i18n/qgis_ru.ts	2025-05-15 15:26:35.264748633 +0300
+++ b/i18n/qgis_ru.ts	2025-05-15 14:47:00.097026221 +0300
@@ -56687,7 +56687,7 @@
     <message>
         <location filename="../src/app/qgisapp.cpp" line="1703"/>
         <source>Updating recent project paths</source>
-        <translation>Обновление путей предыдщих проектов</translation>
+        <translation>Обновление путей предыдущих проектов</translation>
     </message>
     <message>
         <location filename="../src/app/qgisapp.cpp" line="1705"/>
@@ -176910,4 +176910,4 @@
         <translation>Некорректный формат вывода</translation>
     </message>
 </context>
-</TS>
\ No newline at end of file
+</TS>
