--- a/python/plugins/processing/ProcessingPlugin.py	2025-07-15 20:09:37.326986467 +0300
+++ b/python/plugins/processing/ProcessingPlugin.py	2025-07-30 19:58:18.105960835 +0300
@@ -37,8 +37,8 @@
                        QgsSettings)
 from qgis.gui import (QgsGui,
                       QgsOptionsWidgetFactory,
-                      QgsCustomDropHandler,
-                      QgsProcessingHistoryDialog)
+                      QgsCustomDropHandler,)
+                      # QgsProcessingHistoryDialog)
 from qgis.PyQt.QtCore import (
     QObject,
     Qt,
@@ -205,7 +205,7 @@
     def initGui(self):
         # port old log, ONCE ONLY!
         settings = QgsSettings()
-        if not settings.value("/Processing/hasPortedOldLog", False, bool):
+        if False and not settings.value("/Processing/hasPortedOldLog", False, bool):
             processing_history_provider = QgsGui.historyProviderRegistry().providerById('processing')
             if processing_history_provider:
                 processing_history_provider.portOldLog()
@@ -227,12 +227,14 @@
         self.edit_features_locator_filter = InPlaceAlgorithmLocatorFilter()
         iface.registerLocatorFilter(self.edit_features_locator_filter)
 
+        """
         QgsGui.historyProviderRegistry().providerById('processing').executePython.connect(
             self._execute_history_commands
         )
         QgsGui.historyProviderRegistry().providerById('processing').createTest.connect(
             self.create_test
         )
+        """
 
         self.toolbox = ProcessingToolbox()
         self.iface.addDockWidget(Qt.RightDockWidgetArea, self.toolbox)
@@ -269,6 +271,7 @@
                                             QKeySequence('Ctrl+Alt+G').toString(QKeySequence.NativeText))
         self.menu.addAction(self.modelerAction)
 
+        """
         self.historyAction = QAction(
             QgsApplication.getThemeIcon("/mIconHistory.svg"),
             QCoreApplication.translate('ProcessingPlugin', '&History…'), self.iface.mainWindow())
@@ -278,6 +281,7 @@
                                             QKeySequence('Ctrl+Alt+H').toString(QKeySequence.NativeText))
         self.menu.addAction(self.historyAction)
         self.toolbox.processingToolbar.addAction(self.historyAction)
+        """
 
         self.resultsAction = QAction(
             QgsApplication.getThemeIcon("/processingResult.svg"),
@@ -485,7 +489,7 @@
 
         self.iface.unregisterMainWindowAction(self.toolboxAction)
         self.iface.unregisterMainWindowAction(self.modelerAction)
-        self.iface.unregisterMainWindowAction(self.historyAction)
+        # self.iface.unregisterMainWindowAction(self.historyAction)
         self.iface.unregisterMainWindowAction(self.resultsAction)
 
         self.iface.unregisterOptionsWidgetFactory(self.options_factory)
@@ -504,12 +508,14 @@
             self.iface.projectMenu().removeAction(self.projectMenuSeparator)
             self.projectMenuSeparator = None
 
+        """
         QgsGui.historyProviderRegistry().providerById('processing').executePython.disconnect(
             self._execute_history_commands
         )
         QgsGui.historyProviderRegistry().providerById('processing').createTest.disconnect(
             self.create_test
         )
+        """
 
         Processing.deinitialize()
 
@@ -534,10 +540,12 @@
         else:
             self.resultsDock.show()
 
+    """
     def openHistory(self):
         dlg = QgsProcessingHistoryDialog(self.iface.mainWindow())
         dlg.setAttribute(Qt.WA_DeleteOnClose)
         dlg.show()
+    """
 
     def tr(self, message, disambiguation=None, n=-1):
         return QCoreApplication.translate('ProcessingPlugin', message, disambiguation=disambiguation, n=n)
