--- a/python/plugins/processing/gui/AlgorithmDialog.py	2025-05-15 15:26:35.338220074 +0300
+++ b/python/plugins/processing/gui/AlgorithmDialog.py	2025-05-15 14:47:00.415282130 +0300
@@ -345,9 +345,9 @@
                 )
                 if command_ok:
                     self.history_details["process_command"] = process_command
-                self.history_log_id, _ = QgsGui.historyProviderRegistry().addEntry(
-                    "processing", self.history_details
-                )
+                # self.history_log_id, _ = QgsGui.historyProviderRegistry().addEntry(
+                #     "processing", self.history_details
+                # )
 
                 QgsGui.instance().processingRecentAlgorithmLog().push(
                     self.algorithm().id()
@@ -381,9 +381,9 @@
                         }
                         self.history_details["log"] = self.feedback.htmlLog()
 
-                        QgsGui.historyProviderRegistry().updateEntry(
-                            self.history_log_id, self.history_details
-                        )
+                        # QgsGui.historyProviderRegistry().updateEntry(
+                        #     self.history_log_id, self.history_details
+                        # )
 
                     if self.feedback_dialog is not None:
                         self.feedback_dialog.close()
