--- a/python/plugins/processing/gui/AlgorithmDialog.py	2022-02-18 15:06:10.000000000 +0300
+++ b/python/plugins/processing/gui/AlgorithmDialog.py	2022-05-26 20:06:04.551972000 +0300
@@ -275,7 +275,7 @@
                 process_command, command_ok = self.algorithm().asQgisProcessCommand(parameters, self.context)
                 if command_ok:
                     self.history_details['process_command'] = process_command
-                self.history_log_id, _ = QgsGui.historyProviderRegistry().addEntry('processing', self.history_details)
+                #self.history_log_id, _ = QgsGui.historyProviderRegistry().addEntry('processing', self.history_details)
 
                 QgsGui.instance().processingRecentAlgorithmLog().push(self.algorithm().id())
                 self.cancelButton().setEnabled(self.algorithm().flags() & QgsProcessingAlgorithm.FlagCanCancel)
@@ -294,7 +294,7 @@
 
                     if self.history_log_id is not None:
                         self.history_details['results'] = results
-                        QgsGui.historyProviderRegistry().updateEntry(self.history_log_id, self.history_details)
+                        #QgsGui.historyProviderRegistry().updateEntry(self.history_log_id, self.history_details)
 
                     if self.feedback_dialog is not None:
                         self.feedback_dialog.close()
