--- a/src/providers/postgres/qgspostgresconn.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/providers/postgres/qgspostgresconn.cpp	2022-04-06 18:40:09.000000000 +0300
@@ -38,7 +38,7 @@
 
 #include <climits>
 
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 
 // for htonl
 #ifdef Q_OS_WIN
@@ -1324,7 +1324,7 @@
     }
   }
   const auto j = QgsJsonUtils::jsonFromVariant( value );
-  return quotedString( QString::fromStdString( j.dump() ) );
+  return quotedString( QgsJsonUtils::dump(j) );
 }
 
 PGresult *QgsPostgresConn::PQexec( const QString &query, bool logError, bool retry ) const
