--- a/src/providers/wms/qgswmsprovider.cpp	2025-07-15 20:09:38.056551796 +0300
+++ b/src/providers/wms/qgswmsprovider.cpp	2025-07-30 19:58:18.127961619 +0300
@@ -35,7 +35,7 @@
 #include "qgsrectangle.h"
 #include "qgscoordinatereferencesystem.h"
 #include "qgsmapsettings.h"
-#include "qgsmbtiles.h"
+// #include "qgsmbtiles.h"
 #include "qgsmessageoutput.h"
 #include "qgsmessagelog.h"
 #include "qgsnetworkaccessmanager.h"
@@ -131,7 +131,7 @@
     return;
 
   std::unique_ptr< QgsScopedRuntimeProfile > profile;
-
+  /*
   if ( mSettings.mIsMBTiles )
   {
     if ( QgsApplication::profiler()->groupIsActive( QStringLiteral( "projectload" ) ) )
@@ -144,7 +144,7 @@
       return;
     }
   }
-  else if ( mSettings.mXyz )
+  else */ if ( mSettings.mXyz )
   {
     if ( QgsApplication::profiler()->groupIsActive( QStringLiteral( "projectload" ) ) )
       profile = std::make_unique< QgsScopedRuntimeProfile >( tr( "Setup tile capabilities" ), QStringLiteral( "projectload" ) );
@@ -909,6 +909,7 @@
     else
       tm->viewExtentIntersection( viewExtent, tml, col0, row0, col1, row1 );
 
+    /*
 #ifdef QGISDEBUG
     int n = ( col1 - col0 + 1 ) * ( row1 - row0 + 1 );
     QgsDebugMsgLevel( QStringLiteral( "tile number: %1x%2 = %3" ).arg( col1 - col0 + 1 ).arg( row1 - row0 + 1 ).arg( n ), 3 );
@@ -918,6 +919,7 @@
       return image;
     }
 #endif
+    */
 
     TilePositions tiles;
     for ( int row = row0; row <= row1; row++ )
@@ -954,12 +956,14 @@
     QList<TileImage> tileImages;  // in the correct resolution
     QList<QRectF> missing;  // rectangles (in map coords) of missing tiles for this view
 
+    /*
     std::unique_ptr<QgsMbTiles> mbtilesReader;
     if ( mSettings.mIsMBTiles )
     {
       mbtilesReader.reset( new QgsMbTiles( QUrl( mSettings.mBaseUrl ).path() ) );
       mbtilesReader->open();
     }
+    */
 
     QElapsedTimer t;
     t.start();
@@ -973,6 +977,7 @@
 
       QImage localImage;
 
+      /*
       if ( mbtilesReader && !QgsTileCache::tile( r.url, localImage ) )
       {
         QUrlQuery query( r.url );
@@ -983,6 +988,7 @@
           continue;
         QgsTileCache::insertTile( r.url, img );
       }
+      */
 
       if ( QgsTileCache::tile( r.url, localImage ) )
       {
@@ -1867,6 +1873,8 @@
 
 bool QgsWmsProvider::setupMBTilesCapabilities( const QString &uri )
 {
+  return false;
+  /*
   // if it is MBTiles source, let's prepare the reader to get some metadata
   QgsMbTiles mbtilesReader( QUrl( mSettings.mBaseUrl ).path() );
   if ( !mbtilesReader.open() )
@@ -1913,6 +1921,7 @@
 
   setupXyzCapabilities( uri, sourceExtent, sourceMinZoom, sourceMaxZoom, sourceTilePixelRatio );
   return true;
+  */
 }
 
 
@@ -5299,6 +5308,7 @@
     }
     else
     {
+      /*
       // slower scan, check actual mbtiles format
       QgsMbTiles reader( fileName );
       if ( reader.open() )
@@ -5313,6 +5323,7 @@
           return {details};
         }
       }
+      */
     }
   }
   return {};
