--- a/src/providers/wcs/qgswcsdataitems.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/src/providers/wcs/qgswcsdataitems.cpp	2022-04-06 18:40:09.000000000 +0300
@@ -19,8 +19,8 @@
 #include "qgsdatasourceuri.h"
 #include "qgsowsconnection.h"
 
-#include "qgsgeonodeconnection.h"
-#include "qgsgeonoderequest.h"
+// #include "qgsgeonodeconnection.h"
+// #include "qgsgeonoderequest.h"
 #include "qgssettings.h"
 
 #ifdef HAVE_GUI
@@ -263,6 +263,7 @@
 QVector<QgsDataItem *> QgsWcsDataItemProvider::createDataItems( const QString &path, QgsDataItem *parentItem )
 {
   QVector<QgsDataItem *> items;
+  /*
   if ( path.startsWith( QLatin1String( "geonode:/" ) ) )
   {
     QString connectionName = path.split( '/' ).last();
@@ -302,6 +303,6 @@
       }
     }
   }
-
+  */
   return items;
 }
