--- a/tests/src/gui/testprocessinggui.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/tests/src/gui/testprocessinggui.cpp	2022-04-20 02:30:29.228572000 +0300
@@ -92,8 +92,8 @@
 #include "qgsprocessingparameteraggregate.h"
 #include "qgsprocessingparametertininputlayers.h"
 #include "qgsprocessingtininputlayerswidget.h"
-#include "qgsprocessingparameterdxflayers.h"
-#include "qgsprocessingdxflayerswidgetwrapper.h"
+//#include "qgsprocessingparameterdxflayers.h"
+//#include "qgsprocessingdxflayerswidgetwrapper.h"
 #include "qgsprocessingmeshdatasetwidget.h"
 #include "qgsabstractdatabaseproviderconnection.h"
 #include "qgspluginlayer.h"
@@ -311,7 +311,7 @@
     void testFileOutWrapper();
     void testFolderOutWrapper();
     void testTinInputLayerWrapper();
-    void testDxfLayersWrapper();
+    //void testDxfLayersWrapper();
     void testMeshDatasetWrapperLayerInProject();
     void testMeshDatasetWrapperLayerOutsideProject();
     void testModelGraphicsView();
@@ -9789,6 +9789,7 @@
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'source': 'PointLayerForTin','type': 0,'attributeIndex': -1}]" ) );
 }
 
+/*
 void TestProcessingGui::testDxfLayersWrapper()
 {
   QgsProcessingParameterDxfLayers definition( QStringLiteral( "DXF layers" ) ) ;
@@ -9823,6 +9824,7 @@
   QString valueAsPythonString = definition.valueAsPythonString( value, context );
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'layer': '%1','attributeIndex': -1}]" ).arg( vectorLayer->source() ) );
 }
+*/
 
 void TestProcessingGui::testMeshDatasetWrapperLayerInProject()
 {
