--- atests/src/core/testqgsjsonutils.cpp	2022-02-18 15:06:10.000000000 +0300
+++ btests/src/core/testqgsjsonutils.cpp	2022-04-10 23:47:58.428679700 +0300
@@ -16,7 +16,7 @@
 #include "qgstest.h"
 
 #include <qgsjsonutils.h>
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 
 #include "qgsvectorlayer.h"
 #include "qgsfeature.h"
@@ -38,7 +38,7 @@
   private slots:
     void testStringList();
     void testJsonArray();
-    void testParseJson();
+    //void testParseJson();
     void testIntList();
     void testDoubleList();
     void testExportAttributesJson_data();
@@ -109,6 +109,7 @@
   }
 }
 
+/*
 void TestQgsJsonUtils::testParseJson()
 {
   const QStringList tests {{
@@ -142,6 +143,7 @@
             QString( R"raw(["A string"])raw" ) );
 
 }
+*/
 
 void TestQgsJsonUtils::testIntList()
 {
@@ -196,7 +198,7 @@
     QBENCHMARK
     {
       const json j( QgsJsonUtils::exportAttributesToJsonObject( feature, &vl ) );
-      QCOMPARE( QString::fromStdString( j.dump() ), QStringLiteral( R"raw({"flddbl":2.0,"fldint":1,"fldtxt":"a value"})raw" ) );
+      QCOMPARE( QString::fromStdString( j.ToString() ), QStringLiteral( R"raw({"flddbl":2.0,"fldint":1,"fldtxt":"a value"})raw" ) );
     }
   }
   else // 0.0032
@@ -227,7 +229,7 @@
                             ",\"type\":\"Feature\"}" ) };
 
   const auto j( exporter.exportFeatureToJsonObject( feature ) );
-  QCOMPARE( QString::fromStdString( j.dump() ),  expectedJson );
+  QCOMPARE( QString::fromStdString( j.ToString() ),  expectedJson );
   const auto json = exporter.exportFeature( feature );
   QCOMPARE( json, expectedJson );
 
@@ -242,7 +244,7 @@
 
   feature.setId( 123 );
   const auto jPrecision( exporterPrecision.exportFeatureToJsonObject( feature ) );
-  QCOMPARE( QString::fromStdString( jPrecision.dump() ),  expectedJsonPrecision );
+  QCOMPARE( QString::fromStdString( jPrecision.ToString() ),  expectedJsonPrecision );
   const auto jsonPrecision { exporterPrecision.exportFeature( feature ) };
   QCOMPARE( jsonPrecision, expectedJsonPrecision );
 
@@ -284,7 +286,7 @@
     if ( w.first.startsWith( QLatin1String( "CIRCULARSTRING" ) ) )
     {
       QVERIFY( g.asJson( 3 ).startsWith( w.second ) );
-      QCOMPARE( QString::fromStdString( g.asJsonObject( 3 )["type"].dump() ), QStringLiteral( R"("LineString")" ) );
+      QCOMPARE( QString::fromStdString( g.asJsonObject( 3 )["type"].ToString() ), QStringLiteral( R"("LineString")" ) );
     }
     else
     {
