--- a/tests/src/gui/testprocessinggui.cpp	2025-05-15 15:26:35.595219564 +0300
+++ b/tests/src/gui/testprocessinggui.cpp	2025-05-15 14:47:01.142307483 +0300
@@ -93,8 +93,8 @@
 #include "qgsprocessingparameteraggregate.h"
 #include "qgsprocessingparametertininputlayers.h"
 #include "qgsprocessingtininputlayerswidget.h"
-#include "qgsprocessingparameterdxflayers.h"
-#include "qgsprocessingdxflayerswidgetwrapper.h"
+//#include "qgsprocessingparameterdxflayers.h"
+//#include "qgsprocessingdxflayerswidgetwrapper.h"
 #include "qgsprocessingmeshdatasetwidget.h"
 #include "qgsabstractdatabaseproviderconnection.h"
 #include "qgsprocessingpointcloudexpressionlineedit.h"
@@ -311,7 +311,7 @@
     void testFileOutWrapper();
     void testFolderOutWrapper();
     void testTinInputLayerWrapper();
-    void testDxfLayersWrapper();
+    // void testDxfLayersWrapper();
     void testAlignRasterLayersWrapper();
     void testRasterOptionsWrapper();
     void testMeshDatasetWrapperLayerInProject();
@@ -10331,6 +10331,7 @@
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'source': 'PointLayerForTin','type': 0,'attributeIndex': -1}]" ) );
 }
 
+/*
 void TestProcessingGui::testDxfLayersWrapper()
 {
   QgsProcessingParameterDxfLayers definition( QStringLiteral( "DXF layers" ) );
@@ -10366,6 +10367,7 @@
   QString valueAsPythonString = definition.valueAsPythonString( value, context );
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'layer': '%1','attributeIndex': -1,'overriddenLayerName': '','buildDataDefinedBlocks': True,'dataDefinedBlocksMaximumNumberOfClasses': -1}]" ).arg( vectorLayer->source() ) );
 }
+*/
 
 void TestProcessingGui::testAlignRasterLayersWrapper()
 {
