--- a/tests/src/CMakeLists.txt	2025-07-15 20:09:38.136553505 +0300
+++ b/tests/src/CMakeLists.txt	2025-08-06 12:18:57.516667464 +0300
@@ -26,7 +26,14 @@
       set_tests_properties(${TESTNAME} PROPERTIES LABELS ${ARG_QGIS_TEST_LABELS})
     endif()
 
-    set_tests_properties(${TESTNAME} PROPERTIES FIXTURES_REQUIRED SOURCETREE)
+    if (NOT PROJ_DB_POSITION)
+      string(REGEX REPLACE ".*/(.*)" "\\1" LAST_PART_DB_POSITION "${PROJ_INCLUDE_DIR}")
+      if(${LAST_PART_DB_POSITION} STREQUAL "include")
+        string(REPLACE "/include" "/share/proj" PROJ_DB_POSITION "${PROJ_INCLUDE_DIR}")
+      endif()
+      message("PROJ_LIB database position for CTest set as ${PROJ_DB_POSITION}")
+    endif()
+    set_tests_properties(${TESTNAME} PROPERTIES ENVIRONMENT "PROJ_LIB=${PROJ_DB_POSITION}")
   endmacro()
 
 
@@ -56,10 +63,10 @@
   # Since the tests are not actually installed, but rather
   # run directly from the build/src/tests dir we need to
   # ensure the qgis libs can be found.
-  if (APPLE)
-  # For Mac OS X, the executable must be at the root of the bundle's executable folder
-  #  set (CMAKE_INSTALL_NAME_DIR @executable_path/../../../src/core)
-  endif()
+  # if (APPLE)
+    # For Mac OS X, the executable must be at the root of the bundle's executable folder
+    #  set (CMAKE_INSTALL_NAME_DIR @executable_path/../../../src/core)
+  # endif()
 
   add_subdirectory(core)
   if (WITH_GUI)
@@ -70,7 +77,7 @@
   endif()
   if (WITH_ANALYSIS)
     add_subdirectory(analysis)
-    add_subdirectory(geometry_checker)
+    # add_subdirectory(geometry_checker)
   endif()
   if (WITH_AUTH)
     add_subdirectory(auth)
@@ -86,7 +93,15 @@
   if (WITH_QUICK)
     add_subdirectory(quickgui)
   endif()
-  if (WITH_SERVER)
-    add_subdirectory(server)
-  endif()
+  # if (WITH_SERVER)
+  #     add_subdirectory(server)
+  # endif()
 endif()
+
+get_target_property(_PATH Qt5::qmake IMPORTED_LOCATION)
+get_filename_component(QT_BIN_DIR ${_PATH} PATH)
+get_filename_component(QT_ROOT_DIR ${QT_BIN_DIR} PATH)
+set(QT5_TRANSLATION_DEVDIR ${QT_ROOT_DIR}/translations)
+set(QT5_PLUGINS_DEVDIR ${QT_ROOT_DIR}/plugins)
+
+configure_file(${CMAKE_SOURCE_DIR}/cmake/qt.conf.cmake.tmp.in ${BINARY_DIR}/qt.conf IMMEDIATE @ONLY)
