--- a/tests/src/gui/testqgsvaluerelationwidgetwrapper.cpp	2022-02-18 15:06:10.000000000 +0300
+++ b/tests/src/gui/testqgsvaluerelationwidgetwrapper.cpp	2022-04-07 00:39:43.000000000 +0300
@@ -28,7 +28,7 @@
 #include <QComboBox>
 #include "qgsgui.h"
 #include <gdal_version.h>
-#include <nlohmann/json.hpp>
+// #include <nlohmann/json.hpp>
 
 class TestQgsValueRelationWidgetWrapper : public QObject
 {
