--- a/tests/src/gui/testprocessinggui.cpp	2023-08-21 19:30:51.278480516 +0300
+++ b/tests/src/gui/testprocessinggui.cpp	2023-08-29 15:33:31.826173277 +0300
@@ -93,8 +93,8 @@
 #include "qgsprocessingparameteraggregate.h"
 #include "qgsprocessingparametertininputlayers.h"
 #include "qgsprocessingtininputlayerswidget.h"
-#include "qgsprocessingparameterdxflayers.h"
-#include "qgsprocessingdxflayerswidgetwrapper.h"
+//#include "qgsprocessingparameterdxflayers.h"
+//#include "qgsprocessingdxflayerswidgetwrapper.h"
 #include "qgsprocessingmeshdatasetwidget.h"
 #include "qgsabstractdatabaseproviderconnection.h"
 #include "qgsprocessingpointcloudexpressionlineedit.h"
@@ -314,7 +314,7 @@
     void testFileOutWrapper();
     void testFolderOutWrapper();
     void testTinInputLayerWrapper();
-    void testDxfLayersWrapper();
+//    void testDxfLayersWrapper();
     void testMeshDatasetWrapperLayerInProject();
     void testMeshDatasetWrapperLayerOutsideProject();
     void testModelGraphicsView();
@@ -9910,7 +9910,7 @@
   QString valueAsPythonString = definition.valueAsPythonString( value, context );
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'source': 'PointLayerForTin','type': 0,'attributeIndex': -1}]" ) );
 }
-
+/*
 void TestProcessingGui::testDxfLayersWrapper()
 {
   QgsProcessingParameterDxfLayers definition( QStringLiteral( "DXF layers" ) ) ;
@@ -9945,7 +9945,7 @@
   QString valueAsPythonString = definition.valueAsPythonString( value, context );
   QCOMPARE( valueAsPythonString, QStringLiteral( "[{'layer': '%1','attributeIndex': -1}]" ).arg( vectorLayer->source() ) );
 }
-
+*/
 void TestProcessingGui::testMeshDatasetWrapperLayerInProject()
 {
   QgsProcessingParameterMeshLayer layerDefinition( QStringLiteral( "layer" ), QStringLiteral( "layer" ) );
