--- a/tests/src/app/CMakeLists.txt	2025-07-15 20:09:38.141553611 +0300
+++ b/tests/src/app/CMakeLists.txt	2025-08-13 14:51:04.190754987 +0300
@@ -6,7 +6,15 @@
   ${CMAKE_CURRENT_BINARY_DIR}
 )
 
-
+if(IS_MULTI_CONFIG_GENERATOR)
+    include_directories(
+        ${CMAKE_BINARY_DIR}/src/app/${NG_PREFIX}qgis_app_autogen/include_${CMAKE_BUILD_TYPE}
+    )
+else()
+    include_directories(
+        ${CMAKE_BINARY_DIR}/src/app/${NG_PREFIX}qgis_app_autogen/include
+    )
+endif()
 
 set(TESTS
   testqgisapp.cpp
@@ -17,7 +25,7 @@
   testqgsattributetable.cpp
   testqgsapplocatorfilters.cpp
   testqgsdecorationscalebar.cpp
-  testqgsdwgimportdialog.cpp
+  # testqgsdwgimportdialog.cpp
   testqgsfieldcalculator.cpp
   testqgsmapcanvasdockwidget.cpp
   testqgsmaptooleditannotation.cpp
@@ -28,7 +36,7 @@
   testqgsmaptoolrotatefeature.cpp
   testqgsmaptoolscalefeature.cpp
   testqgsmaptoolcircularstring.cpp
-  testqgsmaptooleditmesh.cpp
+  # testqgsmaptooleditmesh.cpp
   testqgsmaptoolmovefeature.cpp
   testqgsmaptoolellipse.cpp
   testqgsmaptoolcircle.cpp
@@ -45,22 +53,22 @@
   testqgsmaptooltrimextendfeature.cpp
   testqgsprojectproperties.cpp
   testqgsapplayoutvaliditychecks.cpp
-  testqgsmeshcalculatordialog.cpp
-  testqgsgpsintegration.cpp
+  # testqgsmeshcalculatordialog.cpp
+  # testqgsgpsintegration.cpp
   testqgslabelpropertydialog.cpp
   testqgsmergeattributesdialog.cpp
 )
 
 if (HAVE_GEOREFERENCER)
-  set(TESTS ${TESTS} testqgsgeoreferencer.cpp)
+  # set(TESTS ${TESTS} testqgsgeoreferencer.cpp)
 endif()
 
 if (WITH_BINDINGS)
-  set(TESTS ${TESTS} testqgisapppython.cpp)
+  # set(TESTS ${TESTS} testqgisapppython.cpp)
 endif()
 
 foreach(TESTSRC ${TESTS})
-  add_qgis_test(${TESTSRC} MODULE app LINKEDLIBRARIES qgis_app)
+  add_qgis_test(${TESTSRC} MODULE app LINKEDLIBRARIES ${NG_PREFIX}qgis_app)
 endforeach(TESTSRC)
 
 add_subdirectory(maptooladdfeaturepoint)
