--- a/tests/src/core/geometry/CMakeLists.txt	2025-07-15 20:09:38.148553761 +0300
+++ b/tests/src/core/geometry/CMakeLists.txt	2025-08-06 15:47:27.447970549 +0300
@@ -15,29 +15,29 @@
 
 set(TESTS
   testqgscircle.cpp
-  testqgscircularstring.cpp
-  testqgscompoundcurve.cpp
-  testqgscurvepolygon.cpp
+  # testqgscircularstring.cpp
+  # testqgscompoundcurve.cpp
+  # testqgscurvepolygon.cpp
   testqgsellipse.cpp
-  testqgslinestring.cpp
-  testqgsmulticurve.cpp
-  testqgsmultilinestring.cpp
-  testqgsmultipoint.cpp
-  testqgsmultipolygon.cpp
-  testqgsmultisurface.cpp
-  testqgspoint.cpp
-  testqgspointxy.cpp
-  testqgspolygon.cpp
+  # testqgslinestring.cpp
+  # testqgsmulticurve.cpp
+  # testqgsmultilinestring.cpp
+  # testqgsmultipoint.cpp
+  # testqgsmultipolygon.cpp
+  # testqgsmultisurface.cpp
+  # testqgspoint.cpp
+  # testqgspointxy.cpp
+  # testqgspolygon.cpp
   testqgsquadrilateral.cpp
-  testqgsrectangle.cpp
+  # testqgsrectangle.cpp
   testqgsregularpolygon.cpp
   testqgstriangle.cpp
   )
 
 foreach(TESTSRC ${TESTS})
-  add_qgis_test(${TESTSRC} MODULE core_geometry LINKEDLIBRARIES qgis_core)
+  add_qgis_test(${TESTSRC} MODULE core_geometry LINKEDLIBRARIES ${NG_PREFIX}qgis_core)
 endforeach(TESTSRC)
 
-add_qgis_test(testqgsgeometry.cpp MODULE core LINKEDLIBRARIES qgis_core)
-add_qgis_test(testqgsgeometrycollection.cpp MODULE core LINKEDLIBRARIES qgis_core)
-add_qgis_test(testqgsgeometryutils.cpp MODULE core LINKEDLIBRARIES qgis_core)
+# add_qgis_test(testqgsgeometry.cpp MODULE core LINKEDLIBRARIES ${NG_PREFIX}qgis_core)
+# add_qgis_test(testqgsgeometrycollection.cpp MODULE core LINKEDLIBRARIES ${NG_PREFIX}qgis_core)
+add_qgis_test(testqgsgeometryutils.cpp MODULE core LINKEDLIBRARIES ${NG_PREFIX}qgis_core)
