--- a/src/test/qgstest.h	2025-05-15 15:26:35.579536177 +0300
+++ b/src/test/qgstest.h	2025-05-15 14:47:01.059304589 +0300
@@ -20,7 +20,7 @@
 #include <QDir>
 #include <QFile>
 #include <QTextStream>
-#include <QDesktopServices>
+//#include <QDesktopServices>
 
 #include "qgsapplication.h"
 
@@ -509,8 +509,8 @@
         stream << report;
         file.close();
 
-        if ( !isCIRun() )
-          QDesktopServices::openUrl( QStringLiteral( "file:///%1" ).arg( reportFile ) );
+        // if ( !isCIRun() )
+        //   QDesktopServices::openUrl( QStringLiteral( "file:///%1" ).arg( reportFile ) );
       }
     }
 
