--- a/src/test/qgstest.h	2025-07-15 20:09:38.077552245 +0300
+++ b/src/test/qgstest.h	2025-07-30 19:58:18.127961619 +0300
@@ -20,7 +20,7 @@
 #include <QDir>
 #include <QFile>
 #include <QTextStream>
-#include <QDesktopServices>
+// #include <QDesktopServices>
 
 #include "qgsapplication.h"
 
@@ -239,8 +239,10 @@
         stream << report;
         file.close();
 
+        /*
         if ( !isCIRun() )
           QDesktopServices::openUrl( QStringLiteral( "file:///%1" ).arg( reportFile ) );
+        */
       }
     }
 
