--- a/tests/src/analysis/CMakeLists.txt	2025-07-15 20:09:38.136553505 +0300
+++ b/tests/src/analysis/CMakeLists.txt	2025-08-06 15:42:09.921728110 +0300
@@ -20,17 +20,17 @@
 set(TESTS
  testqgsgeometrysnapper.cpp
  testqgsinterpolator.cpp
- testqgsprocessingalgspt1.cpp
- testqgsprocessingalgspt2.cpp
- testqgsprocessingmodelalgorithm.cpp
+ # testqgsprocessingalgspt1.cpp
+ # testqgsprocessingalgspt2.cpp
+ # testqgsprocessingmodelalgorithm.cpp
  testqgszonalstatistics.cpp
- testqgsrastercalculator.cpp
+ # testqgsrastercalculator.cpp
  testqgsreclassifyutils.cpp
- testqgsalignraster.cpp
+ # testqgsalignraster.cpp
  testqgsnetworkanalysis.cpp
  testqgsninecellfilters.cpp
- testqgsmeshcalculator.cpp
- testqgsmeshcontours.cpp
+ # testqgsmeshcalculator.cpp
+ # testqgsmeshcontours.cpp
  testqgstriangulation.cpp
 )
 if(HAVE_GSL)
@@ -42,6 +42,6 @@
 endif()
 
 foreach(TESTSRC ${TESTS})
-    add_qgis_test(${TESTSRC} MODULE analysis LINKEDLIBRARIES qgis_analysis)
+    add_qgis_test(${TESTSRC} MODULE analysis LINKEDLIBRARIES ${NG_PREFIX}qgis_analysis)
 endforeach(TESTSRC)
-add_qgis_test(testqgsprocessing.cpp MODULE analysis LINKEDLIBRARIES qgis_analysis LABELS "POSTGRES")
+# add_qgis_test(testqgsprocessing.cpp MODULE analysis LINKEDLIBRARIES ${NG_PREFIX}qgis_analysis LABELS "POSTGRES")
