--- a/src/ui/qgsoptionsbase.ui	2025-07-14 17:28:14.980054478 +0300
+++ b/src/ui/qgsoptionsbase.ui	2025-07-14 13:59:40.999932274 +0300
@@ -5154,6 +5154,7 @@
            </item>
           </layout>
          </widget>
+         <widget class="NGAuthOptions" name="mOptionsPageNextGIS" native="true"/>
         </widget>
        </item>
       </layout>
@@ -5311,6 +5312,12 @@
    <header>qgsautheditorwidgets.h</header>
    <container>1</container>
   </customwidget>
+  <customwidget>
+   <class>NGAuthOptions</class>
+   <extends>QWidget</extends>
+   <header>options/ngauthoptions.h</header>
+   <container>1</container>
+  </customwidget>
  </customwidgets>
  <tabstops>
   <tabstop>mSearchLineEdit</tabstop>
