--- a/src/ui/qgsoptionsbase.ui	2025-05-15 15:26:35.585750729 +0300
+++ b/src/ui/qgsoptionsbase.ui	2025-05-15 14:47:01.074305112 +0300
@@ -5267,6 +5267,7 @@
            </item>
           </layout>
          </widget>
+         <widget class="NGAuthOptions" name="mOptionsPageNextGIS" native="true"/>
         </widget>
        </item>
       </layout>
@@ -5424,6 +5425,12 @@
    <header>qgsautheditorwidgets.h</header>
    <container>1</container>
   </customwidget>
+  <customwidget>
+   <class>NGAuthOptions</class>
+   <extends>QWidget</extends>
+   <header>options/ngauthoptions.h</header>
+   <container>1</container>
+  </customwidget>
  </customwidgets>
  <tabstops>
   <tabstop>mOptionsTreeView</tabstop>
